// ng-cosmos-td-ui/src/base/cmMenuTree/cmMenuTree.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const Subject_1 = require("rxjs/Subject");
let CmMenuTreeService = class CmMenuTreeService {
    constructor() {
        this.grabbleSource = new Subject_1.Subject();
        this.Radio = new Subject_1.Subject();
        this.data = new Subject_1.Subject();
        this.disabledchange = new Subject_1.Subject();
        this.menu = new Subject_1.Subject();
        this.click = new Subject_1.Subject();
        this.selectRadio = null;
        this.missionGrabble$ = this.grabbleSource.asObservable();
        this.missiongetRadio$ = this.Radio.asObservable();
        this.missiondataChange$ = this.data.asObservable();
        this.missiondisabled$ = this.disabledchange.asObservable();
        this.missioncreateMenu$ = this.menu.asObservable();
        this.missionclickItems$ = this.click.asObservable();
    }
    clickItems(select) {
        this.click.next(select);
    }
    createMenu(value, select) {
        let obj = {};
        obj["type"] = value;
        obj["select"] = select;
        this.menu.next(obj);
    }
    disabledChange(value) {
        this.disabledchange.next(value);
    }
    moveFile(select, data) {
        let obj = {};
        obj["select"] = select;
        obj["data"] = data;
        this.grabbleSource.next(obj);
    }
    getRadio(value) {
        this.Radio.next(value);
    }
    dataChange(type, select) {
        // console.log(type,select);
        let obj = {};
        obj["type"] = type;
        obj["select"] = select;
        this.data.next(obj);
    }
};
CmMenuTreeService = __decorate([
    core_1.Injectable()
], CmMenuTreeService);
exports.CmMenuTreeService = CmMenuTreeService;
