// ng-cosmos-td-ui/src/cosmos.all.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var CosmosAllModule_1;
const common_1 = require("@angular/common");
const forms_1 = require("@angular/forms");
const core_1 = require("@angular/core");
const base_component_module_1 = require("./base/base.component.module");
const business_component_module_1 = require("./business/business.component.module");
const chart_component_module_1 = require("./chart/chart.component.module");
// 导出基础组件和模块
__export(require("./base/base.component.module"));
// 导出业务组件和模块
__export(require("./business/business.component.module"));
// 导出图表组件和模块
__export(require("./chart/chart.component.module"));
var Menutree_service_1 = require("./base/cmMenuTree/Menutree.service");
exports.MenuTreeService = Menutree_service_1.MenuTreeService;
let CosmosAllModule = CosmosAllModule_1 = class CosmosAllModule {
    static forRoot(options) {
        return {
            ngModule: CosmosAllModule_1
        };
    }
    static forChild() {
        return {
            ngModule: CosmosAllModule_1
        };
    }
};
CosmosAllModule = CosmosAllModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule,
            forms_1.FormsModule,
            base_component_module_1.BaseComponentModule.forRoot(),
            chart_component_module_1.ChartComponentModule.forRoot(),
            business_component_module_1.BusinessComponentModule.forRoot()
        ],
        exports: [
            ...base_component_module_1.baseComponentModuleArray,
            ...business_component_module_1.businessComponentModuleArray,
            ...chart_component_module_1.chartComponentModuleArray
        ]
    })
], CosmosAllModule);
exports.CosmosAllModule = CosmosAllModule;
