// ng-cosmos-td-ui/src/cosmos.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var CosmosModule_1;
const core_1 = require("@angular/core");
const ng_cs_antd_1 = require("ng-cs-antd");
var MenuTree_service_1 = require("./base/cmMenuTree/MenuTree.service");
exports.MenuTreeService = MenuTree_service_1.MenuTreeService;
exports.iconConfig = {
    extraFontName: "string",
    extraFontUrl: "string"
};
// 导出antd服务
var ng_cs_antd_2 = require("ng-cs-antd");
exports.CmNotificationService = ng_cs_antd_2.NzNotificationService;
exports.CmMessageService = ng_cs_antd_2.NzMessageService;
exports.CmModalService = ng_cs_antd_2.NzModalService;
let CosmosModule = CosmosModule_1 = class CosmosModule {
    static forRoot(options) {
        return {
            ngModule: CosmosModule_1
        };
    }
};
CosmosModule = CosmosModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng_cs_antd_1.NgZorroAntdModule.forRoot(exports.iconConfig)
        ]
    })
], CosmosModule);
exports.CosmosModule = CosmosModule;
