// ng-cosmos-td-ui/src/base/base.component.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var BaseComponentModule_1;
const core_1 = require("@angular/core");
const ng_cs_antd_1 = require("ng-cs-antd");
// 导出组件模块
const button_module_1 = require("./button/button.module");
const badge_module_1 = require("./badge/badge.module");
const breadcrumb_module_1 = require("./breadcrumb/breadcrumb.module");
const card_module_1 = require("./card/card.module");
const carousel_module_1 = require("./carousel/carousel.module");
const dropdown_module_1 = require("./dropdown/dropdown.module");
const pagination_module_1 = require("./pagination/pagination.module");
const popconfirm_module_1 = require("./popconfirm/popconfirm.module");
const progress_module_1 = require("./progress/progress.module");
const radio_module_1 = require("./radio/radio.module");
const select_module_1 = require("./select/select.module");
const slider_module_1 = require("./slider/slider.module");
const spin_module_1 = require("./spin/spin.module");
const switch_module_1 = require("./switch/switch.module");
const timeline_module_1 = require("./timeline/timeline.module");
const timePicker_module_1 = require("./timePicker/timePicker.module");
const tooltip_module_1 = require("./tooltip/tooltip.module");
const tag_module_1 = require("./tag/tag.module");
const tabs_module_1 = require("./tabs/tabs.module");
const table_module_1 = require("./table/table.module");
const steps_module_1 = require("./steps/steps.module");
const rate_module_1 = require("./rate/rate.module");
const popover_module_1 = require("./popover/popover.module");
const modal_module_1 = require("./modal/modal.module");
const message_module_1 = require("./message/message.module");
const menu_module_1 = require("./menu/menu.module");
const inputNumber_module_1 = require("./inputNumber/inputNumber.module");
const input_module_1 = require("./input/input.module");
const form_module_1 = require("./form/form.module");
const datePicker_module_1 = require("./datePicker/datePicker.module");
const collapse_module_1 = require("./collapse/collapse.module");
const checkbox_module_1 = require("./checkbox/checkbox.module");
const calendar_module_1 = require("./calendar/calendar.module");
const avatar_module_1 = require("./avatar/avatar.module");
const alert_module_1 = require("./alert/alert.module");
const affix_module_1 = require("./affix/affix.module");
const code_highlight_module_1 = require("./code-highlight/code-highlight.module");
const cmtTable_module_1 = require("./cmtTable/cmtTable.module");
const cmDatePickerShortcut_module_1 = require("./cmDatePickerShortcut/cmDatePickerShortcut.module");
const cmMenuTree_module_1 = require("./cmMenuTree/cmMenuTree.module");
const upload_module_1 = require("./upload/upload.module");
const cascader_module_1 = require("./cascader/cascader.module");
// 导出组件
var affix_component_1 = require("./affix/affix.component");
exports.AffixComponent = affix_component_1.AffixComponent;
var alert_component_1 = require("./alert/alert.component");
exports.AlertComponent = alert_component_1.AlertComponent;
var avatar_component_1 = require("./avatar/avatar.component");
exports.AvatarComponent = avatar_component_1.AvatarComponent;
var badge_component_1 = require("./badge/badge.component");
exports.BadgeComponent = badge_component_1.BadgeComponent;
var breadcrumb_component_1 = require("./breadcrumb/breadcrumb.component");
exports.BreadcrumbComponent = breadcrumb_component_1.BreadcrumbComponent;
var button_component_1 = require("./button/button.component");
exports.ButtonComponent = button_component_1.ButtonComponent;
var calendar_component_1 = require("./calendar/calendar.component");
exports.CalendarComponent = calendar_component_1.CalendarComponent;
var card_component_1 = require("./card/card.component");
exports.CardComponent = card_component_1.CardComponent;
var carousel_component_1 = require("./carousel/carousel.component");
exports.CarouselComponent = carousel_component_1.CarouselComponent;
var checkbox_component_1 = require("./checkbox/checkbox.component");
exports.CheckboxComponent = checkbox_component_1.CheckboxComponent;
var collapse_component_1 = require("./collapse/collapse.component");
exports.CollapseComponent = collapse_component_1.CollapseComponent;
var datePicker_component_1 = require("./datePicker/datePicker.component");
exports.datePickerComponent = datePicker_component_1.datePickerComponent;
var dropdown_component_1 = require("./dropdown/dropdown.component");
exports.DropdownComponent = dropdown_component_1.DropdownComponent;
var form_component_1 = require("./form/form.component");
exports.FormComponent = form_component_1.FormComponent;
var input_component_1 = require("./input/input.component");
exports.InputComponent = input_component_1.InputComponent;
var inputNumber_component_1 = require("./inputNumber/inputNumber.component");
exports.InputNumberComponent = inputNumber_component_1.InputNumberComponent;
var menu_component_1 = require("./menu/menu.component");
exports.MenuComponent = menu_component_1.MenuComponent;
var message_component_1 = require("./message/message.component");
exports.MessageComponent = message_component_1.MessageComponent;
var modal_component_1 = require("./modal/modal.component");
exports.ModalComponent = modal_component_1.ModalComponent;
var pagination_component_1 = require("./pagination/pagination.component");
exports.PaginationComponent = pagination_component_1.PaginationComponent;
var popconfirm_component_1 = require("./popconfirm/popconfirm.component");
exports.PopconfirmComponent = popconfirm_component_1.PopconfirmComponent;
var popover_component_1 = require("./popover/popover.component");
exports.PopoverComponent = popover_component_1.PopoverComponent;
var progress_component_1 = require("./progress/progress.component");
exports.ProgressComponent = progress_component_1.ProgressComponent;
var radio_component_1 = require("./radio/radio.component");
exports.RadioComponent = radio_component_1.RadioComponent;
var rate_component_1 = require("./rate/rate.component");
exports.RateComponent = rate_component_1.RateComponent;
var select_component_1 = require("./select/select.component");
exports.SelectComponent = select_component_1.SelectComponent;
var slider_component_1 = require("./slider/slider.component");
exports.SliderComponent = slider_component_1.SliderComponent;
var spin_component_1 = require("./spin/spin.component");
exports.SpinComponent = spin_component_1.SpinComponent;
var steps_component_1 = require("./steps/steps.component");
exports.StepsComponent = steps_component_1.StepsComponent;
var switch_component_1 = require("./switch/switch.component");
exports.SwitchComponent = switch_component_1.SwitchComponent;
var table_component_1 = require("./table/table.component");
exports.TableComponent = table_component_1.TableComponent;
var tabs_component_1 = require("./tabs/tabs.component");
exports.TabsComponent = tabs_component_1.TabsComponent;
var tag_component_1 = require("./tag/tag.component");
exports.TagComponent = tag_component_1.TagComponent;
var timeline_component_1 = require("./timeline/timeline.component");
exports.TimelineComponent = timeline_component_1.TimelineComponent;
var timePicker_component_1 = require("./timePicker/timePicker.component");
exports.TimePickerComponent = timePicker_component_1.TimePickerComponent;
var tooltip_component_1 = require("./tooltip/tooltip.component");
exports.TooltipComponent = tooltip_component_1.TooltipComponent;
var transfer_component_1 = require("./transfer/transfer.component");
exports.TransferComponent = transfer_component_1.TransferComponent;
var code_highlight_component_1 = require("./code-highlight/code-highlight.component");
exports.CodeHighlightComponent = code_highlight_component_1.CodeHighlightComponent;
var cmtTable_component_1 = require("./cmtTable/cmtTable.component");
exports.cmtTableComponent = cmtTable_component_1.cmtTableComponent;
var cmDatePickerShortcut_component_1 = require("./cmDatePickerShortcut/cmDatePickerShortcut.component");
exports.CmDatePickerShortcutComponent = cmDatePickerShortcut_component_1.CmDatePickerShortcutComponent;
var cmForm_component_1 = require("./cmForm/cmForm.component");
exports.cmFormComponent = cmForm_component_1.cmFormComponent;
var upload_component_1 = require("./upload/upload.component");
exports.UploadComponent = upload_component_1.UploadComponent;
var cascader_component_1 = require("./cascader/cascader.component");
exports.CascaderComponent = cascader_component_1.CascaderComponent;
var cmMenuTree_component_1 = require("./cmMenuTree/cmMenuTree.component");
exports.cmMenuTreeComponent = cmMenuTree_component_1.cmMenuTreeComponent;
// 导出组件模块
var button_module_2 = require("./button/button.module");
exports.ButtonModule = button_module_2.ButtonModule;
var badge_module_2 = require("./badge/badge.module");
exports.BadgeModule = badge_module_2.BadgeModule;
var breadcrumb_module_2 = require("./breadcrumb/breadcrumb.module");
exports.BreadcrumbModule = breadcrumb_module_2.BreadcrumbModule;
var card_module_2 = require("./card/card.module");
exports.CardModule = card_module_2.CardModule;
var carousel_module_2 = require("./carousel/carousel.module");
exports.CarouselModule = carousel_module_2.CarouselModule;
var dropdown_module_2 = require("./dropdown/dropdown.module");
exports.DropdownModule = dropdown_module_2.DropdownModule;
var pagination_module_2 = require("./pagination/pagination.module");
exports.PaginationModule = pagination_module_2.PaginationModule;
var popconfirm_module_2 = require("./popconfirm/popconfirm.module");
exports.PopconfirmModule = popconfirm_module_2.PopconfirmModule;
var progress_module_2 = require("./progress/progress.module");
exports.ProgressModule = progress_module_2.ProgressModule;
var radio_module_2 = require("./radio/radio.module");
exports.RadioModule = radio_module_2.RadioModule;
var select_module_2 = require("./select/select.module");
exports.SelectModule = select_module_2.SelectModule;
var slider_module_2 = require("./slider/slider.module");
exports.SliderModule = slider_module_2.SliderModule;
var spin_module_2 = require("./spin/spin.module");
exports.SpinModule = spin_module_2.SpinModule;
var switch_module_2 = require("./switch/switch.module");
exports.SwitchModule = switch_module_2.SwitchModule;
var timeline_module_2 = require("./timeline/timeline.module");
exports.TimelineModule = timeline_module_2.TimelineModule;
var timePicker_module_2 = require("./timePicker/timePicker.module");
exports.TimePickerModule = timePicker_module_2.TimePickerModule;
var tooltip_module_2 = require("./tooltip/tooltip.module");
exports.TooltipModule = tooltip_module_2.TooltipModule;
var tag_module_2 = require("./tag/tag.module");
exports.TagModule = tag_module_2.TagModule;
var tabs_module_2 = require("./tabs/tabs.module");
exports.TabsModule = tabs_module_2.TabsModule;
var table_module_2 = require("./table/table.module");
exports.TableModule = table_module_2.TableModule;
var steps_module_2 = require("./steps/steps.module");
exports.StepsModule = steps_module_2.StepsModule;
var rate_module_2 = require("./rate/rate.module");
exports.RateModule = rate_module_2.RateModule;
var popover_module_2 = require("./popover/popover.module");
exports.PopoverModule = popover_module_2.PopoverModule;
var modal_module_2 = require("./modal/modal.module");
exports.ModalModule = modal_module_2.ModalModule;
var message_module_2 = require("./message/message.module");
exports.MessageModule = message_module_2.MessageModule;
var menu_module_2 = require("./menu/menu.module");
exports.MenuModule = menu_module_2.MenuModule;
var inputNumber_module_2 = require("./inputNumber/inputNumber.module");
exports.InputNumberModule = inputNumber_module_2.InputNumberModule;
var input_module_2 = require("./input/input.module");
exports.InputModule = input_module_2.InputModule;
var form_module_2 = require("./form/form.module");
exports.FormModule = form_module_2.FormModule;
var datePicker_module_2 = require("./datePicker/datePicker.module");
exports.datePickerModule = datePicker_module_2.datePickerModule;
var collapse_module_2 = require("./collapse/collapse.module");
exports.CollapseModule = collapse_module_2.CollapseModule;
var checkbox_module_2 = require("./checkbox/checkbox.module");
exports.CheckboxModule = checkbox_module_2.CheckboxModule;
var calendar_module_2 = require("./calendar/calendar.module");
exports.CalendarModule = calendar_module_2.CalendarModule;
var avatar_module_2 = require("./avatar/avatar.module");
exports.AvatarModule = avatar_module_2.AvatarModule;
var alert_module_2 = require("./alert/alert.module");
exports.AlertModule = alert_module_2.AlertModule;
var affix_module_2 = require("./affix/affix.module");
exports.AffixModule = affix_module_2.AffixModule;
var code_highlight_module_2 = require("./code-highlight/code-highlight.module");
exports.CodeHighlightModule = code_highlight_module_2.CodeHighlightModule;
var cmtTable_module_2 = require("./cmtTable/cmtTable.module");
exports.cmtTableModule = cmtTable_module_2.cmtTableModule;
var cmDatePickerShortcut_module_2 = require("./cmDatePickerShortcut/cmDatePickerShortcut.module");
exports.CmDatePickerShortcutModule = cmDatePickerShortcut_module_2.CmDatePickerShortcutModule;
var cmMenuTree_module_2 = require("./cmMenuTree/cmMenuTree.module");
exports.cmMenuTreeModule = cmMenuTree_module_2.cmMenuTreeModule;
var upload_module_2 = require("./upload/upload.module");
exports.UploadModule = upload_module_2.UploadModule;
var cascader_module_2 = require("./cascader/cascader.module");
exports.CascaderModule = cascader_module_2.CascaderModule;
// 导出antd
var ng_cs_antd_2 = require("ng-cs-antd");
exports.zhCN = ng_cs_antd_2.zhCN;
exports.enUS = ng_cs_antd_2.enUS;
exports.CmNotificationService = ng_cs_antd_2.NzNotificationService;
exports.CmMessageService = ng_cs_antd_2.NzMessageService;
exports.CmModalService = ng_cs_antd_2.NzModalService;
let BaseComponentModule = BaseComponentModule_1 = class BaseComponentModule {
    static forRoot() {
        return {
            ngModule: BaseComponentModule_1
        };
    }
};
BaseComponentModule = BaseComponentModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng_cs_antd_1.NgZorroAntdModule.forRoot()
        ]
    })
], BaseComponentModule);
exports.BaseComponentModule = BaseComponentModule;
exports.baseComponentModuleArray = [
    affix_module_1.AffixModule,
    alert_module_1.AlertModule,
    avatar_module_1.AvatarModule,
    badge_module_1.BadgeModule,
    breadcrumb_module_1.BreadcrumbModule,
    button_module_1.ButtonModule,
    calendar_module_1.CalendarModule,
    card_module_1.CardModule,
    carousel_module_1.CarouselModule,
    checkbox_module_1.CheckboxModule,
    collapse_module_1.CollapseModule,
    datePicker_module_1.datePickerModule,
    dropdown_module_1.DropdownModule,
    form_module_1.FormModule,
    input_module_1.InputModule,
    inputNumber_module_1.InputNumberModule,
    menu_module_1.MenuModule,
    message_module_1.MessageModule,
    modal_module_1.ModalModule,
    pagination_module_1.PaginationModule,
    popconfirm_module_1.PopconfirmModule,
    popover_module_1.PopoverModule,
    progress_module_1.ProgressModule,
    radio_module_1.RadioModule,
    rate_module_1.RateModule,
    select_module_1.SelectModule,
    slider_module_1.SliderModule,
    spin_module_1.SpinModule,
    steps_module_1.StepsModule,
    switch_module_1.SwitchModule,
    table_module_1.TableModule,
    tabs_module_1.TabsModule,
    tag_module_1.TagModule,
    timeline_module_1.TimelineModule,
    timePicker_module_1.TimePickerModule,
    tooltip_module_1.TooltipModule,
    code_highlight_module_1.CodeHighlightModule,
    cmtTable_module_1.cmtTableModule,
    cmDatePickerShortcut_module_1.CmDatePickerShortcutModule,
    cmMenuTree_module_1.cmMenuTreeModule,
    upload_module_1.UploadModule,
    cascader_module_1.CascaderModule
];
