// ng-cosmos-td-ui/src/business/tree/treeView/treeView.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
let TreeViewComponent = class TreeViewComponent {
    constructor() {
        this.isOnly = false;
    }
    set cmData(cmData) {
        this._cmData = cmData;
    }
    ngOnInit() {
        // console.log(this._cmData,this._cmData.length)
        if (this._cmData.length == 1) {
            this.isOnly = true;
        }
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], TreeViewComponent.prototype, "cmData", null);
TreeViewComponent = __decorate([
    core_1.Component({
        selector: 'cm-treeView',
        template: `

      <div *ngFor="let comment of _cmData" class="ng-star-0" [ngClass]="isOnly?'only':''">
      	<div class="treeView">
      		<div class="treeView-body">
      			<div class="treeView-box" [ngClass]="comment.cComments != false&&comment.cComments.isShow? 'treeview-box':'' ">
      				{{ comment.content }}
      				<div class="treeView-btn" *ngIf="comment.cComments != false">
      					<i class="anticon anticon-plus" (click)="comment.cComments.isShow = true" *ngIf="!comment.cComments.isShow"></i>
      					<i class="anticon anticon-minus" (click)="comment.cComments.isShow = false" *ngIf="comment.cComments.isShow"></i>
      				</div>
      			</div>
      		</div>
      		<cm-treeView *ngIf="comment.cComments != false&&comment.cComments.isShow" [cmData]="comment.cComments" class="treeView-item"></cm-treeView>
      	</div>
      </div>
    `,
        styles: [`
      .treeView-body {
        display: flex;
        justify-content: center;
      }
      .ng-star-0.ng-star-inserted {
        position: relative;
      }
      .ng-star-0.ng-star-inserted:before {
        content: '';
        position: absolute;
        top: -10px;
        left: 0;
        background: #ddd;
        width: 100%;
        height: 2px;
      }
      .ng-star-0.ng-star-inserted:first-child:before {
        width: 50%;
        left: 50%;
      }
      .ng-star-0.ng-star-inserted:last-child:before {
        width: 50%;
      }
      .only.ng-star-0.ng-star-inserted:before {
        width: 0;
      }
      .ng-star-0.ng-star-inserted:after {
        content: '';
        position: absolute;
        top: -10px;
        left: 50%;
        background: #ddd;
        width: 2px;
        height: 17px;
      }
      .treeview-box:before {
        content: '';
        position: absolute;
        top: 100%;
        left: 50%;
        background: #ddd;
        width: 2px;
        height: 20px;
      }
      .treeView-box {
        border: 2px solid #ddd;
        width: 120px;
        height: 40px;
        line-height: 40px;
        text-align: center;
        border-radius: 5%;
        position: relative;
        margin: 7px;
      }
      .treeView-item {
        display: flex;
        justify-content: center;
        margin-top: 20px;
      }
      .treeView-btn {
        width: 15%;
        height: 41%;
        background: #fff;
        border: 1px solid #ddd;
        border-radius: 50%;
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        top: 84%;
        left: 42%;
      }
      .treeView-btn i {
        color: #2d8cf0;
        font-size: 12px;
      }
    `]
    })
], TreeViewComponent);
exports.TreeViewComponent = TreeViewComponent;
