// ng-cosmos-td-ui/src/chart/cm-chart/chart.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const chart_common_1 = require("./../common/chart.common");
let CosmosChartComponent = class CosmosChartComponent extends chart_common_1.ChartCommon {
    ngOnInit() {
        this.chartContainer = this.canvas.nativeElement;
    }
    ngOnDestroy() {
        this.disposeChart();
    }
};
CosmosChartComponent = __decorate([
    core_1.Component({
        selector: 'cm-chart',
        template: `
      <main>
          <div class="canvas" #canvas></div>
      </main>
    `,
        styles: [`
      main {
        width: 100%;
        height: 100%;
      }
      .canvas {
        width: 100%;
        height: 100%;
      }
    `]
    })
], CosmosChartComponent);
exports.CosmosChartComponent = CosmosChartComponent;
