// ng-cosmos-td-ui/src/business/tree/tree.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
let TreeComponent = class TreeComponent {
    constructor() {
        this.isColumn = true;
        this._cmData = [
            {
                "title": '1',
                "content": "com.android.dazhih",
                "show": false,
                "cComments": [
                    {
                        "title": '1.1',
                        "content": "com.android.dazhih",
                        "show": false,
                        "cComments": [
                            {
                                "title": '1.1.1',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            },
                            {
                                "title": '1.1.2',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            },
                            {
                                "title": '1.1.3',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            }
                        ]
                    },
                    {
                        "title": '1.2',
                        "content": "com.android.dazhih",
                        "show": false,
                        "cComments": [
                            {
                                "title": '1.2.1',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            },
                            {
                                "title": '1.2.2',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            },
                            {
                                "title": '1.2.3',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            }
                        ]
                    },
                    {
                        "title": '1.3',
                        "content": "com.android.dazhih",
                        "show": false,
                        "cComments": [
                            {
                                "title": '1.3.1',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            },
                            {
                                "title": '1.3.2',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": [
                                    {
                                        "title": '1.3.2.1',
                                        "content": "com.android.dazhih",
                                        "show": false,
                                        "cComments": false
                                    },
                                    {
                                        "title": '1.3.2.2',
                                        "content": "com.android.dazhih",
                                        "show": false,
                                        "cComments": false
                                    },
                                    {
                                        "title": '1.3.2.3',
                                        "content": "com.android.dazhih",
                                        "show": false,
                                        "cComments": [
                                            {
                                                "title": '1.3.2.3.1',
                                                "content": "com.android.dazhih",
                                                "show": false,
                                                "cComments": [
                                                    {
                                                        "title": '1.3.2.3.1.1',
                                                        "content": "com.android.dazhih",
                                                        "show": false,
                                                        "cComments": false
                                                    },
                                                    {
                                                        "title": '1.3.2.3.1.2',
                                                        "content": "com.android.dazhih",
                                                        "show": false,
                                                        "cComments": [
                                                            {
                                                                "title": '1.3.2.3.1.2.1',
                                                                "content": "com.android.dazhih",
                                                                "show": false,
                                                                "cComments": false
                                                            },
                                                            {
                                                                "title": '1.3.2.3.1.2.2',
                                                                "content": "com.android.dazhih",
                                                                "show": false,
                                                                "cComments": false
                                                            }
                                                        ]
                                                    },
                                                    {
                                                        "title": '1.3.2.3.1.3',
                                                        "content": "com.android.dazhih",
                                                        "show": false,
                                                        "cComments": [
                                                            {
                                                                "title": '1.3.2.3.1.3.1',
                                                                "content": "com.android.dazhih",
                                                                "show": false,
                                                                "cComments": false
                                                            }
                                                        ]
                                                    }
                                                ]
                                            },
                                            {
                                                "title": '1.3.2.3.2',
                                                "content": "com.android.dazhih",
                                                "show": false,
                                                "cComments": false
                                            },
                                            {
                                                "title": '1.3.2.3.3',
                                                "content": "com.android.dazhih",
                                                "show": false,
                                                "cComments": false
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "title": '1.3.3',
                                "content": "com.android.dazhih",
                                "show": false,
                                "cComments": false
                            }
                        ]
                    }
                ]
            }
        ];
        this.options = [{
                value: 'horizontal',
                label: 'horizontal'
            }, {
                value: 'vertical',
                label: 'vertical'
            }];
        this._cmOrient = 'vertical';
    }
    set cmOrient(cmOrient) {
        this._cmOrient = cmOrient;
    }
    ;
    set cmData(cmData) {
        this._cmData = cmData;
    }
    ;
    ngOnInit() {
        this.selectedChange(this._cmOrient);
    }
    selectedChange(value) {
        if (value == 'horizontal') {
            this.isColumn = false;
        }
        else {
            this.isColumn = true;
        }
        // console.log(value, this.isColumn)
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], TreeComponent.prototype, "cmOrient", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], TreeComponent.prototype, "cmData", null);
TreeComponent = __decorate([
    core_1.Component({
        selector: 'cm-tree',
        template: `
      <ng-container>
          <div class="tree">
              <div class="tree-control" nz-row>
                  <div nz-form-label nz-col [nzSpan]="2" class="tree-label">
                      <label>排列方向：</label>
                  </div>
                  <div nz-form-control nz-col [nzSpan]="6">
                      <nz-select style="width: 100%;" [(ngModel)]="_cmOrient" [nzSize]="default" (ngModelChange)="selectedChange(_cmOrient)">
                          <nz-option *ngFor="let option of options" [nzLabel]="option.label" [nzValue]="option.value">
                          </nz-option>
                      </nz-select>
                  </div>
              </div>
              <div class="tree-body" style="display:-webkit-flex;">
                  <cm-treeView [cmData]="_cmData" class="tree-view" style="flex:1" *ngIf="isColumn"></cm-treeView>
                  <cm-TransversalTree [cmData]="_cmData" class="tree-view" style="flex:1" *ngIf="!isColumn"></cm-TransversalTree>
              </div>
          </div>
      </ng-container>
    `,
        styles: [`
      .tree {
        width: 100%;
        overflow-x: auto;
        overflow-y: hidden;
      }
      .tree-body {
        margin: 20px 10px 10px -7px;
      }
      .tree-label {
        display: flex;
        align-items: center;
        height: 28px;
        justify-content: flex-end;
      }
      .tree-body:after {
        content: '';
        position: absolute;
        top: 90px;
        z-index: 2;
        background: #fff;
        width: 100%;
        height: 17px;
      }
    `]
    })
], TreeComponent);
exports.TreeComponent = TreeComponent;
