// ng-cosmos-td-ui/src/base/cascader/core/util/check.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
// tslint:disable-next-line:no-any
function isNotNil(value) {
    return (typeof (value) !== 'undefined') && value !== null;
}
exports.isNotNil = isNotNil;
/** 校验对象是否相等 */
function shallowEqual(objA, objB) {
    if (objA === objB)
        return true;
    if (typeof objA !== 'object' || !objA || typeof objB !== 'object' || !objB)
        return false;
    const keysA = Object.keys(objA);
    const keysB = Object.keys(objB);
    if (keysA.length !== keysB.length)
        return false;
    const bHasOwnProperty = Object.prototype.hasOwnProperty.bind(objB);
    // tslint:disable-next-line:prefer-for-of
    for (let idx = 0; idx < keysA.length; idx++) {
        const key = keysA[idx];
        if (!bHasOwnProperty(key))
            return false;
        if (objA[key] !== objB[key])
            return false;
    }
    return true;
}
exports.shallowEqual = shallowEqual;
function isInteger(value) {
    return typeof value === 'number' &&
        isFinite(value) &&
        Math.floor(value) === value;
}
exports.isInteger = isInteger;
function isEmpty(element) {
    const nodes = element.childNodes;
    for (let i = 0; i < nodes.length; i++) {
        const node = nodes.item(i);
        if ((node.nodeType === 1) && (node.outerHTML.toString().trim().length !== 0)) {
            return false;
        }
        else if ((node.nodeType === 3) && ((node.textContent.toString().trim().length !== 0))) {
            return false;
        }
    }
    return true;
}
exports.isEmpty = isEmpty;
function isNonEmptyString(value) {
    return typeof value === 'string' && value !== '';
}
exports.isNonEmptyString = isNonEmptyString;
function isTemplateRef(value) {
    return value instanceof core_1.TemplateRef;
}
exports.isTemplateRef = isTemplateRef;
function isComponent(value) {
    return value instanceof core_1.Type;
}
exports.isComponent = isComponent;
