// ng-cosmos-td-ui/src/business/dragsort/dragSort.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const Sortable = require("sortablejs");
let CosmosDragSortComponent = class CosmosDragSortComponent {
    constructor() {
        this.onSortEnd = new core_1.EventEmitter();
        this._chooseOptions = [];
    }
    set cmData(columnData) {
        for (let group of columnData) {
            for (let item of group['list']) {
                item['checked'] = item['checked'] ? true : false;
            }
        }
        this._checkOptions = columnData;
    }
    ngOnInit() {
        this.sortable = Sortable.create(this.handllist.nativeElement, {
            handle: '.sortHandle',
            sort: true,
            animation: 150,
            ghostClass: "sortable-ghost",
            chosenClass: "sortable-chosen",
            onUpdate: (evt) => {
                this.onSortEnd.emit(this.getSortEndOptions());
            }
        });
    }
    checkboxChange(event, option) {
        if (option.checked) {
            this._chooseOptions.push(option);
            this.onSortEnd.emit(this.getSortEndOptions());
        }
        else {
            for (let index in this._chooseOptions) {
                if (this._chooseOptions[index] === option) {
                    this._chooseOptions.splice(Number(index), 1);
                    break;
                }
            }
        }
    }
    getSortEndOptions() {
        return this._chooseOptions;
    }
    ngOnDestroy() {
        this.sortable.destroy();
    }
};
__decorate([
    core_1.ViewChild('handllist'),
    __metadata("design:type", core_1.ElementRef)
], CosmosDragSortComponent.prototype, "handllist", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Array),
    __metadata("design:paramtypes", [Array])
], CosmosDragSortComponent.prototype, "cmData", null);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CosmosDragSortComponent.prototype, "onSortEnd", void 0);
CosmosDragSortComponent = __decorate([
    core_1.Component({
        selector: 'cm-dragsort',
        template: `
      <main class="container">
          <div class="handllist" #handllist>
              <ng-container *ngFor="let option of _chooseOptions">
                  <ng-container *ngIf="option.checked">
                      <cm-button class="sortHandle">
                          <button nz-button [nzType]="'primary'">{{option.label}}</button>
                      </cm-button>
                  </ng-container>
              </ng-container>
          </div>
          <div class="menulist">
              <ng-container *ngFor="let group of _checkOptions">
                  <div class="groupName">{{group.name}}</div>
                  <ng-container *ngFor="let option of group['list']">
                      <cm-checkbox class="checkHandle">
                          <label nz-checkbox [(ngModel)]="option.checked" (ngModelChange)="checkboxChange($event,option)">
                              <span>{{option.label}}</span>
                          </label>
                      </cm-checkbox>
                  </ng-container>
              </ng-container>
          </div>
      </main>
    `,
        styles: [`
      .container {
        width: 100%;
      }
      .handllist {
        width: 100%;
        padding: 10px 0 0 10px;
        min-height: 48px;
        overflow-y: auto;
        border-left: 1px solid #cccccc;
        border-right: 1px solid #cccccc;
        border-top: 1px solid #cccccc;
      }
      .sortHandle {
        display: inline-block;
        margin-right: 10px;
        margin-bottom: 10px;
      }
      .menulist {
        width: 100%;
        max-height: 250px;
        padding: 10px 0 0 10px;
        border: 1px dotted #cccccc;
        overflow-y: auto;
      }
      .checkHandle {
        display: inline-block;
        margin-right: 10px;
        margin-bottom: 10px;
      }
      .groupName {
        padding-top: 10px;
        padding-bottom: 10px;
      }
    `]
    }),
    __metadata("design:paramtypes", [])
], CosmosDragSortComponent);
exports.CosmosDragSortComponent = CosmosDragSortComponent;
