// ng-cosmos-td-ui/src/business/retention/retention.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
let RetentionComponent = class RetentionComponent {
    constructor() {
        this.isShow = false;
        this.colors = ["#f0f9ff", "#c8e8ff", "#9ed3ff", "#75bafb", "#5693d4", "#3e70ad", "#2a5188", "#1e3760"];
        this.positionStyles = {};
        this.width = '200px';
        this.padding = "200px";
        this._cmDataScale = [
            {
                num1: "+1日",
                num2: "+2日",
                num3: "+3日",
                num4: "+4日",
                num5: "+5日",
                num6: "+6日",
                num7: "+7日",
                num8: "+8日",
                num9: "+9日",
                num10: "+10日",
                num11: "+11日",
                num12: "+12日",
                num13: "+13日",
                num14: "+14日"
            },
            {
                num1: 1250,
                num2: 1250,
                num3: 1250,
                num4: 1250,
                num5: 1250,
                num6: 1250,
                num7: 1250,
                num8: 1250,
                num9: 1250,
                num10: 1250,
                num11: 1250,
                num12: 1250,
                num13: 1250,
                num14: 1250,
            },
            {
                num1: 9.4,
                num2: 9.4,
                num3: 16.7,
                num4: 16.7,
                num5: 9.4,
                num6: 9.4,
                num7: 9.4,
                num8: 9.4,
                num9: 9.4,
                num10: 5.3,
                num11: 5.3,
                num12: 3.0,
                num13: 0.2,
                num14: 0.2,
            }, {
                num1: 9.4,
                num2: 16.7,
                num3: 16.7,
                num4: 9.4,
                num5: 9.4,
                num6: 9.4,
                num7: 9.4,
                num8: 9.4,
                num9: 5.3,
                num10: 5.3,
                num11: 5.3,
                num12: 3.0,
                num13: 0.2,
                num14: 0.2,
            }, {
                num1: 16.7,
                num2: 16.7,
                num3: 9.4,
                num4: 9.4,
                num5: 9.4,
                num6: 9.4,
                num7: 9.4,
                num8: 5.3,
                num9: 5.3,
                num10: 5.3,
                num11: 3.0,
                num12: 3.0,
                num13: 0.2,
                num14: 0.2,
            }, {
                num1: 9.4,
                num2: 9.4,
                num3: 5.3,
                num4: 5.3,
                num5: 5.3,
                num6: 5.3,
                num7: 5.3,
                num8: 3.0,
                num9: 3.0,
                num10: 3.0,
                num11: 3.0,
                num12: 0.2,
                num13: 0.2,
                num14: 0.2,
            }, {
                num1: 9.4,
                num2: 9.4,
                num3: 5.3,
                num4: 5.3,
                num5: 5.3,
                num6: 5.3,
                num7: 3.0,
                num8: 3.0,
                num9: 3.0,
                num10: 3.0,
                num11: 3.0,
                num12: 0.2,
                num13: 0.2,
                num14: 0.2,
            }, {
                num1: 9.4,
                num2: 5.3,
                num3: 5.3,
                num4: 5.3,
                num5: 5.3,
                num6: 5.3,
                num7: 3.0,
                num8: 3.0,
                num9: 3.0,
                num10: 3.0,
                num11: 0.2,
                num12: 0.2,
                num13: 0.2,
                num14: 0.2
            }
        ];
        this._cmDataInfo = [
            {
                title1: '日期',
                title2: '新增用户'
            },
            {
                title1: '均值',
                title2: '1250'
            },
            {
                date: '2018-1-17',
                user: 1250
            },
            {
                date: '2018-1-17',
                user: 1250
            },
            {
                date: '2018-1-17',
                user: 1250
            },
            {
                date: '2018-1-17',
                user: 1250
            },
            {
                date: '2018-1-17',
                user: 1250
            },
            {
                date: '2018-1-17',
                user: 1250
            }
        ];
        this.arr = [];
    }
    set cmDataInfo(cmDataInfo) {
        this._cmDataInfo = cmDataInfo;
    }
    ;
    set cmDataScale(cmDataScale) {
        this._cmDataScale = cmDataScale;
    }
    ;
    ngOnInit() {
        this.titleRight = this._cmDataScale.slice(0, 2);
        this.infoRight = this._cmDataScale.slice(2);
        this.titleLeft = this._cmDataInfo.slice(0, 2);
        this.infoLeft = this._cmDataInfo.slice(2);
        // console.log(this.titleRight,this.infoRight,this.titleLeft,this.infoLeft)
        this.setPadding();
        this.dataArr();
        // console.log(this.arr)	
        this.minData = this.arr[0];
        this.maxData = this.arr.pop();
        this.cha = (this.maxData - this.minData) / 8;
        // console.log(this.minData, this.maxData, this.cha)			
    }
    setPadding() {
        this.iL = Object.keys(this.infoLeft[1]);
        this.tL = Object.keys(this.titleLeft[1]);
        // console.log(this.iL,this.tL)
        let length = Object.keys(this.infoLeft[0]).length;
        // console.log(length);
        this.width = `${length * 100}px`;
        this.padding = `${length * 100}px`;
    }
    dataArr() {
        this.iR = Object.keys(this.infoRight[1]);
        this.tR = Object.keys(this.titleRight[1]);
        // console.log(this.iR,this.tR)
        // console.log(this.infoRight.length,this.infoRight)
        for (var i = 0; i < this.infoRight.length; i++) {
            let infoRight = this.infoRight[i];
            for (var key in infoRight) {
                // console.log(key,infoRight[key])
                this.arr.push(infoRight[key]);
            }
        }
        // console.log(this.arr)
        var temp = 0;
        for (var i = 0; i < this.arr.length; i++) { //这里的i控制循环比较的次数
            for (var j = 0; j < this.arr.length - i; j++) {
                //这里的j控制参与比较的数字个数，因为每比较一次都会产生一个最大的数字，那么下次比较的时候就没必要参与比较了，所以每比较一次都会减少一个数字
                if (this.arr[j] > this.arr[j + 1]) {
                    temp = this.arr[j + 1];
                    this.arr[j + 1] = this.arr[j];
                    this.arr[j] = temp;
                }
                if (this.arr[j] == "") {
                    // console.log(this.arr[j])
                    this.arr[j] = 0;
                }
            }
        }
        var res = [this.arr[0]];
        for (var i = 1; i < this.arr.length; i++) {
            if (this.arr[i] !== res[res.length - 1]) {
                res.push(this.arr[i]);
            }
        }
        return res;
    }
    getBgColor(value) {
        if (value >= this.minData && value < this.minData + this.cha) {
            return this.colors[0];
        }
        else if (value >= this.minData + this.cha && value < this.minData + 2 * this.cha) {
            return this.colors[1];
        }
        else if (value >= this.minData + 2 * this.cha && value < this.minData + 3 * this.cha) {
            return this.colors[2];
        }
        else if (value >= this.minData + 3 * this.cha && value < this.minData + 4 * this.cha) {
            return this.colors[3];
        }
        else if (value >= this.minData + 4 * this.cha && value < this.minData + 5 * this.cha) {
            return this.colors[4];
        }
        else if (value >= this.minData + 5 * this.cha && value < this.minData + 6 * this.cha) {
            return this.colors[5];
        }
        else if (value >= this.minData + 6 * this.cha && value < this.minData + 7 * this.cha) {
            return this.colors[6];
        }
        else {
            return this.colors[7];
        }
    }
    getfsColor(value) {
        if (value > this.minData + 3 * this.cha) {
            return '#fff';
        }
        else {
            return '#000';
        }
    }
    mousemove(e, value) {
        this.showData = value;
        this.isShow = true;
        this.positionStyles = {
            'left': `${e.clientX - 300}px`,
            'top': `${e.clientY - e.offsetY}px`,
        };
        // console.log(e.clientX,e.clientX-300,e.clientY,e.offsetY)
    }
    mouseleave(value) {
        this.isShow = false;
    }
    ngOnDestroy() { }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], RetentionComponent.prototype, "cmDataInfo", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], RetentionComponent.prototype, "cmDataScale", null);
RetentionComponent = __decorate([
    core_1.Component({
        selector: 'cm-retention',
        template: `
   <ng-container>
       <div class="table">
           <div class="table1" [ngStyle]="{'width':width}">
               <nz-table #nzTable [nzPageSize]="10" [nzDataSource]="_cmDataInfo" [nzIsPagination]="false">
                   <tbody nz-tbody>
                       <tr nz-tbody-tr *ngFor="let title of titleLeft">
                           <td nz-td *ngFor="let tL of tL">
                               <span>{{title[tL]}}</span>
                           </td>
                       </tr>
                       <tr nz-tbody-tr *ngFor="let data of infoLeft">
                           <td nz-td *ngFor="let iL of iL">
                               <span>{{data[iL]}}</span>
                           </td>
                       </tr>
                   </tbody>
               </nz-table>
           </div>
           <div class="table2" [ngStyle]="{'padding-left':padding}">
               <nz-table #nzTable [nzDataSource]="_cmDataScale" [nzPageSize]="10" [nzIsPagination]="false">
                   <tbody nz-tbody>
                       <tr nz-tbody-tr *ngFor="let title of titleRight">
                           <td nz-td *ngFor="let tR of tR">
                               <span>{{title[tR]}}</span>
                           </td>
                       </tr>
                       <tr nz-tbody-tr *ngFor="let data of infoRight" class="hover">
                           <td nz-td *ngFor="let iR of iR" class="data" [ngStyle]="{'background-color':getBgColor(data[iR])}" (mousemove)="mousemove($event,data[iR])"
                               (mouseleave)="mouseleave(data[iR])">
                               <!-- <span [ngStyle]="{'color':getfsColor(data[iR])}">{{data[iR]}}%</span> -->
                               <nz-tooltip [nzTitle]="showData" [nzPlacement]="'top'">
                                   <span class="td-span" [ngStyle]="{'color':getfsColor(data[iR])}" nz-tooltip>{{data[iR]}}%</span>
                               </nz-tooltip>
                           </td>
                       </tr>
                   </tbody>
               </nz-table>
           </div>
       </div>
   </ng-container>
	`,
        styles: [`
   ::ng-deep table {
     box-sizing: border-box;
     width: 100%;
     position: relative;
   }
   ::ng-deep .table1 {
     width: 20%;
     position: absolute;
     background: #fff;
     z-index: 10;
   }
   ::ng-deep .table1 tr {
     background: #fff;
   }
   ::ng-deep .table1 tr td {
     background: #fff;
   }
   ::ng-deep .table2 {
     width: 100%;
     position: relative;
   }
   ::ng-deep .table2 .ant-table-body {
     width: auto;
     overflow-x: scroll;
   }
   ::ng-deep td {
     min-width: 100px;
     height: 50px;
   }
   ::ng-deep .table1 .ant-table-thead > tr.ant-table-row-hover > td,
   ::ng-deep .table1 .ant-table-tbody > tr.ant-table-row-hover > td,
   ::ng-deep .table1 .ant-table-thead > tr:hover > td,
   ::ng-deep .table1 .ant-table-tbody > tr:hover > td {
     background: none !important;
   }
   ::ng-deep .ant-table-tbody > tr > td {
     box-sizing: border-box;
     height: 50px;
     border: 1px solid transparent;
   }
   ::ng-deep .ant-table-tbody > tr {
     box-sizing: border-box;
     height: 50px;
   }
   ::ng-deep .table2 .ant-table-tbody > .hover > td:hover {
     box-sizing: border-box;
     border-color: blue;
   }
   ::ng-deep .tooltip {
     width: 100px;
     height: 60px;
     background-color: #fff;
     box-shadow: 1px 1px 10px #888888;
     font-size: 20px;
     position: absolute;
     z-index: 1000;
     text-align: center;
     line-height: 60px;
   }
   ::ng-deep .ant-table-thead > tr > th,
   ::ng-deep .ant-table-tbody > tr > td {
     padding: 0px !important;
   }
   ::ng-deep .td-span {
     color: #ffffff;
     display: inline-block;
     width: 100%;
     height: 50px;
     line-height: 50px;
     text-align: center;
   }
	`]
    }),
    __metadata("design:paramtypes", [])
], RetentionComponent);
exports.RetentionComponent = RetentionComponent;
