// ng-cosmos-td-common/src/themeModule/theme.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const Subject_1 = require("rxjs/Subject");
const theme_option_1 = require("./theme.option");
exports.light = "light";
exports.dark = "dark";
let ThemeService = class ThemeService {
    constructor(options) {
        this.options = options;
        this.themeSource = new Subject_1.Subject();
        this.missionTheme$ = this.themeSource.asObservable();
        this.defaultTheme = exports.light;
    }
    get currentTheme() {
        return this._currentTheme;
    }
    use(themeName) {
        let theme = this.getCurrentTheme(themeName);
        if (!theme) {
            return;
        }
        let themeLink = document.querySelector(`link[theme="${theme.value}"]`);
        if (!themeLink) {
            let link = document.querySelector('link[theme-position]');
            if (!link) {
                link = document.createElement('link');
                document.head.insertBefore(link, document.querySelector('link[theme]'));
            }
            else {
                link.removeAttribute('theme-position');
            }
            link.setAttribute('theme', theme.value);
            link.setAttribute('rel', 'stylesheet');
            link.setAttribute('href', theme.path.view);
        }
        document.documentElement.setAttribute('class', "theme-" + theme.value);
        this._currentTheme = theme.value;
        this.themeSource.next(theme);
    }
    getCurrentTheme(themeName) {
        for (let theme of this.options.theme) {
            if (theme.value == themeName) {
                return theme;
            }
        }
        return null;
    }
};
ThemeService = __decorate([
    core_1.Injectable(),
    __param(0, core_1.Inject(theme_option_1.THEME_OPTION)), __param(0, core_1.Optional()),
    __metadata("design:paramtypes", [Object])
], ThemeService);
exports.ThemeService = ThemeService;
