// ng-cosmos-td-ui/src/chart/chart.component.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var ChartComponentModule_1;
const core_1 = require("@angular/core");
const chart_module_1 = require("./cm-chart/chart.module");
// 导出组件模块
var chart_module_2 = require("./cm-chart/chart.module");
exports.CosmosChartModule = chart_module_2.CosmosChartModule;
// 导出组件
var chart_component_1 = require("./cm-chart/chart.component");
exports.CosmosChartComponent = chart_component_1.CosmosChartComponent;
let ChartComponentModule = ChartComponentModule_1 = class ChartComponentModule {
    static forRoot() {
        return {
            ngModule: ChartComponentModule_1
        };
    }
};
ChartComponentModule = ChartComponentModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            chart_module_1.CosmosChartModule
        ]
    })
], ChartComponentModule);
exports.ChartComponentModule = ChartComponentModule;
exports.chartComponentModuleArray = [
    chart_module_1.CosmosChartModule
];
