// ng-cosmos-td-ui/src/business/sortselector/sortselector.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var SortselectorComponent_1;
const core_1 = require("@angular/core");
const pinyinUtil = require("../../scripts/pinyinjs/pinyinUtil");
const overlay_1 = require("@angular/cdk/overlay");
const mesureScrollBar_1 = require("../../utils/mesureScrollBar");
const forms_1 = require("@angular/forms");
let SortselectorComponent = SortselectorComponent_1 = class SortselectorComponent {
    constructor(_elementRef, _renderer, _cdr) {
        this._elementRef = _elementRef;
        this._renderer = _renderer;
        this._cdr = _cdr;
        this._cmSortOptions = [];
        this.selectedOptionModel = [];
        this.optionsList = [];
        this.value = [];
        this._isOpen = false;
        this._cmPlaceHolder = "placeholder";
        this._selectedOptions = []; //已选中选项
        this._searchText = '';
        this.cmShowSearch = false;
        this.cmMultiple = true;
        this._filterOptions = [];
        this._dropDownPosition = 'bottom';
        this._offsetX = 0;
        this._cmTooltip = false;
        this.optionIsNull = false;
        this.isFirst = true;
        // ngModel Access
        this.onChange = () => null;
        this.onTouched = () => null;
        this.cmClosePanel = new core_1.EventEmitter();
        this._el = this._elementRef.nativeElement;
    }
    set cmSortOptions(cmSortOptions) {
        if (cmSortOptions && cmSortOptions.length > 0) {
            this._cmSortOptions = JSON.parse(JSON.stringify(cmSortOptions));
            this._cmSortOptions = this.setSortArray(this._cmSortOptions);
            this.handlerData(this._cmSortOptions);
            this._cmSortOptions = JSON.parse(JSON.stringify(this._filterOptions));
            if (this.selectedOptionModel.length > 0) {
                let historySelectedList = this.selectedOptionModel.toString();
                this.initOptions();
                if (this.selectedOptionModel.toString() !== historySelectedList) {
                    this.cmClosePanel.emit(this.selectedOptionModel);
                    this.onChange(this.selectedOptionModel);
                }
                ;
            }
        }
        else {
            this._cmSortOptions = [];
            if (this.selectedOptionModel.length > 0) {
                this.selectedOptionModel = [];
                this._selectedOptions = [];
                this.cmClosePanel.emit(this.selectedOptionModel);
                this.onChange(this.selectedOptionModel);
            }
        }
    }
    set cmPlaceHolder(cmPlaceHolder) {
        if (cmPlaceHolder.length > 0) {
            this._cmPlaceHolder = cmPlaceHolder;
        }
        else {
            this._cmPlaceHolder = "请选择";
        }
    }
    set cmMode(value) {
        this._mode = value;
        if (this._mode === 'multiple') {
            this.cmMultiple = true;
        }
        else if (this._mode === 'single') {
            this.cmMultiple = false;
        }
    }
    set cmTooltip(cmTooltip) {
        this._cmTooltip = cmTooltip ? true : false;
    }
    set cmSize(value) {
        this._size = { large: 'lg', small: 'sm' }[value] || 'sm';
        this._class = { [`cm-sortselector-${this._size}`]: true };
    }
    get _arrowCls() {
        return {
            [`cm-sortselector-arrow`]: true,
            [`cm-sortselector-arrow-expand`]: this._isOpen
        };
    }
    //取首字母
    setSortArray(cmSortOptions) {
        if (cmSortOptions && cmSortOptions.length > 0) {
            let dataArr = 'abcdefghijklmnopqrstuvwxyz0123456789';
            let optLength = cmSortOptions.length;
            for (let i = 0; i < optLength; i++) {
                let opt = cmSortOptions[i];
                let firstLetter = opt['label'].substr(0, 1).toUpperCase();
                if (dataArr.indexOf(firstLetter.toLowerCase()) >= 0) {
                    opt['code'] = firstLetter;
                }
                else {
                    opt['code'] = (pinyinUtil.pinyinUtil.getFirstLetter(firstLetter)).toUpperCase();
                }
                opt['index'] = dataArr.indexOf(opt['code'].toLowerCase());
            }
            cmSortOptions.sort(function (a, b) {
                if (a.index < b.index) {
                    return -1;
                }
                if (a.index > b.index) {
                    return 1;
                }
                return 0;
            });
            return cmSortOptions;
        }
    }
    //处理数据
    handlerData(cmSortOptions) {
        this._filterOptions = [];
        let optLength = cmSortOptions.length;
        for (let i = 0; i < optLength; i++) {
            let opt = cmSortOptions[i];
            this._filterOptions = this.changeOptionByCode(opt, this._filterOptions);
        }
    }
    //分组
    changeOptionByCode(obj, data) {
        if (data) {
            obj['isSelected'] = false;
            let hasCode = false;
            let len = data.length;
            for (let i = 0; i < len; i++) {
                if (obj['code'] && data[i]['code'] == obj['code']) {
                    hasCode = true;
                    data[i]['list'].push(obj);
                    break;
                }
            }
            //不存在
            if (!hasCode && obj['code']) {
                data[len] = {
                    code: obj['code'],
                    list: [obj]
                };
            }
            return data;
        }
    }
    clcikOptionHandler(e, option) {
        this.isFirst = false;
        this._searchText = '';
        if (this.cmMultiple) {
            if (this.selectedOptionModel.indexOf(option.value) < 0) {
                this._selectedOptions.push(option);
                this.selectedOptionModel.push(option.value);
                this.updateFilterOption(option, this._cmSortOptions);
                this._filterOptions = JSON.parse(JSON.stringify(this._cmSortOptions));
                this.cmClosePanel.emit(this.selectedOptionModel);
                this.onChange(this.selectedOptionModel);
            }
            else {
                this.updateFilterOption(option, this._cmSortOptions);
                this.unSelectMultipleOption(option, null);
            }
            setTimeout(() => {
                this._cdkOverlay.overlayRef.updatePosition();
            }, 0);
        }
        else {
            this.selectedOptionModel = [];
            this._selectedOptions = [];
            this.selectedOptionModel[0] = option.value;
            this._selectedOptions[0] = option;
            this.updateFilterOption(option, this._cmSortOptions);
            this._filterOptions = JSON.parse(JSON.stringify(this._cmSortOptions));
            this.closeDropDown();
        }
        e.stopPropagation();
    }
    closeDropDown() {
        if (!this._isOpen) {
            return;
        }
        this._isOpen = !this._isOpen;
        this._searchText = '';
        if (this._selectedOptions.length <= 0)
            this.cmShowSearch = false;
        this.cmClosePanel.emit(this.selectedOptionModel);
        if (!this.isFirst)
            this.onChange(this.selectedOptionModel);
    }
    onClick(e) {
        e.stopPropagation();
        this._isOpen = !this._isOpen;
        this.cmShowSearch = true;
        /** wait for search display */
        setTimeout(() => {
            this.searchInputElementRef.nativeElement.focus();
        });
    }
    clearSearchText() {
        this._searchText = '';
        setTimeout(() => {
            if (this._cmSortOptions.length > 0) {
                this.updateFilterOption(null, null);
            }
        }, 500);
    }
    updateWidth(element, text) {
        this.isFirst = false;
        if (text) {
            /** wait for scroll width change */
            setTimeout(() => {
                this._renderer.setStyle(element, 'width', `${element.scrollWidth}px`);
            });
        }
        else {
            this._renderer.removeStyle(element, 'width');
        }
    }
    //删除某个选中的选项
    unSelectMultipleOption(option, $event) {
        let len = this._selectedOptions.length;
        for (let i = 0; i < len; i++) {
            let item = this._selectedOptions[i];
            if (item.value == option.value) {
                this._selectedOptions.splice(i, 1);
                this.selectedOptionModel.splice(i, 1);
                if (this._selectedOptions.length <= 0) {
                    this.cmShowSearch = false;
                }
                break;
            }
        }
        this.updateFilterOption(option, this._cmSortOptions);
        this.updateFilterOption(option, this._filterOptions);
        this.cmClosePanel.emit(this.selectedOptionModel);
        this.onChange(this.selectedOptionModel);
        $event ? $event.stopPropagation() : null;
    }
    //更新下拉框数据
    updateFilterOption(option, data) {
        if (option && data) {
            data.forEach((obj) => {
                let len = obj.list.length;
                for (let i = 0; i < len; i++) {
                    let listItem = obj.list[i];
                    if (option.value == listItem.value) {
                        obj.list[i]['isSelected'] = !obj.list[i]['isSelected'];
                        if (this.cmMultiple)
                            return;
                    }
                    else {
                        if (!this.cmMultiple) {
                            obj.list[i]['isSelected'] = false;
                        }
                    }
                }
            });
        }
        else if (this._searchText) { //搜索
            let searchtext = this._searchText;
            let newOptions = [];
            newOptions = JSON.parse(JSON.stringify(this._cmSortOptions));
            let len = newOptions.length;
            for (let i = 0; i < len; i++) {
                newOptions[i]['list'] = [];
                let listLength = this._cmSortOptions[i]['list'].length;
                for (let j = 0; j < listLength; j++) {
                    let listItem = this._cmSortOptions[i]['list'][j];
                    if (listItem['label'].indexOf(searchtext) >= 0) {
                        newOptions[i]['list'].push(listItem);
                    }
                }
            }
            this._filterOptions = newOptions;
            this.checkOptionsIsNull(this._filterOptions);
        }
        else {
            this._filterOptions = JSON.parse(JSON.stringify(this._cmSortOptions));
            this.optionIsNull = false;
        }
    }
    checkOptionsIsNull(options) {
        let isNull = true;
        if (options.length > 0) {
            let len = options.length;
            for (let i = 0; i < len; i++) {
                if (options[i]['list'] && options[i]['list'].length > 0) {
                    isNull = false;
                }
            }
            this.optionIsNull = isNull;
        }
    }
    _getTriggerRect() {
        return this.trigger.nativeElement.getBoundingClientRect();
    }
    onPositionChange(position) {
        this.reposition();
        this._dropDownPosition = position.connectionPair.originY;
    }
    reposition() {
        if (typeof window !== 'undefined' && this._isOpen && this._cdkOverlay && this._cdkOverlay.overlayRef) {
            const originElement = this._cdkOverlay.origin.elementRef.nativeElement;
            const overlayElement = this._cdkOverlay.overlayRef.overlayElement;
            const originX = originElement.getBoundingClientRect().x;
            const overlayWidth = overlayElement.getBoundingClientRect().width < this._getTriggerRect().width ? this._getTriggerRect().width : overlayElement.getBoundingClientRect().width;
            overlayElement.style.width = overlayWidth + 'px';
            const margin = window.innerWidth - originX - overlayWidth;
            this._offsetX = margin > 0 ? 0 : margin - (mesureScrollBar_1.measureScrollbar() || 15);
            this._cdr.detectChanges();
        }
    }
    _updateValue(value) {
        if (this.selectedOptionModel === value) {
            return;
        }
        if ((value == null) || value.length === 0) {
            this.selectedOptionModel = [];
            this.cmShowSearch = false;
        }
        else {
            this.selectedOptionModel = value;
            this.cmShowSearch = true;
        }
        this.initOptions();
    }
    initOptions() {
        let selectArr = JSON.parse(JSON.stringify(this.selectedOptionModel));
        this._filterOptions = JSON.parse(JSON.stringify(this._cmSortOptions));
        this._selectedOptions = [];
        this.selectedOptionModel = [];
        this.setOptionsUnselected(); //全部未选中
        for (let i = 0; i < this._filterOptions.length; i++) {
            let listItems = this._filterOptions[i]['list'];
            for (let value of selectArr) {
                for (let j = 0; j < listItems.length; j++) {
                    if (value == listItems[j]['value']) {
                        this.updateFilterOption(listItems[j], this._cmSortOptions);
                        this.updateFilterOption(listItems[j], this._filterOptions);
                        this._selectedOptions.push(listItems[j]);
                        this.selectedOptionModel.push(value);
                        break;
                    }
                }
            }
        }
    }
    setOptionsUnselected() {
        for (let i = 0; i < this._filterOptions.length; i++) {
            let listItems = this._filterOptions[i]['list'];
            for (let j = 0; j < listItems.length; j++) {
                listItems[j]['isSelected'] = false;
                this._cmSortOptions[i]['list'][j]['isSelected'] = false;
            }
        }
    }
    writeValue(value) {
        this._updateValue(value);
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    onAttach() {
        this.reposition();
    }
    ngOnInit() {
    }
    ngOnDestroy() {
    }
};
__decorate([
    core_1.ViewChild(overlay_1.CdkConnectedOverlay),
    __metadata("design:type", overlay_1.CdkConnectedOverlay)
], SortselectorComponent.prototype, "_cdkOverlay", void 0);
__decorate([
    core_1.ViewChild('trigger'),
    __metadata("design:type", core_1.ElementRef)
], SortselectorComponent.prototype, "trigger", void 0);
__decorate([
    core_1.ViewChild('searchInput'),
    __metadata("design:type", Object)
], SortselectorComponent.prototype, "searchInputElementRef", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Array),
    __metadata("design:paramtypes", [Array])
], SortselectorComponent.prototype, "cmSortOptions", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], SortselectorComponent.prototype, "cmPlaceHolder", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], SortselectorComponent.prototype, "cmMode", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], SortselectorComponent.prototype, "cmTooltip", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], SortselectorComponent.prototype, "cmSize", null);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], SortselectorComponent.prototype, "cmClosePanel", void 0);
__decorate([
    core_1.HostListener('click', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [MouseEvent]),
    __metadata("design:returntype", void 0)
], SortselectorComponent.prototype, "onClick", null);
SortselectorComponent = SortselectorComponent_1 = __decorate([
    core_1.Component({
        encapsulation: core_1.ViewEncapsulation.None,
        selector: 'cm-sortselector',
        template: `
      <div #trigger cdkOverlayOrigin #origin="cdkOverlayOrigin" class="cm-sortselector-selection" [ngClass]="_class">
          <div class="cm-sortselector-selection__rendered" *ngIf="!cmShowSearch">
              <div class="cm-sortselector-selection__placeholder" *ngIf="!cmShowSearch && !_isOpen">
                  {{ _cmPlaceHolder }}
              </div>
          </div>
          <div class="cm-sortselector-selection__rendered" *ngIf="cmShowSearch">

              <ul class="ant-select-selection--multiple clrfix">
                  <ng-container *ngIf="!_cmTooltip">
                      <li *ngFor="let option of _selectedOptions" class="ant-select-selection__choice" style="-webkit-user-select: none;">
                          <div class="ant-select-selection__choice__content">{{ option?.label }}</div>
                          <span class="ant-select-selection__choice__remove" (click)="unSelectMultipleOption(option,$event)"></span>
                      </li>
                  </ng-container>
                  <ng-container *ngIf="_cmTooltip">
                      <li *ngFor="let option of _selectedOptions" class="ant-select-selection__choice" style="-webkit-user-select: none;">
                          <nz-tooltip [nzTitle]="option?.label" [nzPlacement]="'top'">
                              <div nz-tooltip class="ant-select-selection__choice__content">{{ option?.label }}</div>
                          </nz-tooltip>
                          <span class="ant-select-selection__choice__remove" (click)="unSelectMultipleOption(option,$event)"></span>
                      </li>
                  </ng-container>
                  <li class="ant-select-search ant-select-search--inline">
                      <div class="ant-select-search__field__wrap">
                          <input class="ant-select-search__field" [(ngModel)]="_searchText" (ngModelChange)="updateFilterOption();" (keydown)="updateWidth(searchInput,_searchText)" (input)="updateWidth(searchInput,_searchText)" (blur)="clearSearchText()" #searchInput>
                      </div>
                  </li>
              </ul>

          </div>

          <span [ngClass]="_arrowCls"></span>
      </div>


      <ng-template cdkConnectedOverlay cdkConnectedOverlayHasBackdrop [cdkConnectedOverlayOrigin]="origin" [cdkConnectedOverlayOffsetX]="_offsetX" (backdropClick)="closeDropDown()" (detach)="closeDropDown();" (positionChange)="onPositionChange($event)" (attach)="onAttach()"
          [cdkConnectedOverlayWidth]="_triggerWidth" [cdkConnectedOverlayOpen]="_isOpen">
          <div style="overflow: auto;" class="cm-sortselector-dropdown">
              <ng-container *ngIf="_filterOptions.length > 0">
                  <div *ngFor="let optGroup of _filterOptions" class="clrfix">
                      <div *ngIf="optGroup.list.length > 0" class="cm-sortselector-dropdown-title fl"> {{ optGroup.code }}</div>
                      <ul class="clrfix fl" class="cm-sortselector-dropdown-menu">
                          <ng-container *ngIf="_cmTooltip">
                              <nz-tooltip *ngFor="let option of optGroup.list" [nzTitle]="option?.label" [nzPlacement]="'top'">
                                  <li nz-tooltip [ngClass]="{'isSelected': option.isSelected}" class="cm-sortselector-dropdown-menu-item fl" (click)="clcikOptionHandler($event,option)" value="option.value">{{option.label}}</li>
                              </nz-tooltip>
                          </ng-container>

                          <ng-container *ngIf="!_cmTooltip">
                              <li [ngClass]="{'isSelected': option.isSelected}" class="cm-sortselector-dropdown-menu-item fl" *ngFor="let option of optGroup.list" (click)="clcikOptionHandler($event,option)" value="option.value">{{option.label}}</li>
                          </ng-container>
                      </ul>
                  </div>
              </ng-container>

              <ng-container *ngIf="_filterOptions.length <= 0 || optionIsNull">
                  <div style="padding:7px 8px;">暂无数据</div>
              </ng-container>

          </div>
      </ng-template>
    `,
        styles: [`
      /* button*/
      .cm-light-btn-default {
        color: #495060;
        background-color: #fff;
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-btn-default:hover {
        color: #2a6cF0;
        border-color: #2a6cF0;
      }
      .cm-light-btn-default:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-info {
        color: #fff;
        background-color: #2db7f5;
        border: 1px solid #2db7f5;
        border-radius: 4px;
      }
      .cm-light-btn-info:hover {
        background-color: #57CFFF;
        color: #fff;
        border-color: #2db7f5;
      }
      .cm-light-btn-info:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-primary {
        color: #fff;
        background-color: #2d8cf0;
        border: 1px solid #2d8cf0;
        border-radius: 4px;
      }
      .cm-light-btn-primary:hover {
        background-color: #2a6cF0;
        color: #fff;
        border-color: #2d8cf0;
      }
      .cm-light-btn-primary:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-success {
        color: #fff;
        background-color: #19be6b;
        border: 1px solid #19be6b;
        border-radius: 4px;
      }
      .cm-light-btn-success:hover {
        color: #fff;
        background-color: #47cb89;
        border-color: #19be6b;
      }
      .cm-light-btn-success:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-warning {
        color: #fff;
        background-color: #f90;
        border: 1px solid #f90;
        border-radius: 4px;
      }
      .cm-light-btn-warning:hover {
        color: #fff;
        background-color: #f59e2d;
        border-color: #f90;
      }
      .cm-light-btn-warning:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-failed {
        color: #fff;
        background-color: #ed3f14;
        border: 1px solid #ed3f14;
        border-radius: 4px;
      }
      .cm-light-btn-failed:hover {
        color: #fff;
        background-color: #f16543;
        border-color: #ed3f14;
      }
      .cm-light-btn-failed:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-text {
        color: #fff;
        background-color: #fff;
        border: 1px solid #fff;
        border-radius: 4px;
      }
      .cm-light-btn-text:hover {
        color: #2DacF0;
        border-color: #fff;
      }
      .cm-light-btn-text:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-ghost {
        background: transparent!important;
        border-color: #fff;
        color: #fff;
      }
      .cm-light-btn-ghost:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-ghost-primary {
        background: transparent;
        border-color: #2d8cf0;
        color: #2d8cf0;
      }
      .cm-light-btn-ghost-primary:hover {
        border-color: #2DacF0;
        color: #2DacF0;
      }
      .cm-light-btn-ghost-danger {
        background: transparent;
        border-color: #ed3f14;
        color: #ed3f14;
      }
      .cm-light-btn-ghost-danger:hover {
        border-color: #f16543;
        color: #f16543;
      }
      .cm-light-btn-dashed {
        color: #495060;
        background-color: #fff;
        border: 1px dashed #dddee1;
        border-radius: 4px;
      }
      .cm-light-btn-dashed:hover {
        color: #2a6cF0;
        border-color: #2a6cF0;
      }
      .cm-light-btn-dashed:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-radio {
        color: #495060;
        background-color: #fff;
        border-color: #dddee1;
      }
      .cm-light-btn-radio:first-child {
        border-radius: 4px 0 0 4px;
        border: 1px solid #dddee1;
      }
      .cm-light-btn-radio:hover {
        color: #2DacF0;
      }
      .cm-light-btn-radio-checked {
        background: #fff;
        border-color: #2d8cf0;
        color: #2d8cf0;
        box-shadow: -1px 0 0 0 #2d8cf0;
      }
      .cm-light-btn-radio-checked:first-child {
        box-shadow: none!important;
        border-color: #2d8cf0;
      }
      .cm-light-btn-radio-checked:hover {
        border-color: #2DacF0;
        box-shadow: -1px 0 0 0 #2DacF0;
        color: #2DacF0;
      }
      .cm-light-btn-radio-checked:before {
        border-color: #2DacF0;
      }
      .cm-light-btn-disabled {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
        border-radius: 4px;
        cursor: not-allowed;
      }
      /*Dropdown*/
      .cm-light-dropdown-default {
        color: #2d8cf0;
        background-color: #fff;
        border: none;
      }
      .cm-light-dropdown-default:hover {
        color: #2a6cF0;
      }
      .cm-light-dropdown-btn {
        color: #fff;
        background-color: #2d8cf0;
        border: transparent;
        border-radius: 4px;
      }
      .cm-light-dropdown-btn:hover {
        background-color: #2a6cF0;
      }
      .cm-light-dropdown-menu {
        color: #495060;
        background-color: #fff;
        border: 1px solid #dddee1;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
      }
      .cm-light-dropdown-menu-item {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-dropdown-menu-item:hover {
        background-color: #dddee1;
      }
      .cm-light-dropdown-disbabled-item {
        color: #bbbec4;
        background-color: #f8f8f9;
      }
      .cm-light-dropdown-menu-current {
        background-color: #2DacF0;
      }
      /*Menu*/
      .cm-light-menu-color {
        color: #495060;
      }
      .cm-light-menu-color:hover {
        color: #2a6cF0;
      }
      .cm-light-menu-bg {
        background-color: #fff;
      }
      .cm-light-menu-border {
        border-color: #dddee1;
      }
      .cm-light-menu-border-hover {
        background-color: transparent;
      }
      .cm-light-menu-border-hover:hover {
        border-color: #2a6cF0;
      }
      .cm-light-menu-active {
        color: #2a6cF0;
        border-color: #2a6cF0;
        background-color: #ecf6fd;
      }
      .cm-light-menu-disabled {
        color: #bbbec4;
      }
      /*Pagination*/
      .cm-light-pagination-current-page {
        color: #fff;
        border-color: #2d8cf0;
        background-color: #2d8cf0;
      }
      .cm-light-pagination-current-page:hover a,
      .cm-light-pagination-current-page:focus a {
        color: #fff !important;
      }
      .cm-light-pagination-page {
        color: #495060;
        background-color: #fff;
        border-color: #dddee1;
        border-radius: 4px;
      }
      .cm-light-pagination-page:hover {
        border-color: #2d8cf0;
        color: #2d8cf0;
      }
      .cm-light-pagination-page:hover a {
        color: #2d8cf0;
      }
      .cm-light-pagination-border:hover {
        border-color: #2d8cf0;
      }
      .cm-light-pagination-text {
        color: #495060;
      }
      .cm-light-pagination-disabled {
        color: #bbbec4;
        border-color: #bbbec4;
      }
      /*Checkbox*/
      .cm-light-checkbox-checked {
        color: #2a6cF0;
      }
      .cm-light-checkbox-unchecked {
        color: #fff;
      }
      .cm-light-checkbox-label {
        color: #495060;
      }
      .cm-light-checkbox-disabled {
        color: #bbbec4;
      }
      .cm-light-checkbox-base {
        border-color: #2d8cf0;
        background-color: #2d8cf0;
      }
      /*DatePicker*/
      .cm-light-datepicker-default {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-datepicker-default:hover {
        border-color: #2a6cF0;
      }
      .cm-light-datepicker-date {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-datepicker-active-font {
        color: #495060;
      }
      .cm-light-datepicker-active-font:hover {
        background-color: #f8f8f9;
      }
      .cm-light-datepicker-unactive-font {
        color: #bbbec4;
      }
      .cm-light-datepicker-current-font {
        background-color: #2d8cf0;
      }
      .cm-light-datepicker-disabled-date {
        color: #bbbec4;
      }
      /*Radio*/
      .cm-light-radio-label {
        color: #495060;
      }
      .cm-light-radio-default {
        border-color: #dddee1;
        background: #fff;
      }
      .cm-light-radio-default:hover {
        border-color: #2d8cf0;
      }
      .cm-light-radio-bg {
        background: #fff;
      }
      .cm-light-radio-inner {
        background: #2d8cf0;
      }
      .cm-light-radio-border-checked {
        border-color: #2d8cf0;
      }
      .cm-light-radio-border-shadow {
        box-shadow: -1px 0 0 0 #2d8cf0;
      }
      .cm-light-radio-disabled {
        color: #bbbec4;
        border-color: #bbbec4 !important;
        background-color: #f4f4f4;
        cursor: not-allowed;
      }
      /*Select*/
      .cm-light-select-default {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
        background-color: #fff;
      }
      .cm-light-select-default:hover {
        border-color: #2a6cF0;
      }
      .cm-light-select-disabled {
        color: #bbbec4;
        border: 1px solid #dddee1;
        border-radius: 4px;
        background-color: #f4f4f4;
      }
      .cm-light-select-menu {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-select-menu-item {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-select-menu-item:hover {
        background-color: #dddee1;
      }
      .cm-light-select-menu-current {
        color: #fff;
        background-color: #2a6cF0;
      }
      .cm-light-select-item-disabled {
        color: #495060;
      }
      /*Slider*/
      .cm-light-slider-color {
        color: #2DacF0;
        background-color: #2DacF0;
      }
      .cm-light-slider-border-color {
        border-color: #2DacF0;
      }
      .cm-light-slider-range {
        color: #dddee1;
        background-color: #dddee1;
      }
      .cm-light-slider-disabled {
        color: #bbbec4;
        background-color: #bbbec4;
      }
      .cm-light-slider-step {
        color: #666;
        background-color: #2DacF0;
      }
      .cm-light-slider-step:hover {
        color: #666;
        background-color: #57a3f3;
      }
      .cm-light-slider-input {
        color: #495060;
        background-color: #fff;
        border: 2px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-slider-border {
        border: 1px solid #dddee1;
        border-radius: 50%;
      }
      .cm-light-slider-border:hover {
        border-color: #2DacF0;
      }
      /*Switch*/
      .cm-light-switch-default {
        background-color: #666;
      }
      .cm-light-switch-default:focus {
        box-shadow: 0 0 0 1px rgba(16, 142, 233, 0.2);
      }
      .cm-light-switch-default-bg {
        background-color: #666;
      }
      .cm-light-switch-default-color {
        color: #fff;
      }
      .cm-light-switch-open {
        background-color: #2d8cf0;
      }
      .cm-light-switch-disabled {
        background-color: #f4f4f4;
      }
      /*Popover*/
      .cm-light-popover-default {
        background-color: #fff;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-popover-title {
        color: #000;
        border-bottom: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-popovert-body {
        color: #495060;
      }
      /*Tooltip*/
      .cm-light-tooltip-default {
        color: #fff;
        background-color: #1c2438;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      /*Table*/
      .cm-light-table-thead {
        color: #000;
        background-color: #f8f8f9;
      }
      .cm-light-table-tbody {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-table-tbody:hover {
        background-color: #2a6cF0;
      }
      .cm-light-table-border {
        border-color: #dddee1;
      }
      .cm-light-table-cell {
        padding: 0;
        padding-left: 18px;
        padding-right: 18px;
        height: 40px;
      }
      .cm-light-table-tbody-td-hover {
        background-color: #ebf7ff;
      }
      /*Tabs*/
      .cm-light-tabs-default {
        color: #495060;
        background-color: #fff;
        border-bottom: 1px solid #dddee1;
      }
      .cm-light-tabs-default:hover {
        background-color: #2a6cF0;
      }
      .cm-light-tabs-current {
        color: #2a6cF0;
        border-bottom: 1px solid #2a6cF0;
      }
      .cm-light-tabs-disabled {
        color: #bbbec4;
        border-bottom: 1px solid #bbbec4;
      }
      .cm-light-tabs-hover-color {
        color: #2d8cf0;
      }
      .cm-light-tabs-hover-background {
        background-color: #2d8cf0;
      }
      /*Alert*/
      .cm-light-alert-info {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #2db7f5;
      }
      .cm-light-alert-info-color {
        color: #495060;
      }
      .cm-light-alert-info-main-color {
        color: #1c2438;
      }
      .cm-light-alert-success {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #19be6b;
      }
      .cm-light-alert-success-color {
        color: #495060;
      }
      .cm-light-alert-success-main-color {
        color: #1c2438;
      }
      .cm-light-alert-warning {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #f90;
      }
      .cm-light-alert-warning-color {
        color: #495060;
      }
      .cm-light-alert-warning-main-color {
        color: #1c2438;
      }
      .cm-light-alert-error {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #ed3f14;
      }
      .cm-light-alert-error-color {
        color: #495060;
      }
      .cm-light-alert-main-error-color {
        color: #1c2438;
      }
      /*Message*/
      .cm-light-message-default {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
      }
      /*Modal*/
      .cm-light-modal-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-light-modal-title {
        color: #495060;
      }
      .cm-light-modal-body {
        color: #495060;
      }
      .cm-light-modal-footer {
        color: #495060;
      }
      .cm-light-modal-border {
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      /*Notification*/
      .cm-light-norification-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-light-norification-title {
        color: #000;
      }
      .cm-light-norification-body {
        color: #495060;
      }
      .cm-light-norification-border {
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-norification-info {
        color: #2db7f5;
      }
      .cm-light-norification-success {
        color: #19be6b;
      }
      .cm-light-norification-warning {
        color: #f90;
      }
      .cm-light-norification-error {
        color: #ed3f14;
      }
      /*Popconfirm*/
      .cm-light-popconfirm-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
        color: #495060;
      }
      /*input*/
      .cm-light-input {
        background-color: #fff;
        border-radius: 4px;
        border-style: solid;
        border-width: 1px;
        border-color: #dddee1;
        color: #495060;
      }
      .cm-light-input:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-input:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-disabled {
        color: #bbbec4;
        background-color: #f4f4f4;
        border: 1px solid #dddee1;
      }
      .cm-light-input-disabled:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-disabled:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-group-border-last {
        border-left-width: 0;
        border-right-width: 1px;
      }
      .cm-light-input-group-border-last:hover {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-light-input-group-border-last:focus {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-light-input-group-border {
        border-right-width: 0;
      }
      .cm-light-input-group-border:hover {
        border-right-width: 1px;
      }
      .cm-input-disabled {
        color: #bbbec4;
      }
      /*progress*/
      .cm-light-progress-bg {
        background-color: #fff;
      }
      .cm-light-progress-bd {
        border-color: #f8f8f9;
      }
      .cm-light-progress-info-bg {
        background-color: #2db7f5;
      }
      .cm-light-progress-info-color {
        color: #495060;
      }
      .cm-light-progress-failed-bg {
        background-color: #ed3f14;
      }
      .cm-light-progress-failed-color {
        color: #ed3f14;
      }
      .cm-light-progress-success-bg {
        background-color: #19be6b;
      }
      .cm-light-progress-success-color {
        color: #19be6b;
      }
      /*inputNumber*/
      .cm-light-inputNumber {
        border-color: #dddee1;
        color: #495060;
      }
      .cm-light-inputNumber:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-inputNumber:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-inputNumber-inner-color {
        color: #495060;
      }
      .cm-light-inputNumber-inner-border {
        border-color: #57a3f3;
      }
      .cm-light-inputNumber-inner-disabled {
        color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-inputNumber-inner-disabled:hover {
        color: #bbbec4;
      }
      /*弹框下箭头：白色背景*/
      .cm-light-popper-arrow-down {
        border-color: transparent;
        border-width: 6px;
        border-top-color: #dddee1;
        border-bottom-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-light-popper-arrow-down:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        bottom: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-top-color: #fff;
        border-bottom-width: 0;
      }
      /*弹框上箭头：白色背景*/
      .cm-light-popper-arrow-up {
        border-color: transparent;
        border-width: 6px;
        border-bottom-color: #dddee1;
        border-top-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-light-popper-arrow-up:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        top: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-bottom-color: #fff;
        border-top-width: 0;
      }
      /*提示框下三角：黑色背景*/
      .cm-light-tooltip-arrow-down {
        border-width: 6px;
        border-style: solid;
        border-top-color: #000;
        border-bottom-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框上三角：黑色背景*/
      .cm-light-tooltip-arrow-up {
        border-width: 6px;
        border-style: solid;
        border-bottom-color: #000;
        border-top-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框左三角：黑色背景*/
      .cm-light-tooltip-arrow-left {
        border-width: 6px;
        border-style: solid;
        border-right-color: #000;
        border-left-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*提示框右三角：黑色背景*/
      .cm-light-tooltip-arrow-right {
        border-width: 6px;
        border-style: solid;
        border-left-color: #000;
        border-right-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*timeline*/
      /*timeline 圆圈颜色*/
      .cm-light-timeline-circle-blue {
        color: #2d8cf0;
        border-color: #2d8cf0;
      }
      .cm-light-timeline-circle-green {
        color: #19be6b;
        border-color: #19be6b;
      }
      .cm-light-timeline-circle-red {
        color: #ed3f14;
        border-color: #ed3f14;
      }
      /*timeline 字体颜色*/
      .cm-light-timeline-text {
        color: #2d8cf0;
      }
      /*timeline 图标颜色*/
      .cm-light-timeline-icon {
        color: #ed3f14;
      }
      /*steps*/
      .cm-light-step {
        border: 1px solid #dddee1;
        border-radius: 50%;
      }
      .cm-light-step-finish {
        border-color: #57a3f3;
        background-color: #fff;
      }
      .cm-light-step-wait {
        border-color: #dddee1;
        background-color: #fff;
      }
      .cm-light-step-progress {
        border-color: #dddee1;
        background-color: #57a3f3;
      }
      .cm-light-step-line-error {
        border-color: #ed3f14;
        background-color: #fff;
      }
      .cm-light-step-line-wait {
        box-shadow: 0 0 0 0 #dddee1;
        background: #dddee1;
      }
      .cm-light-step-line-progress {
        box-shadow: 0 0 0 0 #57a3f3;
        background: #57a3f3;
      }
      .cm-light-step-color1 {
        color: #dddee1;
      }
      .cm-light-step-color2 {
        color: #fff;
      }
      .cm-light-step-color3 {
        color: #57a3f3;
      }
      .cm-light-step-color4 {
        color: #ed3f14;
      }
      .cm-light-step-color5 {
        color: #495060;
      }
      /*cascader*/
      .cm-light-a-color {
        color: #2d8cf0;
      }
      .cm-light-a-color:hover {
        color: #2DacF0;
      }
      /*breadcrumb*/
      .cm-light-text-color {
        color: #464c5b;
      }
      .cm-light-text-color:hover {
        color: #2DacF0;
      }
      .cm-light-start-color {
        color: #657180;
      }
      .cm-light-interval-color {
        color: #dddee1;
      }
      .cm-light-lasttext-color {
        color: #1c2438;
      }
      /*upload*/
      .cm-light-texterror-color {
        color: #ed3f14;
      }
      .cm-light-border-color {
        border-color: #dddee1;
      }
      .cm-light-text-color {
        color: #495060;
      }
      .cm-light-color {
        color: #2a6cF0;
      }
      .cm-light-hover-color {
        border-color: #2a6cF0;
      }
      .cm-light-bg-color {
        background-color: #f4f4f4;
      }
      /*字体图标颜色*/
      .cm-icon-nav-1:before {
        color: #495060;
      }
      .cm-icon-nav-2:before {
        color: #495060;
      }
      .cm-icon-nav-3:before {
        color: #495060;
      }
      .cm-icon-arrow-down:before {
        color: #dddee1;
      }
      .cm-icon-basic-components:before {
        color: #dddee1;
      }
      .cm-icon-professional-components:before {
        color: #495060;
      }
      .cm-icon-checkbox-focus:before {
        color: #2DacF0;
      }
      .cm-icon-functional-components:before {
        color: #495060;
      }
      .cm-icon-document:before {
        color: #495060;
      }
      .cm-icon-checkbox-default:before {
        color: #495060;
      }
      .cm-icon-info:before {
        color: #2db7f5;
      }
      .cm-icon-arrow-up:before {
        color: #dddee1;
      }
      .cm-icon-close:before {
        color: #dddee1;
      }
      .cm-icon-clock:before {
        color: #dddee1;
      }
      .cm-icon-arrow-left:before {
        color: #dddee1;
      }
      .cm-icon-radio-disabled:before {
        color: #bbbec4;
      }
      .cm-icon-arrow-right:before {
        color: #dddee1;
      }
      .cm-icon-not-all-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-radio-default:before {
        color: #dddee1;
      }
      .cm-icon-menu-fold:before {
        color: #dddee1;
      }
      .cm-icon-menu-unfold:before {
        color: #dddee1;
      }
      .cm-icon-radio-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-error:before {
        color: #ed3f14;
      }
      .cm-icon-disabled-checked:before {
        color: #bbbec4;
      }
      .cm-icon-warning:before {
        color: #f90;
      }
      .cm-icon-checkbox-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-success:before {
        color: #19be6b;
      }
      a {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
      }
      .large {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 14px;
        padding: 12px 20px;
        height: auto;
      }
      .medium {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 14px;
        padding: 10px 20px;
        height: auto;
      }
      .small {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 12px;
        padding: 9px 15px;
        height: auto;
      }
      .mini {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 12px;
        padding: 7px 15px;
        height: auto;
      }
      .theme-light .fl {
        float: left;
      }
      .theme-light .fr {
        float: right;
      }
      .theme-light .clrfix {
        zoom: 1;
      }
      .theme-light .clrfix:after {
        content: "";
        display: block;
        height: 0;
        clear: both;
        visibility: hidden;
        overflow: hidden;
      }
      .theme-light .cm-sortselector-selection {
        display: inline-block;
        background-color: #fff;
        border-radius: 4px;
        border: 1px solid #dddee1;
        transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        max-width: 600px;
        min-width: 200px;
        position: relative;
        overflow: hidden;
      }
      .theme-light .cm-sortselector-lg.cm-sortselector-selection {
        height: 32px;
      }
      .theme-light .cm-sortselector-lg .cm-sortselector-selection__rendered {
        height: 30px;
        line-height: 30px;
      }
      .theme-light .cm-sortselector-selection__rendered {
        display: inline-block;
        margin-left: 7px;
        padding-right: 30px;
        position: relative;
        line-height: 26px;
        min-height: 26px;
        width: 100%;
        margin-bottom: -4px;
      }
      .cm-sortselector-selection__placeholder {
        position: absolute;
        top: 50%;
        left: 0;
        right: 9px;
        color: #dddee1;
        line-height: 20px;
        height: 20px;
        max-width: 100%;
        margin-top: -10px;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        text-align: left;
      }
      .theme-light .cm-sortselector-dropdown {
        top: 100%;
        left: 0;
        position: relative;
        width: 100%;
        margin-top: 4px;
        margin-bottom: 4px;
        background-color: #fff;
        box-shadow: 0 1px 6px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
        box-sizing: border-box;
        z-index: 1050;
        outline: none;
        overflow: hidden;
        font-size: 12px;
        max-width: 600px;
      }
      .theme-light .cm-sortselector-dropdown-menu {
        outline: none;
        margin-bottom: 0;
        padding-left: 0;
        list-style: none;
        max-height: 250px;
        overflow: auto;
      }
      .cm-sortselector-dropdown-title {
        width: 30px;
        height: 100%;
        padding: 7px 8px;
      }
      .theme-light .cm-sortselector-dropdown-menu-item {
        max-width: 100px;
        position: relative;
        display: block;
        padding: 4px 8px!important;
        margin: 2px 5px;
        font-weight: 400;
        color: #495060;
        white-space: nowrap;
        cursor: pointer;
        overflow: hidden;
        text-overflow: ellipsis;
        transition: background .3s ease;
        border-radius: 4px;
      }
      .theme-light .cm-sortselector-dropdown-menu-item:hover {
        background-color: #2d8cf0;
        color: #fff;
      }
      .theme-light .cm-sortselector-dropdown-menu-item.isSelected {
        background-color: #2d8cf0;
        color: #fff;
      }
      .theme-light .cm-sortselector-arrow {
        font-style: normal;
        vertical-align: baseline;
        text-align: center;
        text-transform: none;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        position: absolute;
        top: 50%;
        right: 8px;
        line-height: 1;
        margin-top: -6px;
        color: rgba(0, 0, 0, 0.43);
        display: inline-block;
        font-size: 12px;
        font-size: 9px\\9;
        -webkit-transform: scale(0.75) rotate(0deg);
        transform: scale(0.75) rotate(0deg);
        zoom: 1;
      }
      .theme-light .cm-sortselector-arrow::before {
        content: "\\E61D";
        display: block;
        font-family: anticon!important;
        -webkit-transition: -webkit-transform 0.2s ease;
        transition: -webkit-transform 0.2s ease;
        transition: transform .2s ease;
        transition: transform 0.2s ease, -webkit-transform 0.2s ease;
      }
      .theme-light .cm-sortselector-arrow.cm-sortselector-arrow-expand:before {
        -webkit-transform: rotate(180deg);
        transform: rotate(180deg);
      }
      .theme-light .cm-sortselector-lg .ant-select-selection--multiple .ant-select-selection__choice {
        height: 24px;
        line-height: 24px;
      }
      .theme-light .cm-sortselector-lg .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove {
        top: 4px;
      }
      .theme-light .ant-select-selection--multiple .ant-select-selection__choice {
        color: rgba(0, 0, 0, 0.65);
        background-color: #f3f3f3;
        border-radius: 4px;
        cursor: default;
        float: left;
        margin-right: 4px;
        max-width: 99%;
        position: relative;
        overflow: hidden;
        transition: padding 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        padding: 0 20px 0 10px;
        margin-top: 3px;
        height: 20px;
        line-height: 20px;
      }
      .theme-light .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__content {
        max-width: 70px;
        display: inline-block;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        transition: margin 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
      }
      .theme-light .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove {
        font-style: normal;
        vertical-align: middle;
        text-align: center;
        text-transform: none;
        line-height: 1;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        color: rgba(0, 0, 0, 0.43);
        line-height: inherit;
        cursor: pointer;
        font-weight: 700;
        transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        display: inline-block;
        font-size: 12px;
        font-size: 8px\\9;
        -webkit-transform: scale(0.66666667) rotate(0deg);
        transform: scale(0.66666667) rotate(0deg);
        zoom: 1;
        position: absolute;
        right: 10px;
        padding: 0 0 0 8px;
        top: 3px;
      }
      .theme-light .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove::before {
        content: "\\E633";
        display: block;
        font-family: anticon!important;
        position: absolute;
      }
      .theme-light .ant-select-selection--multiple .ant-select-search--inline {
        width: auto;
        padding: 0;
        max-width: 100%;
        margin-top: 3px;
        height: 20px;
        line-height: 20px;
        float: left;
        position: static;
      }
      .theme-light .ant-select-selection--multiple .ant-select-search--inline .ant-select-search__field__wrap {
        width: 100%;
        height: 100%;
        display: inline-block;
        position: relative;
      }
      .theme-light .ant-select-selection--multiple .ant-select-search--inline .ant-select-search__field__wrap .ant-select-search__field {
        border-width: 0;
        font-size: 100%;
        height: 100%;
        width: 100%;
        background: transparent;
        outline: 0;
        border-radius: 4px;
        line-height: 1;
        max-width: 100%;
        width: .75em;
      }
      /* button*/
      .cm-dark-btn-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-btn-default:hover {
        color: #080F26;
        border-color: #080F26;
      }
      .cm-dark-btn-default:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-info {
        color: #fff;
        background-color: #0D6DA8;
        border: 1px solid #0D6DA8;
        border-radius: 4px;
      }
      .cm-dark-btn-info:hover {
        color: #fff;
        background-color: #1B90CF;
      }
      .cm-dark-btn-info:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-primary {
        color: #fff;
        background-color: #14224D;
        border: 1px solid #14224D;
        border-radius: 4px;
      }
      .cm-dark-btn-primary:hover {
        color: #fff;
        background-color: #253459;
        border-color: #14224D;
      }
      .cm-dark-btn-primary:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-success {
        color: #fff;
        background-color: #037343;
        border: 1px solid #037343;
        border-radius: 4px;
      }
      .cm-dark-btn-success:hover {
        color: #fff;
        background-color: #0C9957;
        border-color: #037343;
      }
      .cm-dark-btn-success:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-warning {
        color: #fff;
        background-color: #B35F00;
        border: 1px solid #B35F00;
        border-radius: 4px;
      }
      .cm-dark-btn-warning:hover {
        color: #fff;
        background-color: #D97B00;
        border-color: #B35F00;
      }
      .cm-dark-btn-warning:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-failed {
        color: #fff;
        background-color: #A11500;
        border: 1px solid #A11500;
        border-radius: 4px;
      }
      .cm-dark-btn-failed:hover {
        color: #fff;
        background-color: #C72606;
        border-color: #A11500;
      }
      .cm-dark-btn-failed:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-text {
        color: #fff;
        background-color: #fff;
        border: 1px solid #fff;
        border-radius: 4px;
      }
      .cm-dark-btn-text:hover {
        color: #253459;
        border-color: #fff;
      }
      .cm-dark-btn-text:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-ghost {
        background: transparent!important;
        border-color: #fff;
        color: #fff;
      }
      .cm-dark-btn-ghost:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-ghost-primary {
        background: transparent;
        border-color: #14224D;
        color: #14224D;
      }
      .cm-dark-btn-ghost-primary:hover {
        border-color: #253459;
        color: #253459;
      }
      .cm-dark-btn-ghost-danger {
        background: transparent;
        border-color: #A11500;
        color: #A11500;
      }
      .cm-dark-btn-ghost-danger:hover {
        border-color: #C72606;
        color: #C72606;
      }
      .cm-dark-btn-dashed {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px dashed #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-btn-dashed:hover {
        color: #1B90CF;
        border-color: #080F26;
      }
      .cm-dark-btn-dashed:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-radio {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border-color: #9CA0A6;
      }
      .cm-dark-btn-radio:first-child {
        border-radius: 4px 0 0 4px;
        border: 1px solid #9CA0A6;
      }
      .cm-dark-btn-radio:hover {
        color: #253459;
      }
      .cm-dark-btn-radio-checked {
        background: #fff;
        border-color: #14224D;
        color: #14224D;
        box-shadow: -1px 0 0 0 #14224D;
      }
      .cm-dark-btn-radio-checked:first-child {
        box-shadow: none!important;
        border-color: #14224D;
      }
      .cm-dark-btn-radio-checked:hover {
        border-color: #253459;
        box-shadow: -1px 0 0 0 #253459;
        color: #253459;
      }
      .cm-dark-btn-disabled {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
        border-radius: 4px;
        cursor: not-allowed;
      }
      /*Dropdown*/
      .cm-dark-dropdown-default {
        color: #14224D;
        background-color: #fff;
        border: none;
      }
      .cm-dark-dropdown-default:hover {
        color: #253459;
      }
      .cm-dark-dropdown-btn {
        color: #fff;
        background-color: #14224D;
        border: transparent;
        border-radius: 4px;
      }
      .cm-dark-dropdown-btn:hover {
        background-color: #253459;
      }
      .cm-dark-dropdown-menu {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
      }
      .cm-dark-dropdown-menu-item {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-dropdown-menu-item:hover {
        background-color: #9CA0A6;
      }
      .cm-dark-dropdown-disbabled-item {
        color: #84878C;
        background-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-dropdown-menu-current {
        background-color: #253459;
      }
      /*Menu*/
      .cm-dark-menu-color {
        color: #a4a5a7;
      }
      .cm-dark-menu-color:hover {
        color: #d3d5d8;
      }
      .cm-dark-menu-bg {
        background-color: #14224D;
      }
      .cm-dark-submenu-bg {
        background-color: #253459;
      }
      .cm-dark-menu-border {
        border-color: #9CA0A6;
      }
      .cm-dark-menu-border-hover {
        background-color: #0D6DA8;
      }
      .cm-dark-menu-border-hover:hover {
        border-color: #080F26;
      }
      .cm-dark-menu-active {
        color: #d3d5d8;
        border-color: #080F26;
        background-color: #0D6DA8;
      }
      .cm-dark-menu-disabled {
        color: #84878C;
      }
      /*Pagination*/
      .cm-dark-pagination-current-page {
        color: #fff;
        border-color: #0D6DA8;
        background-color: #0D6DA8;
      }
      .cm-dark-pagination-current-page:hover a,
      .cm-dark-pagination-current-page:focus a {
        color: #fff !important;
      }
      .cm-dark-pagination-page {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-pagination-page:hover {
        border-color: #0D6DA8;
        color: #0D6DA8;
      }
      .cm-dark-pagination-page:hover a {
        color: #0D6DA8;
      }
      .cm-dark-pagination-border:hover {
        border-color: #0D6DA8;
      }
      .cm-dark-pagination-text {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-pagination-disabled {
        color: #84878C;
        border-color: #84878C;
      }
      /*Checkbox*/
      .cm-dark-checkbox-checked {
        color: #080F26;
      }
      .cm-dark-checkbox-unchecked {
        color: #fff;
      }
      .cm-dark-checkbox-label {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-checkbox-text {
        color: #14224D;
      }
      .cm-dark-checkbox-disabled {
        color: #84878C;
      }
      .cm-dark-checkbox-move {
        border-color: #14224D;
      }
      .cm-dark-checkbox-base {
        border-color: #0D6DA8;
        background-color: #0D6DA8;
      }
      /*DatePicker*/
      .cm-dark-datepicker-default {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-datepicker-default:hover {
        border-color: #080F26;
      }
      .cm-dark-datepicker-date {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-datepicker-active-font {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-datepicker-active-font:hover {
        background-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-datepicker-unactive-font {
        color: #84878C;
      }
      .cm-dark-datepicker-current-font {
        background-color: #14224D;
      }
      .cm-dark-datepicker-disabled-date {
        color: #84878C;
      }
      /*Radio*/
      .cm-dark-radio-label {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-radio-default {
        border-color: #9CA0A6;
        background: #fff;
      }
      .cm-dark-radio-default:hover {
        border-color: #14224D;
      }
      .cm-dark-radio-bg {
        background: #fff;
      }
      .cm-dark-radio-inner {
        background: #14224D;
      }
      .cm-dark-radio-border-checked {
        border-color: #14224D;
      }
      .cm-dark-radio-border-shadow {
        box-shadow: -1px 0 0 0 #14224D;
      }
      .cm-dark-radio-disabled {
        color: #84878C;
        border-color: #84878C !important;
        background-color: #d8cbcb;
        cursor: not-allowed;
      }
      /*Select*/
      .cm-dark-select-default {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        background-color: #fff;
      }
      .cm-dark-select-default:hover {
        border-color: #080F26;
      }
      .cm-dark-select-disabled {
        color: #84878C;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        background-color: #d8cbcb;
      }
      .cm-dark-select-menu {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-select-menu-item {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-select-menu-item:hover {
        background-color: #9CA0A6;
      }
      .cm-dark-select-menu-current {
        color: #fff;
        background-color: #080F26;
      }
      .cm-dark-select-item-disabled {
        color: rgba(0, 0, 0, 0.56);
      }
      /*Slider*/
      .cm-dark-slider-color {
        color: #253459;
        background-color: #253459;
      }
      .cm-dark-slider-border-color {
        border-color: #253459;
      }
      .cm-dark-slider-range {
        color: #9CA0A6;
        background-color: #9CA0A6;
      }
      .cm-dark-slider-disabled {
        color: #84878C;
        background-color: #84878C;
      }
      .cm-dark-slider-step {
        color: rgba(0, 0, 0, 0.4);
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-slider-step:hover {
        color: rgba(0, 0, 0, 0.56);
        background-color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-slider-input {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 2px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-slider-border {
        border: 1px solid #9CA0A6;
        border-radius: 50%;
      }
      .cm-dark-slider-border:hover {
        border-color: #253459;
      }
      /*Switch*/
      .cm-dark-switch-default {
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-switch-default:focus {
        box-shadow: 0 0 0 1px rgba(28, 28, 29, 0.2);
      }
      .cm-dark-switch-default-bg {
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-switch-default-color {
        color: #fff;
      }
      .cm-dark-switch-open {
        background-color: #14224D;
      }
      .cm-dark-switch-disabled {
        background-color: #84878C;
      }
      /*Popover*/
      .cm-dark-popover-default {
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-popover-title {
        color: #000;
        border-bottom: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-popovert-body {
        color: rgba(0, 0, 0, 0.56);
      }
      /*Tooltip*/
      .cm-dark-tooltip-default {
        color: #fff;
        background-color: #1c2438;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      /*Table*/
      .cm-dark-table-thead {
        color: #000;
        background-color: #acacbb;
      }
      .cm-dark-table-tbody {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-table-tbody:hover {
        background-color: #253459;
      }
      .cm-dark-table-border {
        border-color: #9CA0A6;
      }
      .cm-dark-table-cell {
        padding: 0;
        padding-left: 18px;
        padding-right: 18px;
        height: 40px;
      }
      .cm-dark-table-tbody-td-hover {
        background-color: #ebf7ff;
      }
      /*Tabs*/
      .cm-dark-tabs-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border-bottom: 1px solid #9CA0A6;
      }
      .cm-dark-tabs-default:hover {
        background-color: #253459;
      }
      .cm-dark-tabs-current {
        color: #080F26;
        border-bottom: 1px solid #080F26;
      }
      .cm-dark-tabs-disabled {
        color: #84878C;
        border-bottom: 1px solid #84878C;
      }
      .cm-dark-tabs-hover-color {
        color: #0D6DA8;
      }
      .cm-dark-tabs-hover-background {
        background-color: #0D6DA8;
      }
      /*Alert*/
      .cm-dark-alert-info {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #0D6DA8;
      }
      .cm-dark-alert-info-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-info-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-success {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #037343;
      }
      .cm-dark-alert-success-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-success-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-warning {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #B35F00;
      }
      .cm-dark-alert-warning-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-warning-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-error {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #A11500;
      }
      .cm-dark-alert-error-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-error-main-color {
        color: #d3d5d8;
      }
      /*Message*/
      .cm-dark-message-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
      }
      /*Modal*/
      .cm-dark-modal-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-dark-modal-title {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-body {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-footer {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-border {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      /*Notification*/
      .cm-dark-norification-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-dark-norification-title {
        color: #000;
      }
      .cm-dark-norification-body {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-norification-border {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-norification-info {
        color: #0D6DA8;
      }
      .cm-dark-norification-success {
        color: #037343;
      }
      .cm-dark-norification-warning {
        color: #B35F00;
      }
      .cm-dark-norification-error {
        color: #A11500;
      }
      /*Popconfirm*/
      .cm-dark-popconfirm-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
        color: rgba(0, 0, 0, 0.56);
      }
      /*input*/
      .cm-dark-input {
        background-color: #fff;
        border-radius: 4px;
        border-style: solid;
        border-width: 1px;
        border-color: #9CA0A6;
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-input:hover {
        border: 1px solid #253459;
      }
      .cm-dark-input:focus {
        border: 1px solid #253459;
      }
      .cm-dark-input-disabled {
        color: #84878C;
        background-color: #d8cbcb;
        border: 1px solid #9CA0A6;
      }
      .cm-dark-input-disabled:hover {
        border: 1px solid #253459;
      }
      .cm-dark-input-disabled:focus {
        border: 1px solid #253459;
      }
      .cm-dark-input-group-border-last {
        border-left-width: 0;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border-last:hover {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border-last:focus {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border {
        border-right-width: 0;
      }
      .cm-dark-input-group-border:hover {
        border-right-width: 1px;
      }
      .cm-input-disabled {
        color: #84878C;
      }
      /*progress*/
      .cm-dark-progress-bg {
        background-color: #fbfbfb;
      }
      .cm-dark-progress-bd {
        border-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-progress-info-bg {
        background-color: #0D6DA8;
      }
      .cm-dark-progress-info-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-progress-failed-bg {
        background-color: #A11500;
      }
      .cm-dark-progress-failed-color {
        color: #A11500;
      }
      .cm-dark-progress-success-bg {
        background-color: #037343;
      }
      .cm-dark-progress-success-color {
        color: #037343;
      }
      /*inputNumber*/
      .cm-dark-inputNumber {
        border-color: #9CA0A6;
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber:hover {
        border: 1px solid #253459;
      }
      .cm-dark-inputNumber:focus {
        border: 1px solid #253459;
      }
      .cm-dark-inputNumber-inner-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber-inner-color:hover {
        color: #14224D;
      }
      .cm-dark-inputNumber-inner-border {
        border-color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber-inner-disabled {
        color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-inputNumber-inner-disabled:hover {
        color: #84878C;
      }
      /*弹框下箭头：白色背景*/
      .cm-dark-popper-arrow-down {
        border-color: transparent;
        border-width: 6px;
        border-top-color: #9CA0A6;
        border-bottom-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-dark-popper-arrow-down:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        bottom: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-top-color: #fff;
        border-bottom-width: 0;
      }
      /*弹框上箭头：白色背景*/
      .cm-dark-popper-arrow-up {
        border-color: transparent;
        border-width: 6px;
        border-bottom-color: #9CA0A6;
        border-top-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-dark-popper-arrow-up:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        top: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-bottom-color: #fff;
        border-top-width: 0;
      }
      /*提示框下三角：黑色背景*/
      .cm-dark-tooltip-arrow-down {
        border-width: 6px;
        border-style: solid;
        border-top-color: #000;
        border-bottom-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框上三角：黑色背景*/
      .cm-dark-tooltip-arrow-up {
        border-width: 6px;
        border-style: solid;
        border-bottom-color: #000;
        border-top-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框左三角：黑色背景*/
      .cm-dark-tooltip-arrow-left {
        border-width: 6px;
        border-style: solid;
        border-right-color: #000;
        border-left-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*提示框右三角：黑色背景*/
      .cm-dark-tooltip-arrow-right {
        border-width: 6px;
        border-style: solid;
        border-left-color: #000;
        border-right-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*timeline*/
      /*timeline 圆圈颜色*/
      .cm-dark-timeline-circle-blue {
        color: #14224D;
        border-color: #14224D;
      }
      .cm-dark-timeline-circle-green {
        color: #037343;
        border-color: #037343;
      }
      .cm-dark-timeline-circle-red {
        color: #A11500;
        border-color: #A11500;
      }
      /*timeline 字体颜色*/
      .cm-dark-timeline-text {
        color: #14224D;
      }
      /*timeline 图标颜色*/
      .cm-dark-timeline-icon {
        color: #A11500;
      }
      /*steps*/
      .cm-dark-step {
        border: 1px solid #9CA0A6;
        border-radius: 50%;
      }
      .cm-dark-step-finish {
        border-color: #253459;
        background-color: #fff;
      }
      .cm-dark-step-wait {
        border-color: #9CA0A6;
        background-color: #fff;
      }
      .cm-dark-step-progress {
        border-color: #9CA0A6;
        background-color: #253459;
      }
      .cm-dark-step-line-error {
        border-color: #A11500;
        background-color: #fff;
      }
      .cm-dark-step-line-wait {
        box-shadow: 0 0 0 0 #9CA0A6;
        background: #9CA0A6;
      }
      .cm-dark-step-line-progress {
        box-shadow: 0 0 0 0 #253459;
        background: #253459;
      }
      .cm-dark-step-color1 {
        color: #9CA0A6;
      }
      .cm-dark-step-color2 {
        color: #fff;
      }
      .cm-dark-step-color3 {
        color: #253459;
      }
      .cm-dark-step-color4 {
        color: #A11500;
      }
      .cm-dark-step-color5 {
        color: rgba(0, 0, 0, 0.56);
      }
      /*cascader*/
      .cm-dark-a-color {
        color: #14224D;
      }
      .cm-dark-a-color:hover {
        color: #253459;
      }
      /*breadcrumb*/
      .cm-dark-text-color {
        color: #d3d5d8;
      }
      .cm-dark-text-color:hover {
        color: #253459;
      }
      .cm-dark-start-color {
        color: #a4a5a7;
      }
      .cm-dark-interval-color {
        color: #9CA0A6;
      }
      .cm-dark-lasttext-color {
        color: #1c2438;
      }
      /*upload*/
      .cm-dark-texterror-color {
        color: #A11500;
      }
      .cm-dark-border-color {
        border-color: #9CA0A6;
      }
      .cm-dark-text-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-color {
        color: #080F26;
      }
      .cm-dark-hover-color {
        border-color: #080F26;
      }
      .cm-dark-bg-color {
        background-color: #d8cbcb;
      }
      /*字体图标颜色*/
      .cm-icon-nav-1:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-nav-2:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-nav-3:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-arrow-down:before {
        color: #9CA0A6;
      }
      .cm-icon-basic-components:before {
        color: #9CA0A6;
      }
      .cm-icon-professional-components:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-checkbox-focus:before {
        color: #253459;
      }
      .cm-icon-functional-components:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-document:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-checkbox-default:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-info:before {
        color: #0D6DA8;
      }
      .cm-icon-arrow-up:before {
        color: #9CA0A6;
      }
      .cm-icon-close:before {
        color: #9CA0A6;
      }
      .cm-icon-clock:before {
        color: #9CA0A6;
      }
      .cm-icon-arrow-left:before {
        color: #9CA0A6;
      }
      .cm-icon-radio-disabled:before {
        color: #84878C;
      }
      .cm-icon-arrow-right:before {
        color: #9CA0A6;
      }
      .cm-icon-not-all-checked:before {
        color: #14224D;
      }
      .cm-icon-radio-default:before {
        color: #9CA0A6;
      }
      .cm-icon-menu-fold:before {
        color: #9CA0A6;
      }
      .cm-icon-menu-unfold:before {
        color: #9CA0A6;
      }
      .cm-icon-radio-checked:before {
        color: #14224D;
      }
      .cm-icon-error:before {
        color: #A11500;
      }
      .cm-icon-disabled-checked:before {
        color: #84878C;
      }
      .cm-icon-warning:before {
        color: #B35F00;
      }
      .cm-icon-checkbox-checked:before {
        color: #14224D;
      }
      .cm-icon-success:before {
        color: #037343;
      }
      .theme-dark .fl {
        float: left;
      }
      .theme-dark .fr {
        float: right;
      }
      .theme-dark .clrfix {
        zoom: 1;
      }
      .theme-dark .clrfix:after {
        content: "";
        display: block;
        height: 0;
        clear: both;
        visibility: hidden;
        overflow: hidden;
      }
      .theme-dark .cm-sortselector-selection {
        display: inline-block;
        background-color: #fff;
        border-radius: 4px;
        border: 1px solid #9CA0A6;
        transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        max-width: 600px;
        min-width: 200px;
        position: relative;
        height: 28px;
        overflow: hidden;
      }
      .theme-dark .cm-sortselector-lg.cm-sortselector-selection {
        height: 32px;
      }
      .theme-dark .cm-sortselector-lg .cm-sortselector-selection__rendered {
        height: 30px;
        line-height: 30px;
      }
      .theme-dark .cm-sortselector-selection__rendered {
        display: inline-block;
        margin-left: 7px;
        padding-right: 30px;
        position: relative;
        line-height: 26px;
        min-height: 28px;
        width: 100%;
      }
      .cm-sortselector-selection__placeholder {
        position: absolute;
        top: 50%;
        left: 0;
        right: 9px;
        color: #9CA0A6;
        line-height: 20px;
        height: 20px;
        max-width: 100%;
        margin-top: -10px;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        text-align: left;
      }
      .theme-dark .cm-sortselector-dropdown {
        top: 100%;
        left: 0;
        position: relative;
        width: 100%;
        margin-top: 4px;
        margin-bottom: 4px;
        background-color: #fff;
        box-shadow: 0 1px 6px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
        box-sizing: border-box;
        z-index: 1050;
        outline: none;
        overflow: hidden;
        font-size: 12px;
        max-width: 600px;
      }
      .theme-dark .cm-sortselector-dropdown-menu {
        outline: none;
        margin-bottom: 0;
        padding-left: 0;
        list-style: none;
        max-height: 250px;
        overflow: auto;
      }
      .cm-sortselector-dropdown-title {
        width: 30px;
        height: 100%;
        padding: 7px 8px;
      }
      .theme-dark .cm-sortselector-dropdown-menu-item {
        max-width: 100px;
        position: relative;
        display: block;
        padding: 4px 8px!important;
        margin: 2px 5px;
        font-weight: 400;
        color: -dark-text-color;
        white-space: nowrap;
        cursor: pointer;
        overflow: hidden;
        text-overflow: ellipsis;
        transition: background .3s ease;
        border-radius: 4px;
      }
      .theme-dark .cm-sortselector-dropdown-menu-item:hover {
        background-color: #14224D;
        color: #fff;
      }
      .theme-dark .cm-sortselector-dropdown-menu-item.isSelected {
        background-color: #14224D;
        color: #fff;
      }
      .theme-dark .cm-sortselector-arrow {
        font-style: normal;
        vertical-align: baseline;
        text-align: center;
        text-transform: none;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        position: absolute;
        top: 50%;
        right: 8px;
        line-height: 1;
        margin-top: -6px;
        color: rgba(0, 0, 0, 0.43);
        display: inline-block;
        font-size: 12px;
        font-size: 9px\\9;
        -webkit-transform: scale(0.75) rotate(0deg);
        transform: scale(0.75) rotate(0deg);
        zoom: 1;
      }
      .theme-dark .cm-sortselector-arrow::before {
        content: "\\E61D";
        display: block;
        font-family: anticon!important;
      }
      .theme-dark .cm-sortselector-lg .ant-select-selection--multiple .ant-select-selection__choice {
        height: 24px;
        line-height: 24px;
      }
      .theme-dark .cm-sortselector-lg .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove {
        top: 4px;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-selection__choice {
        color: rgba(0, 0, 0, 0.65);
        background-color: #f3f3f3;
        border-radius: 4px;
        cursor: default;
        float: left;
        margin-right: 4px;
        max-width: 99%;
        position: relative;
        overflow: hidden;
        transition: padding 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        padding: 0 20px 0 10px;
        margin-top: 3px;
        height: 20px;
        line-height: 20px;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__content {
        max-width: 70px;
        display: inline-block;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        transition: margin 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
      }
      .theme-dark .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove {
        font-style: normal;
        vertical-align: middle;
        text-align: center;
        text-transform: none;
        line-height: 1;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        color: rgba(0, 0, 0, 0.43);
        line-height: inherit;
        cursor: pointer;
        font-weight: 700;
        transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
        display: inline-block;
        font-size: 12px;
        font-size: 8px\\9;
        -webkit-transform: scale(0.66666667) rotate(0deg);
        transform: scale(0.66666667) rotate(0deg);
        zoom: 1;
        position: absolute;
        right: 10px;
        padding: 0 0 0 8px;
        top: 3px;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-selection__choice .ant-select-selection__choice__remove::before {
        content: "\\E633";
        display: block;
        font-family: anticon!important;
        position: absolute;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-search--inline {
        width: auto;
        padding: 0;
        max-width: 100%;
        margin-top: 3px;
        height: 20px;
        line-height: 20px;
        float: left;
        position: static;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-search--inline .ant-select-search__field__wrap {
        width: 100%;
        height: 100%;
        display: inline-block;
        position: relative;
      }
      .theme-dark .ant-select-selection--multiple .ant-select-search--inline .ant-select-search__field__wrap .ant-select-search__field {
        border-width: 0;
        font-size: 100%;
        height: 100%;
        width: 100%;
        background: transparent;
        outline: 0;
        border-radius: 4px;
        line-height: 1;
        max-width: 100%;
        width: .75em;
      }
    `],
        providers: [
            {
                provide: forms_1.NG_VALUE_ACCESSOR,
                useExisting: core_1.forwardRef(() => SortselectorComponent_1),
                multi: true
            }
        ],
    }),
    __metadata("design:paramtypes", [core_1.ElementRef, core_1.Renderer2, core_1.ChangeDetectorRef])
], SortselectorComponent);
exports.SortselectorComponent = SortselectorComponent;
