// ng-cosmos-td-ui/src/business/tree/TransversalTree/TransversalTree.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
let TransversalTreeComponent = class TransversalTreeComponent {
    constructor() {
        this.isOnly = false;
    }
    set cmData(cmData) {
        this._cmData = cmData;
    }
    ngOnInit() {
        if (this._cmData.length == 1) {
            this.isOnly = true;
        }
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], TransversalTreeComponent.prototype, "cmData", null);
TransversalTreeComponent = __decorate([
    core_1.Component({
        selector: 'cm-TransversalTree',
        template: `

      <div *ngFor="let comment of _cmData" class="ng-star-0" [ngClass]="isOnly?'only':''">
      	<div class="TransversalTree">
      		<div class="TransversalTree-body">
      			<div class="TransversalTree-box" [ngClass]="comment.cComments != false&&comment.cComments.isShow ?'TransversalTreeBox':'' ">
      				{{ comment.content }}
      				<div class="TransversalTree-btn" *ngIf="comment.cComments != false">
      					<i class="anticon anticon-plus" (click)="comment.cComments.isShow = true" *ngIf="!comment.cComments.isShow"></i>
      					<i class="anticon anticon-minus" (click)="comment.cComments.isShow = false" *ngIf="comment.cComments.isShow"></i>
      				</div>
      			</div>
      		</div>
      		<cm-TransversalTree *ngIf="comment.cComments != false&&comment.cComments.isShow" [cmData]="comment.cComments" class="TransversalTree-item"></cm-TransversalTree>
      	</div>
      </div>
    `,
        styles: [`
      .TransversalTree {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        height: 100%;
      }
      .TransversalTree-box {
        border: 2px solid #ddd;
        width: 120px;
        height: 40px;
        line-height: 40px;
        text-align: center;
        border-radius: 5%;
        position: relative;
        margin: 7px;
      }
      .TransversalTree-item {
        display: flex;
        flex-direction: column;
        justify-content: center;
        margin-left: 20px;
      }
      .TransversalTree-btn {
        width: 14%;
        height: 43%;
        background: #fff;
        border: 1px solid #ddd;
        border-radius: 50%;
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        left: 93%;
        top: 33%;
      }
      .TransversalTree-btn i {
        color: #2d8cf0;
        font-size: 12px;
      }
      .ng-star-0.ng-star-inserted {
        position: relative;
      }
      .ng-star-0.ng-star-inserted:after {
        content: '';
        position: absolute;
        top: 0px;
        left: -10px;
        background: #ddd;
        height: 100%;
        width: 2px;
      }
      .ng-star-0.ng-star-inserted:first-child:after {
        height: 50%;
        top: 50%;
      }
      .ng-star-0.ng-star-inserted:last-child:after {
        height: 50%;
      }
      .only.ng-star-0.ng-star-inserted:last-child:after {
        height: 0;
      }
      .ng-star-0.ng-star-inserted:before {
        content: '';
        position: absolute;
        left: -10px;
        top: 50%;
        background: #ddd;
        height: 2px;
        width: 17px;
      }
      .TransversalTreeBox::before {
        content: '';
        position: absolute;
        left: 103%;
        top: 50%;
        background: #ddd;
        height: 2px;
        width: 17px;
      }
    `]
    })
], TransversalTreeComponent);
exports.TransversalTreeComponent = TransversalTreeComponent;
