// ng-cosmos-td-ui/src/chart/common/chart.base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const echarts = require("echarts");
const timer_1 = require("./../../utils/timer");
class BaseChart {
    constructor() {
        // 绑定 resize 函数
        this.resizeBindHandler = this.resizeHandler.bind(this);
        // 默认值
        this.resizeEnable = true;
    }
    // 供外部引用的 echarts 实例
    get echarts() {
        return this.chart;
    }
    // 设置 echarts 容器，此时会自动初始化 chart
    set chartContainer(container) {
        if (container) {
            setTimeout(() => {
                this.chart = echarts.init(container);
                this.init();
            }, 300);
        }
    }
    init() {
        if (this.chart && this.chartOption) {
            this.inputOption();
        }
    }
    inputOption() {
        if (this.chart && this.chartOption) {
            this.chart.setOption(this.chartOption, true);
            this.onInitChart();
        }
    }
    onInitChart() {
        // console.log('onInitChart...');
    }
    disposeChart() {
        this.resizeEnable = false;
        if (this.chart) {
            this.chart.dispose();
            this.chart = null;
        }
        this.onDestroyChart();
    }
    onDestroyChart() {
        // console.log('onDisposeChart...');
    }
    // 缩放窗口时自动缩放
    set resizeEnable(resize) {
        if (resize) {
            this.timer = new timer_1.Timer();
            window.addEventListener('resize', this.resizeBindHandler, false);
        }
        else {
            this.timer = null;
            window.removeEventListener('resize', this.resizeBindHandler, false);
        }
    }
    resizeHandler() {
        // console.log('resizeHandler')
        this.timer.startTimeOut(0.5, () => {
            this.timer.stopTimeOut();
            if (this.chart) {
                this.chart.resize();
            }
        }, this);
    }
}
exports.BaseChart = BaseChart;
