// ng-cosmos-td-ui/src/chart/common/chart.common.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const chart_base_1 = require("./chart.base");
class ChartCommon extends chart_base_1.BaseChart {
    constructor() {
        super();
        this.chartOption = this.defineChartOption();
    }
    // 定义 echarts option
    defineChartOption() {
        return {};
    }
    // 通用option
    set option(option) {
        this.mergeChartOption(option);
    }
    set chartData(data) {
        this._chartData = data;
        this.mergeChartOption(this.chartDataChange(data));
    }
    chartDataChange(data) {
        return null;
    }
    set chartSettings(settings) {
        this._chartSettings = settings;
        this.mergeChartOption(this.chartSettingsChange(settings));
    }
    chartSettingsChange(Settings) {
        return null;
    }
    mergeChartOption(option) {
        Object.assign(this.chartOption, option);
        if (this.echarts && this.chartOption) {
            this.inputOption();
        }
    }
}
__decorate([
    core_1.ViewChild('canvas'),
    __metadata("design:type", core_1.ElementRef)
], ChartCommon.prototype, "canvas", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], ChartCommon.prototype, "option", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], ChartCommon.prototype, "chartData", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], ChartCommon.prototype, "chartSettings", null);
exports.ChartCommon = ChartCommon;
