// ng-cosmos-td-ui/src/business/business.component.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var BusinessComponentModule_1;
const core_1 = require("@angular/core");
const gender_module_1 = require("./gender/gender.module");
const age_module_1 = require("./age/age.module");
const help_module_1 = require("./help/help.module");
const retention_module_1 = require("./retention/retention.module");
const sortselector_module_1 = require("./sortselector/sortselector.module");
const tree_module_1 = require("./tree/tree.module");
const checkcascader_module_1 = require("./checkcascader/checkcascader.module");
const advancedfilter_module_1 = require("./advancedfilter/advancedfilter.module");
const colorPicker_module_1 = require("./colorPicker/colorPicker.module");
// 导出组件
var gender_component_1 = require("./gender/gender.component");
exports.GenderComponent = gender_component_1.GenderComponent;
var help_component_1 = require("./help/help.component");
exports.HelpComponent = help_component_1.HelpComponent;
var retention_component_1 = require("./retention/retention.component");
exports.RetentionComponent = retention_component_1.RetentionComponent;
var sortselector_component_1 = require("./sortselector/sortselector.component");
exports.SortselectorComponent = sortselector_component_1.SortselectorComponent;
var dragSort_component_1 = require("./dragsort/dragSort.component");
exports.CosmosDragSortComponent = dragSort_component_1.CosmosDragSortComponent;
var tree_component_1 = require("./tree/tree.component");
exports.TreeComponent = tree_component_1.TreeComponent;
var table_component_1 = require("./table/table.component");
exports.TableDocComponent = table_component_1.TableDocComponent;
var colorPicker_component_1 = require("./colorPicker/colorPicker.component");
exports.colorPickerComponent = colorPicker_component_1.colorPickerComponent;
// 导出组件模块
var gender_module_2 = require("./gender/gender.module");
exports.GenderModule = gender_module_2.GenderModule;
var age_module_2 = require("./age/age.module");
exports.AgeModule = age_module_2.AgeModule;
var help_module_2 = require("./help/help.module");
exports.HelpModule = help_module_2.HelpModule;
var retention_module_2 = require("./retention/retention.module");
exports.RetentionModule = retention_module_2.RetentionModule;
var sortselector_module_2 = require("./sortselector/sortselector.module");
exports.SortselectorModule = sortselector_module_2.SortselectorModule;
var tree_module_2 = require("./tree/tree.module");
exports.TreeModule = tree_module_2.TreeModule;
var checkcascader_module_2 = require("./checkcascader/checkcascader.module");
exports.CheckCascaderModule = checkcascader_module_2.CheckCascaderModule;
var advancedfilter_module_2 = require("./advancedfilter/advancedfilter.module");
exports.AdvancedFilterModule = advancedfilter_module_2.AdvancedFilterModule;
var colorPicker_module_2 = require("./colorPicker/colorPicker.module");
exports.colorPickerModule = colorPicker_module_2.colorPickerModule;
let BusinessComponentModule = BusinessComponentModule_1 = class BusinessComponentModule {
    static forRoot() {
        return {
            ngModule: BusinessComponentModule_1,
            providers: []
        };
    }
};
BusinessComponentModule = BusinessComponentModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            gender_module_1.GenderModule,
            age_module_1.AgeModule,
            help_module_1.HelpModule,
            retention_module_1.RetentionModule,
            sortselector_module_1.SortselectorModule,
            tree_module_1.TreeModule,
            checkcascader_module_1.CheckCascaderModule,
            advancedfilter_module_1.AdvancedFilterModule,
            colorPicker_module_1.colorPickerModule,
        ]
    })
], BusinessComponentModule);
exports.BusinessComponentModule = BusinessComponentModule;
exports.businessComponentModuleArray = [
    gender_module_1.GenderModule,
    age_module_1.AgeModule,
    help_module_1.HelpModule,
    retention_module_1.RetentionModule,
    sortselector_module_1.SortselectorModule,
    tree_module_1.TreeModule,
    checkcascader_module_1.CheckCascaderModule,
    advancedfilter_module_1.AdvancedFilterModule,
    colorPicker_module_1.colorPickerModule,
];
