// ng-cosmos-td-ui/src/business/table/highlightColor.directive.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
/* tslint:disable:member-ordering */
const core_1 = require("@angular/core");
let highlightcolor = class highlightcolor {
    constructor(el) {
        this.el = el;
        this.el;
    }
    set highlightcolor(obj) {
        this.highlightColor = obj.highlightcolor;
        this._Index = obj.Index;
        // debugger
    }
    ;
    onMouseEnter() {
        this.el;
        this.highlight(this.highlightColor || '#ebf7ff', this._Index);
    }
    onMouseLeave() {
        this.highlight(null, this._Index);
    }
    highlight(background, index) {
        this.el.nativeElement.style.backgroundColor = background;
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], highlightcolor.prototype, "highlightcolor", null);
__decorate([
    core_1.HostListener('mouseenter'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], highlightcolor.prototype, "onMouseEnter", null);
__decorate([
    core_1.HostListener('mouseleave'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], highlightcolor.prototype, "onMouseLeave", null);
highlightcolor = __decorate([
    core_1.Directive({
        selector: '[highlightcolor]'
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], highlightcolor);
exports.highlightcolor = highlightcolor;
/*
Copyright 2017 Google Inc. All Rights Reserved.
Use of this source code is governed by an MIT-style license that
can be found in the LICENSE file at http://angular.io/license
*/
