// ng-cosmos-td-ui/src/business/table/table.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const forms_1 = require("@angular/forms");
const ng_cs_antd_1 = require("ng-cs-antd");
const common_1 = require("@angular/common");
const cmttable_directive_1 = require("./cmttable.directive");
const cmtborderbottom_directive_1 = require("./cmtborderbottom.directive");
const cmtborderright_directive_1 = require("./cmtborderright.directive");
const highlightColor_directive_1 = require("./highlightColor.directive");
// import { RowHighlight } from './cmthighlight.directive';
const table_component_1 = require("./table.component");
let CosmosTableModule = class CosmosTableModule {
    constructor() {
    }
    ngOnInit() {
    }
    ngOnDestroy() {
    }
};
CosmosTableModule = __decorate([
    core_1.NgModule({
        declarations: [
            table_component_1.TableDocComponent,
            cmttable_directive_1.cmttable,
            cmtborderbottom_directive_1.cmtborderbottom,
            cmtborderright_directive_1.cmtborderright,
            highlightColor_directive_1.highlightcolor,
        ],
        imports: [
            common_1.CommonModule,
            forms_1.FormsModule,
            forms_1.ReactiveFormsModule,
            ng_cs_antd_1.NzPaginationModule
        ],
        exports: [table_component_1.TableDocComponent, cmttable_directive_1.cmttable, cmtborderbottom_directive_1.cmtborderbottom, highlightColor_directive_1.highlightcolor, cmtborderright_directive_1.cmtborderright],
    }),
    __metadata("design:paramtypes", [])
], CosmosTableModule);
exports.CosmosTableModule = CosmosTableModule;
