// ng-cosmos-td-ui/src/business/checkcascader/checkcascader.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var CheckCascaderComponent_1;
const core_1 = require("@angular/core");
const forms_1 = require("@angular/forms");
const keycodes_1 = require("@angular/cdk/keycodes");
const cloneDeep = require('lodash.clonedeep');
const ESC = 27;
function toBoolean(value) {
    return value === '' || (value && value !== 'false');
}
function toArray(value) {
    let ret;
    if (value == null) {
        ret = [];
    }
    else if (!Array.isArray(value)) {
        ret = [value];
    }
    else {
        ret = value;
    }
    return ret;
}
function arrayEquals(array1, array2) {
    if (!array1 || !array2 || array1.length !== array2.length) {
        return false;
    }
    const len = array1.length;
    for (let i = 0; i < len; i++) {
        if (array1[i] !== array2[i]) {
            return false;
        }
    }
    return true;
}
const defaultDisplayRender = (label) => label.join(' ; ');
const defaultIntervalRender = (label) => label.join('、');
let CheckCascaderComponent = CheckCascaderComponent_1 = class CheckCascaderComponent {
    constructor(_elementRef, _render, _cdr) {
        this._elementRef = _elementRef;
        this._render = _render;
        this._cdr = _cdr;
        this.allowClear = true;
        this.disabled = false;
        this.enableCache = true;
        this.showArrow = true;
        this.showInput = true;
        this.showSearch = false;
        this.changeOnSelect = false;
        this._cmPlaceHolder = "Please select";
        this._dropDownPosition = 'bottom';
        this._prefixCls = 'ant-cascader';
        this._inputPrefixCls = 'ant-input';
        this._focused = false;
        this.menuVisible = false;
        this._displayLabelIsTemplate = false;
        this._displayLabelContext = {};
        /* tslint:disable-next-line:variable-name */
        this.__inputValue = '';
        this._searchValue = [];
        // selection will trigger value change
        this._selectedOptions = [];
        // activaction will not trigger value change
        this._activatedOptions = [];
        // all data columns
        this._cmColumns = [];
        this._filterColumns = [];
        // ngModel Access
        this.onChange = Function.prototype;
        this.onTouched = Function.prototype;
        /** Input size, one of `large` `default` `small` */
        this.cmSize = 'default';
        /** Hover text for the clear icon */
        this.cmClearText = 'Clear';
        /** Specify content to show when no result matches. */
        this.cmNotFoundContent = 'Not Found';
        /** Expand column item when click or hover, one of 'click' 'hover' */
        this.cmExpandTrigger = 'click';
        /** Delay time to show when mouse enter, when `cmExpandTrigger` is `hover`. */
        this.cmMouseEnterDelay = 150; // ms
        /** Delay time to hide when mouse enter, when `cmExpandTrigger` is `hover`. */
        this.cmMouseLeaveDelay = 150; // ms
        /** Triggering mode: can be Array<'click'|'hover'> */
        this.cmTriggerAction = ['click'];
        /** Property name for getting `value` in the option */
        this.cmValueProperty = 'value';
        /** Property name for getting `label` in the option */
        this.cmLabelProperty = 'label';
        this.tabIndex = '0';
        /** Event: emit on popup show or hide */
        this.cmVisibleChange = new core_1.EventEmitter();
        /** Event: emit on values changed */
        this.cmChange = new core_1.EventEmitter();
        /** Event: emit on values and selection changed */
        this.cmSelectionChange = new core_1.EventEmitter();
        /**
         * Event: emit on option selected, event data：{option: any, index: number}
         */
        this.cmSelect = new core_1.EventEmitter();
        /**
         * Event: emit on option unselected, event data：{option: any, index: number}
         */
        this.cmUnSelect = new core_1.EventEmitter();
        /** Event: emit on the clear button clicked */
        this.cmClear = new core_1.EventEmitter();
        this._el = this._elementRef.nativeElement;
    }
    get _inputValue() {
        return this.__inputValue;
    }
    set _inputValue(inputValue) {
        this.__inputValue = inputValue;
        if (inputValue.length) {
            this._addHostClass(`${this._prefixCls}-picker-with-value`);
        }
        else {
            this._removeHostClass(`${this._prefixCls}-picker-with-value`);
        }
    }
    /** Whether is disabled */
    set cmDisabled(value) {
        this.disabled = toBoolean(value);
    }
    get cmDisabled() {
        return this.disabled;
    }
    /** Input placeholder */
    set cmPlaceHolder(placeHolder) {
        this._cmPlaceHolder = placeHolder;
    }
    /** Whether show input box. Defaults to `true`. */
    set cmShowInput(value) {
        this.showInput = toBoolean(value);
    }
    get cmShowInput() {
        return this.showInput;
    }
    /** Whether can search. Defaults to `false`. */
    set cmShowSearch(value) {
        this.showSearch = toBoolean(value);
    }
    get cmShowSearch() {
        return this.showSearch;
    }
    /** Whether allow clear. Defaults to `true`. */
    set cmAllowClear(value) {
        this.allowClear = toBoolean(value);
    }
    get cmAllowClear() {
        return this.allowClear;
    }
    /** Whether to show arrow */
    set cmShowArrow(value) {
        this.showArrow = toBoolean(value);
    }
    get cmShowArrow() {
        return this.showArrow;
    }
    /** Whether cache children when they were loaded asych */
    set cmEnableCache(value) {
        this.enableCache = toBoolean(value);
    }
    get cmEnableCache() {
        return this.enableCache;
    }
    /** Change value on each selection if set to true */
    set cmChangeOnSelect(value) {
        this.changeOnSelect = toBoolean(value);
    }
    get cmChangeOnSelect() {
        return this.changeOnSelect;
    }
    onPositionChange(position) {
        const _position = position.connectionPair.originY === 'bottom' ? 'bottom' : 'top';
        if (this._dropDownPosition !== _position) {
            this._dropDownPosition = _position;
            this._cdr.detectChanges();
        }
    }
    cmFocus() {
        this._focused = true;
        this._addHostClass(`${this._prefixCls}-focused`);
    }
    cmBlur() {
        this._focused = false;
        this._removeHostClass(`${this._prefixCls}-focused`);
    }
    get _pickerLabelCls() {
        return {
            [`${this._prefixCls}-picker-label`]: true
        };
    }
    get _arrowCls() {
        return {
            [`${this._prefixCls}-picker-arrow`]: true,
            [`${this._prefixCls}-picker-arrow-expand`]: this.menuVisible
        };
    }
    get _clearCls() {
        return {
            [`${this._prefixCls}-picker-clear`]: true
        };
    }
    get _inputCls() {
        return {
            [`${this._prefixCls}-input`]: 1,
            [`${this._inputPrefixCls}-disabled`]: this.cmDisabled,
            [`${this._inputPrefixCls}-lg`]: this.cmSize === 'large',
            [`${this._inputPrefixCls}-sm`]: this.cmSize === 'small',
        };
    }
    get _menuCls() {
        return {
            [`${this._prefixCls}-menus`]: true,
            [`${this._prefixCls}-menus-hidden`]: !this.menuVisible,
            [`${this.cmPopupClassName}`]: this.cmPopupClassName
        };
    }
    /** 获取菜单中列的样式 */
    get _columnCls() {
        return {
            [`${this._prefixCls}-menu`]: true,
            [`${this.cmColumnClassName}`]: this.cmColumnClassName
        };
    }
    /** 获取列中Option的样式 */
    getOptionCls(option, index) {
        return {
            [`${this._prefixCls}-menu-item`]: true,
            [`${this._prefixCls}-menu-item-expand`]: !option.isLeaf,
            [`${this._prefixCls}-menu-item-active`]: this.isActiveOption(option, index),
            [`${this._prefixCls}-menu-item-disabled`]: option.disabled,
            [`${this._prefixCls}-menu-item-loading`]: option.loading
        };
    }
    _getLabel() {
        return this._displayLabelIsTemplate ? '' : this._displayLabel;
    }
    /** prevent input change event */
    _handlerInputChange(event) {
        event.stopPropagation();
    }
    /** input blur */
    _handleInputBlur(event) {
        if (!this.cmShowSearch) {
            return;
        }
        if (this.menuVisible) {
            this.cmFocus();
        }
        else {
            this.cmBlur();
        }
    }
    /** input focus */
    _handleInputFocus(event) {
        if (!this.cmShowSearch) {
            return;
        }
        this.cmFocus();
    }
    /** input key down */
    _handleInputKeyDown(event) { }
    setInputValue(inputValue, fireSearch = true) {
        if (inputValue !== this._inputValue) {
            this._inputValue = inputValue;
        }
    }
    hasInput() {
        return this._inputValue.length > 0;
    }
    hasSelection() {
        return this._selectedOptions.length > 0;
    }
    /** Whether the clear button is visible */
    get showClearIcon() {
        const isSelected = this.hasSelection();
        const isHasInput = this.hasInput();
        return this.cmAllowClear && !this.cmDisabled && (isSelected || isHasInput);
    }
    get _displayRender() {
        return this.cmDisplayRender || defaultDisplayRender;
    }
    get _intervalRender() {
        return this.cmIntervalRender || defaultIntervalRender;
    }
    _buildDisplayLabel() {
        let labels = [];
        const selectedOptions = this._selectedOptions;
        let len = selectedOptions && selectedOptions.length || 0;
        for (let i = 0; i < len; i++) {
            let label = selectedOptions[i].map(o => o[this.cmLabelProperty || 'label']);
            labels[i] = this._intervalRender.call(this, label, selectedOptions[i]);
        }
        // 设置当前控件的显示值
        this._displayLabel = this._displayRender.call(this, labels, selectedOptions);
        this._displayLabelIsTemplate = !(typeof this._displayLabel === 'string');
        this._displayLabelContext = { labels, selectedOptions };
    }
    /** 由用户来定义点击后是否变更 */
    _isChangeOn(option, index) {
        if (typeof this.cmChangeOn === 'function') {
            return this.cmChangeOn(option, index) === true;
        }
        return false;
    }
    onKeyDown(event) {
        const keyCode = event.keyCode;
        if (keyCode !== keycodes_1.DOWN_ARROW &&
            keyCode !== keycodes_1.UP_ARROW &&
            keyCode !== keycodes_1.LEFT_ARROW &&
            keyCode !== keycodes_1.RIGHT_ARROW &&
            keyCode !== keycodes_1.ENTER &&
            keyCode !== keycodes_1.BACKSPACE &&
            keyCode !== ESC) {
            return;
        }
        // Press any keys above to reopen menu
        if (!this._isPopupVisible() && keyCode !== keycodes_1.BACKSPACE && keyCode !== ESC) {
            this.setPopupVisible(true);
            return;
        }
        // Press ESC to close menu
        if (keyCode === ESC) {
            this.setPopupVisible(false);
            return;
        }
        if (this._isPopupVisible()) {
            event.preventDefault();
            if (keyCode === keycodes_1.DOWN_ARROW) {
                this._moveDown();
            }
            if (keyCode === keycodes_1.UP_ARROW) {
                this._moveUp();
            }
            if (keyCode === keycodes_1.LEFT_ARROW) {
                this._moveLeft();
            }
            if (keyCode === keycodes_1.RIGHT_ARROW) {
                this._moveRight();
            }
        }
    }
    _onTriggerClick(event) {
        if (this.cmDisabled) {
            return;
        }
        this.onTouched(); // set your control to 'touched'
        if (this._isClickTiggerAction()) {
            this._delaySetPopupVisible(!this.menuVisible, 100);
        }
    }
    _onTriggerMouseEnter(event) {
        if (this.cmDisabled) {
            return;
        }
        if (this._isPointerTiggerAction()) {
            this._delaySetPopupVisible(true, this.cmMouseEnterDelay);
        }
    }
    _onTriggerMouseLeave(event) {
        if (this.cmDisabled) {
            return;
        }
        if (!this._isPopupVisible()) {
            return;
        }
        if (this._isPointerTiggerAction()) {
            const currEl = this._el;
            const popupEl = this.menu && this.menu.nativeElement;
            if (currEl.contains(event.target) ||
                (popupEl && popupEl.contains(event.target))) {
                return; // 还在菜单内部
            }
            this._delaySetPopupVisible(false, this.cmMouseLeaveDelay);
        }
    }
    _isClickTiggerAction() {
        if (typeof this.cmTriggerAction === 'string') {
            return this.cmTriggerAction === 'click';
        }
        return this.cmTriggerAction.indexOf('click') !== -1;
    }
    _isPointerTiggerAction() {
        if (typeof this.cmTriggerAction === 'string') {
            return this.cmTriggerAction === 'hover';
        }
        return this.cmTriggerAction.indexOf('hover') !== -1;
    }
    closeMenu() {
        this._clearDelayTimer();
        this.setPopupVisible(false);
    }
    /**
     * 显示或者隐藏菜单
     *
     * @param visible true-显示，false-隐藏
     * @param delay 延迟时间
     */
    _delaySetPopupVisible(visible, delay) {
        this._clearDelayTimer();
        if (delay) {
            this._delayTimer = setTimeout(() => {
                this.setPopupVisible(visible);
                this._clearDelayTimer();
            }, delay);
        }
        else {
            this.setPopupVisible(visible);
        }
    }
    _isPopupVisible() {
        return this.menuVisible;
    }
    setPopupVisible(popupVisible) {
        if (this.cmDisabled) {
            return;
        }
        if (this.menuVisible !== popupVisible) {
            this.menuVisible = popupVisible;
            // We must listen to `mousedown` or `touchstart`, edge case:
            // https://github.com/ant-design/ant-design/issues/5804
            // https://github.com/react-component/calendar/issues/250
            // https://github.com/react-component/trigger/issues/50
            if (popupVisible) {
                if (!this._clickOutsideHandler) {
                    this._clickOutsideHandler = this._render.listen('document', 'mousedown', this._onDocumentClick.bind(this));
                }
                // always hide on mobile
                if (!this._touchOutsideHandler) {
                    this._touchOutsideHandler = this._render.listen('document', 'touchstart', this._onDocumentClick.bind(this));
                }
            }
            if (!popupVisible) {
                this._clearOutsideHandler();
            }
            if (popupVisible) {
                this._beforeVisible();
            }
            this.cmVisibleChange.emit(popupVisible);
        }
    }
    /** load init data if necessary */
    _beforeVisible() {
        this.loadRootOptions();
    }
    loadRootOptions() {
        if (!this._cmColumns.length) {
            const root = {};
            this.loadChildren(root, -1);
        }
    }
    _onDocumentClick(event) {
        const target = event.target;
        const popupEl = this.menu && this.menu.nativeElement;
        if (!this._el.contains(target) && !popupEl.contains(target)) {
            this.setPopupVisible(false);
        }
    }
    _clearOutsideHandler() {
        if (this._clickOutsideHandler) {
            this._clickOutsideHandler(); // Removes "listen" listener
            this._clickOutsideHandler = null;
        }
        if (this._touchOutsideHandler) {
            this._touchOutsideHandler(); // Removes "listen" listener
            this._touchOutsideHandler = null;
        }
    }
    _clearDelayTimer() {
        if (this._delayTimer) {
            clearTimeout(this._delayTimer);
            this._delayTimer = null;
        }
    }
    /**
     * press `up` or `down` arrow to select the sibling option.
     */
    _moveUpOrDown(isUp) {
        const columnIndex = Math.max(this._activatedOptions.length - 1, 0);
        // 该组中已经被激活的选项
        const activeOption = this._activatedOptions[columnIndex];
        // 该组所有的选项，用于遍历获取下一个被激活的选项
        const options = this._filterColumns[columnIndex];
        if (!options || !options.length) {
            return;
        }
        const length = options.length;
        let nextOptIndex = -1;
        if (!activeOption || !activeOption.length) { // 该列还没有选中的选项
            nextOptIndex = isUp ? length : -1;
        }
        else {
            nextOptIndex = options.indexOf(activeOption);
        }
        while (true) {
            nextOptIndex = isUp ? nextOptIndex - 1 : nextOptIndex + 1;
            if (nextOptIndex < 0 || nextOptIndex >= length) {
                break;
            }
            const nextOption = options[nextOptIndex];
            if (!nextOption || nextOption.disabled) {
                continue;
            }
            //this.setActiveOption(nextOption, columnIndex);
            break;
        }
    }
    _moveUp() {
        this._moveUpOrDown(true);
    }
    _moveDown() {
        this._moveUpOrDown(false);
    }
    /**
     * press `left` arrow to remove the last selected option.
     * If there is no option selected, emit `cmClear` event.
     */
    _moveLeft() {
        const options = this._selectedOptions;
        if (options.length) {
            options.pop(); // Remove the last one
            const len = options.length;
            if (len) {
                this.setActiveOption(options[len - 1], len - 1);
            }
            else {
                this.cmClear.emit();
            }
        }
    }
    /**
     * press `right` arrow to select the next column option.
     */
    _moveRight() {
        const columns = this._filterColumns;
        const length = this._selectedOptions.length;
        if (length === 0) {
            return;
        }
        const nextColIndex = length;
        const options = columns.length > nextColIndex ? columns[nextColIndex] : null;
        if (options) { // 存在`下级选项`
            const len = options.length;
            for (let i = 0; i < len; i++) {
                const activeOpt = options[i];
                if (activeOpt && !activeOpt.disabled) {
                    this.setActiveOption(activeOpt, nextColIndex);
                    return;
                }
            }
        }
    }
    /** 获取Option的值，例如，可以指定labelProperty="name"来取Name */
    getOptionLabel(option) {
        return option[this.cmLabelProperty || 'label'];
    }
    /** 获取Option的值，例如，可以指定valueProperty="id"来取ID */
    getOptionValue(option) {
        return option[this.cmValueProperty || 'value'];
    }
    checkChange(event, option) {
    }
    /** clear the input box and selected options */
    clearSelection(event) {
        if (event) {
            event.preventDefault();
            event.stopPropagation();
        }
        this._displayLabel = '';
        this._displayLabelIsTemplate = false;
        this._searchValue = [];
        this._selectedOptions = [];
        this._activatedOptions = [];
        this._cmColumns = this._filterColumns = [];
        this.setInputValue('', false);
        this.setPopupVisible(false);
        if (this.cmOptions && this.cmOptions.length) {
            this._cmColumns.push(cloneDeep(this.cmOptions));
        }
        this._cmColumns.forEach((item, index) => {
            this._filterColumns[index] = item;
        });
        // trigger change event
        this.onValueChange();
    }
    onSearch(event, index) {
        if (!this._cmColumns || !this._cmColumns.length) {
            return;
        }
        this._filterColumns[index] = [];
        let len = this._cmColumns[index].length;
        for (let i = 0; i < len; i++) {
            if (-1 == this._cmColumns[index][i].label.indexOf(event)) {
                continue;
            }
            this._filterColumns[index].push(this._cmColumns[index][i]);
        }
    }
    filterColumnData() {
        this._filterColumns = [];
        for (let i = 0; i < this._cmColumns.length; i++) {
            if (this._searchValue[i]) {
                this.onSearch(this._searchValue[i], i);
            }
            else {
                this._filterColumns[i] = this._cmColumns[i];
            }
        }
    }
    isActiveOption(option, index) {
        const activeOpt = this._activatedOptions[index];
        if (activeOpt === option) {
            return true;
        }
        if (activeOpt && this.getOptionValue(activeOpt) === this.getOptionValue(option)) {
            return true;
        }
        return false;
    }
    setColumnData(options, index) {
        if (!options || (0 == options.length)) {
            return;
        }
        if (!arrayEquals(this._cmColumns[index], options)) {
            if (!this._cmColumns[index]) {
                this._cmColumns[index] = [].concat(options);
            }
            else {
                let colLen = this._cmColumns[index].length;
                for (let i = 0; i < options.length; i++) {
                    let find = false;
                    for (let j = 0; j < colLen; j++) {
                        if (this._cmColumns[index][j].value == options[i].value) {
                            find = true;
                            break;
                        }
                    }
                    if (!find) {
                        this._cmColumns[index].push(options[i]);
                    }
                }
            }
            this.filterColumnData();
        }
    }
    selectOption(option, index) {
        // trigger `cmSelect` event
        this.cmSelect.emit({ option, index });
        // load children directly
        if (option.children && option.children.length) {
            option.isLeaf = false;
            option.children.forEach(child => {
                child.parent = option;
                child.checked = false;
            });
            this.setColumnData(option.children, index + 1);
        }
        else if (!option.isLeaf) {
            // load children async
            this.loadChildren(option, index);
        }
        else {
            // clicking leaf node will't remove any children columns
            /*if (index < this._cmColumns.length - 1) {
             this._cmColumns = this._cmColumns.slice(0, index + 1);
             }*/
        }
    }
    loadChildren(option, index, success, failure) {
        if (this.cmLoadData) {
            this.cmLoadData(option, index).then(() => {
                if (option.children) {
                    if (option.children.length) {
                        option.children.forEach(child => {
                            child.parent = index < 0 ? undefined : option;
                            child.checked = false;
                        });
                        this.setColumnData(option.children, index + 1);
                    }
                    else {
                        option.isLeaf = true;
                    }
                }
                if (success) {
                    success();
                }
            }, () => {
                option.isLeaf = true;
                if (failure) {
                    failure();
                }
            });
        }
    }
    addActiveOption(option, index) {
        if (!option || option.disabled) {
            return;
        }
        let node = cloneDeep(option);
        delete node.checked;
        if (!this._activatedOptions[index]) {
            this._activatedOptions[index] = [];
        }
        this._activatedOptions[index].push(node);
        // 当直接选择最后一级时，前面的选项要补全。例如，选择“城市”，则自动补全“国家”、“省份”
        node = option;
        for (let i = index - 1; i >= 0; i--) {
            if (!this._activatedOptions[i]) {
                this._activatedOptions[i] = [];
            }
            if (!this._activatedOptions[i].length) {
                node.parent.checked = true;
                let pnode = cloneDeep(node.parent);
                delete pnode.checked;
                this._activatedOptions[i].push(pnode);
            }
            else {
                let tag = false;
                let currentlen = this._activatedOptions[i].length;
                for (let j = 0; j < currentlen; j++) {
                    if (this._activatedOptions[i][j].value == node.parent.value) {
                        tag = true;
                        break;
                    }
                }
                if (!tag) {
                    node.parent.checked = true;
                    let pnode = cloneDeep(node.parent);
                    delete pnode.checked;
                    this._activatedOptions[i].push(pnode);
                }
            }
            node = node.parent;
        }
    }
    resetColumnData(option, index) {
        if (!option) {
            return;
        }
        let len = this._cmColumns[index] && this._cmColumns[index].length || 0;
        for (let i = 0; i < option.length; i++) {
            for (let j = 0; j < len; j++) {
                if (option[i].value == this._cmColumns[index][j].value) {
                    this.resetColumnData(option[i].children, index + 1);
                    this._cmColumns[index].splice(j, 1);
                    break;
                }
            }
        }
        if (this._cmColumns[index] && !this._cmColumns[index].length) {
            this._cmColumns.splice(index, 1);
        }
    }
    unselectOption(option, index) {
        // trigger `cmUnSelect` event
        this.cmUnSelect.emit({ option, index });
        if (option.children && option.children.length) {
            this.resetColumnData(option.children, index + 1);
        }
    }
    delActiveOption(option, index) {
        if (!option || option.disabled) {
            return;
        }
        let len = this._activatedOptions[index] && this._activatedOptions[index].length || 0;
        // 截断多余的选项，如选择“省份”，则只会有“国家”、“省份”，去掉“城市”、“区县”
        if (index < this._activatedOptions.length - 1) {
            if (1 == len) {
                this._activatedOptions = this._activatedOptions.slice(0, index);
            }
            else {
                for (let i = 0; i < len; i++) {
                    if (option.value == this._activatedOptions[index][i].value) {
                        if (option.children && option.children.length) {
                            for (let j = 0; j < option.children.length; j++) {
                                this.delActiveOption(option.children[j], index + 1);
                            }
                        }
                        this._activatedOptions[index].splice(i, 1);
                        break;
                    }
                }
            }
        }
        else {
            for (let i = 0; i < len; i++) {
                if (option.value == this._activatedOptions[index][i].value) {
                    this._activatedOptions[index].splice(i, 1);
                    break;
                }
            }
        }
        if (this._activatedOptions[index] && !this._activatedOptions[index].length) {
            this._activatedOptions.splice(index, 1);
        }
    }
    /**
     * 设置某列的激活的菜单选项
     *
     * @param option 菜单选项
     * @param index  选项所在的列组的索引
     */
    setActiveOption(option, index) {
        if (!option || option.disabled) {
            return;
        }
        if (!this._activatedOptions[index]) {
            this._activatedOptions[index] = [];
        }
        if (option.checked) {
            // trigger select event, and display label
            this.selectOption(option, index);
            this.addActiveOption(option, index);
        }
        else {
            // trigger select event, and display label
            this.unselectOption(option, index);
            this.delActiveOption(option, index);
        }
        this.filterColumnData();
        // 生成显示
        if (option.isLeaf || this.cmChangeOnSelect || this._isChangeOn(option, index)) {
            this._selectedOptions = this._activatedOptions;
            // 设置当前控件的显示值
            this._buildDisplayLabel();
            // 触发变更事件
            this.onValueChange();
        }
    }
    /**
     * 鼠标点击选项
     *
     * @param option 菜单选项
     * @param index 选项所在的列组的索引
     * @param event 鼠标事件
     */
    onOptionClick(option, index, event) {
        event.preventDefault();
        // Keep focused state for keyboard support
        this._el.focus();
        if (option && option.disabled) {
            return;
        }
        option.checked = !option.checked;
        this.setActiveOption(option, index);
    }
    /**
     * 鼠标划入选项
     *
     * @param option 菜单选项
     * @param index 选项所在的列组的索引
     * @param event 鼠标事件
     */
    onOptionMouseEnter(option, index, event) {
        event.preventDefault();
        if (this.cmExpandTrigger === 'hover' && !option.isLeaf) {
            this.delaySelect(option, index, true);
        }
    }
    /**
     * 鼠标划出选项
     *
     * @param option 菜单选项
     * @param index 选项所在的列组的索引
     * @param event 鼠标事件
     */
    onOptionMouseLeave(option, index, event) {
        event.preventDefault();
        if (this.cmExpandTrigger === 'hover' && !option.isLeaf) {
            this.delaySelect(option, index, false);
        }
    }
    delaySelect(option, index, doSelect) {
        if (this._delayTimer) {
            clearTimeout(this._delayTimer);
            this._delayTimer = null;
        }
        if (doSelect) {
            this._delayTimer = setTimeout(() => {
                this.setActiveOption(option, index);
                this._delayTimer = null;
            }, 150);
        }
    }
    handleTreeData(option, index) {
        delete option.checked;
        let child = this._selectedOptions[index + 1];
        if (option.children && option.children.length) {
            if (child && child.length) {
                for (let i = 0; i < option.children.length; i++) {
                    let find = false;
                    for (let j = 0; j < child.length; j++) {
                        if (child[j].value == option.children[i].value) {
                            delete option.children[i].checked;
                            delete option.children[i].parent;
                            this.handleTreeData(option.children[i], index + 1);
                            find = true;
                            break;
                        }
                    }
                    if (!find) {
                        option.children.splice(i, 1);
                        i--;
                    }
                }
            }
            else {
                option.children = [];
            }
        }
        else {
            if (child && child.length) {
                for (let j = 0; j < child.length; j++) {
                    if (child[j].parent.value == option.value) {
                        const childNode = cloneDeep(child[j]);
                        if (!option.children) {
                            option.children = [];
                        }
                        delete childNode.checked;
                        delete childNode.parent;
                        option.children.push(childNode);
                        this.handleTreeData(childNode, index + 1);
                    }
                }
            }
        }
    }
    getSubmitValue() {
        let index = 0;
        if (!this._selectedOptions || !this._selectedOptions.length) {
            return [];
        }
        const values = cloneDeep(this._selectedOptions[index]);
        let len = this._selectedOptions[index].length;
        for (let i = 0; i < len; i++) {
            let node = values[i];
            this.handleTreeData(node, index);
        }
        return values;
    }
    onValueChange() {
        const value = this.getSubmitValue();
        if (!arrayEquals(this._lastValue, value)) {
            this._lastValue = value;
            this.onChange(value); // Angular need this
            if (value.length === 0) {
                this.cmClear.emit(); // first trigger `clear` and then `change`
            }
            this.cmSelectionChange.emit(this._selectedOptions);
            this.cmChange.emit(value);
        }
    }
    _addHostClass(classname) {
        this._render.addClass(this._el, classname);
    }
    _removeHostClass(classname) {
        this._render.removeClass(this._el, classname);
    }
    writeColumns(array, index) {
        if (!array || !array.length) {
            return;
        }
        for (let i = 0; i < array.length; i++) {
            let find = false;
            let len = this._cmColumns[index] && this._cmColumns[index].length || 0;
            for (let j = 0; j < len; j++) {
                if (array[i].value == this._cmColumns[index][j].value) {
                    this._cmColumns[index][j].checked = true;
                    this.writeColumns(array[i].children, index + 1);
                    find = true;
                    break;
                }
            }
            if (!find) {
                if (!this._cmColumns[index]) {
                    this._cmColumns[index] = [];
                }
                let node = cloneDeep(array[i]);
                node.checked = true;
                this._cmColumns[index].push(node);
                this.writeColumns(array[i].children, index + 1);
            }
        }
    }
    writeSelected(array, index) {
        if (!array || !array.length) {
            return;
        }
        if (!this._selectedOptions[index]) {
            this._selectedOptions[index] = [];
        }
        this._selectedOptions[index] = this._selectedOptions[index].concat(array);
        for (let i = 0; i < array.length; i++) {
            if (array[i].children && array[i].children.length) {
                this.writeSelected(array[i].children, index + 1);
            }
        }
    }
    /**
     * Write a new value to the element.
     *
     * @Override (From ControlValueAccessor interface)
     */
    writeValue(value) {
        const array = [];
        toArray(value).forEach((v, index) => {
            if (typeof v !== 'object') {
                const obj = {};
                obj[this.cmValueProperty] = v;
                obj[this.cmLabelProperty] = v;
                array[index] = obj;
            }
            else {
                array[index] = v;
            }
        });
        this.writeSelected(array, 0);
        this._activatedOptions = this._selectedOptions;
        this.writeColumns(array, 0);
        this.filterColumnData();
        this._buildDisplayLabel();
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        if (isDisabled) {
            this.closeMenu();
            this._addHostClass(`${this._prefixCls}-picker-disabled`);
        }
        else {
            this._removeHostClass(`${this._prefixCls}-picker-disabled`);
        }
        this.cmDisabled = isDisabled;
    }
    ngOnInit() {
        // 设置第一列
        if (this.cmOptions && this.cmOptions.length) {
            this._cmColumns.push(cloneDeep(this.cmOptions));
        }
        this._cmColumns.forEach((item, index) => {
            this._filterColumns[index] = item;
        });
    }
    ngOnDestroy() {
        if (this._delayTimer) {
            clearTimeout(this._delayTimer);
            this._delayTimer = null;
        }
    }
    ngOnChanges(changes) {
        const cmDisabled = changes.cmDisabled;
        if (cmDisabled) {
            if (cmDisabled.currentValue) {
                this._addHostClass(`${this._prefixCls}-picker-disabled`);
            }
            else {
                this._removeHostClass(`${this._prefixCls}-picker-disabled`);
            }
        }
        const cmOptions = changes.cmOptions;
        if (cmOptions && !cmOptions.isFirstChange()) {
            this._cmColumns.splice(0);
            const newOptions = cmOptions.currentValue;
            if (newOptions && newOptions.length) {
                this._cmColumns.push(newOptions);
                this.clearSelection(null);
            }
        }
        this.filterColumnData();
    }
    ngAfterViewInit() {
        this._addHostClass(this._prefixCls);
        this._addHostClass(`${this._prefixCls}-picker`);
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmDisabled", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], CheckCascaderComponent.prototype, "cmSize", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], CheckCascaderComponent.prototype, "cmPlaceHolder", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmShowInput", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmShowSearch", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmAllowClear", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmClearText", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmShowArrow", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmNotFoundContent", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], CheckCascaderComponent.prototype, "cmPopupClassName", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], CheckCascaderComponent.prototype, "cmColumnClassName", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Array)
], CheckCascaderComponent.prototype, "cmOptions", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmEnableCache", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], CheckCascaderComponent.prototype, "cmExpandTrigger", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Boolean])
], CheckCascaderComponent.prototype, "cmChangeOnSelect", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Function)
], CheckCascaderComponent.prototype, "cmChangeOn", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmMouseEnterDelay", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmMouseLeaveDelay", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmTriggerAction", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Function)
], CheckCascaderComponent.prototype, "cmDisplayRender", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Function)
], CheckCascaderComponent.prototype, "cmIntervalRender", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmValueProperty", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmLabelProperty", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Function)
], CheckCascaderComponent.prototype, "cmLoadData", void 0);
__decorate([
    core_1.HostBinding('attr.tabIndex'),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "tabIndex", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmVisibleChange", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmChange", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmSelectionChange", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmSelect", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmUnSelect", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CheckCascaderComponent.prototype, "cmClear", void 0);
__decorate([
    core_1.ViewChild('menu'),
    __metadata("design:type", core_1.ElementRef)
], CheckCascaderComponent.prototype, "menu", void 0);
__decorate([
    core_1.HostListener('keydown', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [KeyboardEvent]),
    __metadata("design:returntype", void 0)
], CheckCascaderComponent.prototype, "onKeyDown", null);
__decorate([
    core_1.HostListener('click', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [MouseEvent]),
    __metadata("design:returntype", void 0)
], CheckCascaderComponent.prototype, "_onTriggerClick", null);
__decorate([
    core_1.HostListener('mouseenter', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [MouseEvent]),
    __metadata("design:returntype", void 0)
], CheckCascaderComponent.prototype, "_onTriggerMouseEnter", null);
__decorate([
    core_1.HostListener('mouseleave', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [MouseEvent]),
    __metadata("design:returntype", void 0)
], CheckCascaderComponent.prototype, "_onTriggerMouseLeave", null);
CheckCascaderComponent = CheckCascaderComponent_1 = __decorate([
    core_1.Component({
        encapsulation: core_1.ViewEncapsulation.None,
        selector: 'cm-checkcascader',
        template: `
      <div #trigger cdkOverlayOrigin #origin="cdkOverlayOrigin">
          <div *ngIf="cmShowInput">
              <input #input
                     nz-input
                     [attr.autoComplete]="'off'"
                     [attr.placeholder]="hasInput() || hasSelection() ? null : cmPlaceHolder"
                     [readonly]="true"
                     [disabled]="cmDisabled"
                     [nzSize]="cmSize"
                     [ngClass]="_inputCls"
                     [(ngModel)]="_inputValue"
                     (blur)="_handleInputBlur($event)"
                     (focus)="_handleInputFocus($event)"
                     (keydown)="_handleInputKeyDown($event)"
                     (change)="_handlerInputChange($event)"
              >
              <i *ngIf="showClearIcon"
                 [class]="'anticon anticon-cross-circle'"
                 [ngClass]="_clearCls"
                 [attr.title]="cmClearText"
                 (click)="clearSelection($event)"></i>
              <i *ngIf="cmShowArrow" class="anticon anticon-down" [ngClass]="_arrowCls"></i>
              <span [ngClass]="_pickerLabelCls">
                  <ng-container *ngIf="_displayLabelIsTemplate">
                      <ng-container *ngTemplateOutlet="_displayLabel; context: _displayLabelContext"></ng-container>
                  </ng-container>
                  <ng-container *ngIf="!_displayLabelIsTemplate">{{ _displayLabel }}</ng-container>
              </span>
          </div>
          <ng-content></ng-content>

      </div>


      <ng-template
              cdkConnectedOverlay
              cdkConnectedOverlayHasBackdrop
              [cdkConnectedOverlayOrigin]="origin"
              (backdropClick)="closeMenu()"
              (detach)="closeMenu()"
              (positionChange)="onPositionChange($event)"
              [cdkConnectedOverlayOpen]="menuVisible"
      >
          <div #menu [ngClass]="_menuCls">
              <ul *ngFor="let options of _filterColumns; let i = index;" [ngClass]="_columnCls" [attr.data-key]="i">
                  <nz-input *ngIf="cmShowSearch" [nzType]="'search'" [nzPlaceHolder]="'input search text'" [(ngModel)]="_searchValue[i]" (nzOnSearch)="onSearch($event, i)" style="width: 95%;margin: 2px 5px;"></nz-input>
                  <li *ngFor="let option of options"
                      [attr.data-key]="getOptionValue(option)"
                      [attr.title]="option.title || getOptionLabel(option)"
                      [ngClass]="getOptionCls(option, i)"
                      (mouseenter)="onOptionMouseEnter(option, i, $event)"
                      (mouseleave)="onOptionMouseLeave(option, i, $event)"
                      (click)="onOptionClick(option, i, $event)">
                      <cm-checkbox>
                          <label nz-checkbox [ngModel]="option.checked" (ngModelChange)="checkChange($event,option)"></label>
                      </cm-checkbox>
                      <span>{{ getOptionLabel(option) }}</span>
                  </li>
              </ul>
          </div>
      </ng-template>
    `,
        styles: [`
      /* button*/
      .cm-light-btn-default {
        color: #495060;
        background-color: #fff;
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-btn-default:hover {
        color: #2a6cF0;
        border-color: #2a6cF0;
      }
      .cm-light-btn-default:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-info {
        color: #fff;
        background-color: #2db7f5;
        border: 1px solid #2db7f5;
        border-radius: 4px;
      }
      .cm-light-btn-info:hover {
        background-color: #57CFFF;
        color: #fff;
        border-color: #2db7f5;
      }
      .cm-light-btn-info:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-primary {
        color: #fff;
        background-color: #2d8cf0;
        border: 1px solid #2d8cf0;
        border-radius: 4px;
      }
      .cm-light-btn-primary:hover {
        background-color: #2a6cF0;
        color: #fff;
        border-color: #2d8cf0;
      }
      .cm-light-btn-primary:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-success {
        color: #fff;
        background-color: #19be6b;
        border: 1px solid #19be6b;
        border-radius: 4px;
      }
      .cm-light-btn-success:hover {
        color: #fff;
        background-color: #47cb89;
        border-color: #19be6b;
      }
      .cm-light-btn-success:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-warning {
        color: #fff;
        background-color: #f90;
        border: 1px solid #f90;
        border-radius: 4px;
      }
      .cm-light-btn-warning:hover {
        color: #fff;
        background-color: #f59e2d;
        border-color: #f90;
      }
      .cm-light-btn-warning:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-failed {
        color: #fff;
        background-color: #ed3f14;
        border: 1px solid #ed3f14;
        border-radius: 4px;
      }
      .cm-light-btn-failed:hover {
        color: #fff;
        background-color: #f16543;
        border-color: #ed3f14;
      }
      .cm-light-btn-failed:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-text {
        color: #fff;
        background-color: #fff;
        border: 1px solid #fff;
        border-radius: 4px;
      }
      .cm-light-btn-text:hover {
        color: #2DacF0;
        border-color: #fff;
      }
      .cm-light-btn-text:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-ghost {
        background: transparent!important;
        border-color: #fff;
        color: #fff;
      }
      .cm-light-btn-ghost:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-ghost-primary {
        background: transparent;
        border-color: #2d8cf0;
        color: #2d8cf0;
      }
      .cm-light-btn-ghost-primary:hover {
        border-color: #2DacF0;
        color: #2DacF0;
      }
      .cm-light-btn-ghost-danger {
        background: transparent;
        border-color: #ed3f14;
        color: #ed3f14;
      }
      .cm-light-btn-ghost-danger:hover {
        border-color: #f16543;
        color: #f16543;
      }
      .cm-light-btn-dashed {
        color: #495060;
        background-color: #fff;
        border: 1px dashed #dddee1;
        border-radius: 4px;
      }
      .cm-light-btn-dashed:hover {
        color: #2a6cF0;
        border-color: #2a6cF0;
      }
      .cm-light-btn-dashed:disabled:hover {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-btn-radio {
        color: #495060;
        background-color: #fff;
        border-color: #dddee1;
      }
      .cm-light-btn-radio:first-child {
        border-radius: 4px 0 0 4px;
        border: 1px solid #dddee1;
      }
      .cm-light-btn-radio:hover {
        color: #2DacF0;
      }
      .cm-light-btn-radio-checked {
        background: #fff;
        border-color: #2d8cf0;
        color: #2d8cf0;
        box-shadow: -1px 0 0 0 #2d8cf0;
      }
      .cm-light-btn-radio-checked:first-child {
        box-shadow: none!important;
        border-color: #2d8cf0;
      }
      .cm-light-btn-radio-checked:hover {
        border-color: #2DacF0;
        box-shadow: -1px 0 0 0 #2DacF0;
        color: #2DacF0;
      }
      .cm-light-btn-radio-checked:before {
        border-color: #2DacF0;
      }
      .cm-light-btn-disabled {
        color: #bbbec4;
        border-color: #bbbec4;
        background-color: #f4f4f4;
        border-radius: 4px;
        cursor: not-allowed;
      }
      /*Dropdown*/
      .cm-light-dropdown-default {
        color: #2d8cf0;
        background-color: #fff;
        border: none;
      }
      .cm-light-dropdown-default:hover {
        color: #2a6cF0;
      }
      .cm-light-dropdown-btn {
        color: #fff;
        background-color: #2d8cf0;
        border: transparent;
        border-radius: 4px;
      }
      .cm-light-dropdown-btn:hover {
        background-color: #2a6cF0;
      }
      .cm-light-dropdown-menu {
        color: #495060;
        background-color: #fff;
        border: 1px solid #dddee1;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
      }
      .cm-light-dropdown-menu-item {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-dropdown-menu-item:hover {
        background-color: #dddee1;
      }
      .cm-light-dropdown-disbabled-item {
        color: #bbbec4;
        background-color: #f8f8f9;
      }
      .cm-light-dropdown-menu-current {
        background-color: #2DacF0;
      }
      /*Menu*/
      .cm-light-menu-color {
        color: #495060;
      }
      .cm-light-menu-color:hover {
        color: #2a6cF0;
      }
      .cm-light-menu-bg {
        background-color: #fff;
      }
      .cm-light-menu-border {
        border-color: #dddee1;
      }
      .cm-light-menu-border-hover {
        background-color: transparent;
      }
      .cm-light-menu-border-hover:hover {
        border-color: #2a6cF0;
      }
      .cm-light-menu-active {
        color: #2a6cF0;
        border-color: #2a6cF0;
        background-color: #ecf6fd;
      }
      .cm-light-menu-disabled {
        color: #bbbec4;
      }
      /*Pagination*/
      .cm-light-pagination-current-page {
        color: #fff;
        border-color: #2d8cf0;
        background-color: #2d8cf0;
      }
      .cm-light-pagination-current-page:hover a,
      .cm-light-pagination-current-page:focus a {
        color: #fff !important;
      }
      .cm-light-pagination-page {
        color: #495060;
        background-color: #fff;
        border-color: #dddee1;
        border-radius: 4px;
      }
      .cm-light-pagination-page:hover {
        border-color: #2d8cf0;
        color: #2d8cf0;
      }
      .cm-light-pagination-page:hover a {
        color: #2d8cf0;
      }
      .cm-light-pagination-border:hover {
        border-color: #2d8cf0;
      }
      .cm-light-pagination-text {
        color: #495060;
      }
      .cm-light-pagination-disabled {
        color: #bbbec4;
        border-color: #bbbec4;
      }
      /*Checkbox*/
      .cm-light-checkbox-checked {
        color: #2a6cF0;
      }
      .cm-light-checkbox-unchecked {
        color: #fff;
      }
      .cm-light-checkbox-label {
        color: #495060;
      }
      .cm-light-checkbox-disabled {
        color: #bbbec4;
      }
      .cm-light-checkbox-base {
        border-color: #2d8cf0;
        background-color: #2d8cf0;
      }
      /*DatePicker*/
      .cm-light-datepicker-default {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-datepicker-default:hover {
        border-color: #2a6cF0;
      }
      .cm-light-datepicker-date {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-datepicker-active-font {
        color: #495060;
      }
      .cm-light-datepicker-active-font:hover {
        background-color: #f8f8f9;
      }
      .cm-light-datepicker-unactive-font {
        color: #bbbec4;
      }
      .cm-light-datepicker-current-font {
        background-color: #2d8cf0;
      }
      .cm-light-datepicker-disabled-date {
        color: #bbbec4;
      }
      /*Radio*/
      .cm-light-radio-label {
        color: #495060;
      }
      .cm-light-radio-default {
        border-color: #dddee1;
        background: #fff;
      }
      .cm-light-radio-default:hover {
        border-color: #2d8cf0;
      }
      .cm-light-radio-bg {
        background: #fff;
      }
      .cm-light-radio-inner {
        background: #2d8cf0;
      }
      .cm-light-radio-border-checked {
        border-color: #2d8cf0;
      }
      .cm-light-radio-border-shadow {
        box-shadow: -1px 0 0 0 #2d8cf0;
      }
      .cm-light-radio-disabled {
        color: #bbbec4;
        border-color: #bbbec4 !important;
        background-color: #f4f4f4;
        cursor: not-allowed;
      }
      /*Select*/
      .cm-light-select-default {
        color: #495060;
        border: 1px solid #dddee1;
        border-radius: 4px;
        background-color: #fff;
      }
      .cm-light-select-default:hover {
        border-color: #2a6cF0;
      }
      .cm-light-select-disabled {
        color: #bbbec4;
        border: 1px solid #dddee1;
        border-radius: 4px;
        background-color: #f4f4f4;
      }
      .cm-light-select-menu {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-select-menu-item {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-select-menu-item:hover {
        background-color: #dddee1;
      }
      .cm-light-select-menu-current {
        color: #fff;
        background-color: #2a6cF0;
      }
      .cm-light-select-item-disabled {
        color: #495060;
      }
      /*Slider*/
      .cm-light-slider-color {
        color: #2DacF0;
        background-color: #2DacF0;
      }
      .cm-light-slider-border-color {
        border-color: #2DacF0;
      }
      .cm-light-slider-range {
        color: #dddee1;
        background-color: #dddee1;
      }
      .cm-light-slider-disabled {
        color: #bbbec4;
        background-color: #bbbec4;
      }
      .cm-light-slider-step {
        color: #666;
        background-color: #2DacF0;
      }
      .cm-light-slider-step:hover {
        color: #666;
        background-color: #57a3f3;
      }
      .cm-light-slider-input {
        color: #495060;
        background-color: #fff;
        border: 2px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-slider-border {
        border: 1px solid #dddee1;
        border-radius: 50%;
      }
      .cm-light-slider-border:hover {
        border-color: #2DacF0;
      }
      /*Switch*/
      .cm-light-switch-default {
        background-color: #666;
      }
      .cm-light-switch-default:focus {
        box-shadow: 0 0 0 1px rgba(16, 142, 233, 0.2);
      }
      .cm-light-switch-default-bg {
        background-color: #666;
      }
      .cm-light-switch-default-color {
        color: #fff;
      }
      .cm-light-switch-open {
        background-color: #2d8cf0;
      }
      .cm-light-switch-disabled {
        background-color: #f4f4f4;
      }
      /*Popover*/
      .cm-light-popover-default {
        background-color: #fff;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-light-popover-title {
        color: #000;
        border-bottom: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-popovert-body {
        color: #495060;
      }
      /*Tooltip*/
      .cm-light-tooltip-default {
        color: #fff;
        background-color: #1c2438;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      /*Table*/
      .cm-light-table-thead {
        color: #000;
        background-color: #f8f8f9;
      }
      .cm-light-table-tbody {
        color: #495060;
        background-color: #fff;
      }
      .cm-light-table-tbody:hover {
        background-color: #2a6cF0;
      }
      .cm-light-table-border {
        border-color: #dddee1;
      }
      .cm-light-table-cell {
        padding: 0;
        padding-left: 18px;
        padding-right: 18px;
        height: 40px;
      }
      .cm-light-table-tbody-td-hover {
        background-color: #ebf7ff;
      }
      /*Tabs*/
      .cm-light-tabs-default {
        color: #495060;
        background-color: #fff;
        border-bottom: 1px solid #dddee1;
      }
      .cm-light-tabs-default:hover {
        background-color: #2a6cF0;
      }
      .cm-light-tabs-current {
        color: #2a6cF0;
        border-bottom: 1px solid #2a6cF0;
      }
      .cm-light-tabs-disabled {
        color: #bbbec4;
        border-bottom: 1px solid #bbbec4;
      }
      .cm-light-tabs-hover-color {
        color: #2d8cf0;
      }
      .cm-light-tabs-hover-background {
        background-color: #2d8cf0;
      }
      /*Alert*/
      .cm-light-alert-info {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #2db7f5;
      }
      .cm-light-alert-info-color {
        color: #495060;
      }
      .cm-light-alert-info-main-color {
        color: #1c2438;
      }
      .cm-light-alert-success {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #19be6b;
      }
      .cm-light-alert-success-color {
        color: #495060;
      }
      .cm-light-alert-success-main-color {
        color: #1c2438;
      }
      .cm-light-alert-warning {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #f90;
      }
      .cm-light-alert-warning-color {
        color: #495060;
      }
      .cm-light-alert-warning-main-color {
        color: #1c2438;
      }
      .cm-light-alert-error {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #ed3f14;
      }
      .cm-light-alert-error-color {
        color: #495060;
      }
      .cm-light-alert-main-error-color {
        color: #1c2438;
      }
      /*Message*/
      .cm-light-message-default {
        color: #495060;
        background-color: #f8f8f9;
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
      }
      /*Modal*/
      .cm-light-modal-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-light-modal-title {
        color: #495060;
      }
      .cm-light-modal-body {
        color: #495060;
      }
      .cm-light-modal-footer {
        color: #495060;
      }
      .cm-light-modal-border {
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      /*Notification*/
      .cm-light-norification-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-light-norification-title {
        color: #000;
      }
      .cm-light-norification-body {
        color: #495060;
      }
      .cm-light-norification-border {
        border: 1px solid #dddee1;
        border-radius: 4px;
      }
      .cm-light-norification-info {
        color: #2db7f5;
      }
      .cm-light-norification-success {
        color: #19be6b;
      }
      .cm-light-norification-warning {
        color: #f90;
      }
      .cm-light-norification-error {
        color: #ed3f14;
      }
      /*Popconfirm*/
      .cm-light-popconfirm-default {
        border: 1px solid #dddee1;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
        color: #495060;
      }
      /*input*/
      .cm-light-input {
        background-color: #fff;
        border-radius: 4px;
        border-style: solid;
        border-width: 1px;
        border-color: #dddee1;
        color: #495060;
      }
      .cm-light-input:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-input:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-disabled {
        color: #bbbec4;
        background-color: #f4f4f4;
        border: 1px solid #dddee1;
      }
      .cm-light-input-disabled:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-disabled:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-input-group-border-last {
        border-left-width: 0;
        border-right-width: 1px;
      }
      .cm-light-input-group-border-last:hover {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-light-input-group-border-last:focus {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-light-input-group-border {
        border-right-width: 0;
      }
      .cm-light-input-group-border:hover {
        border-right-width: 1px;
      }
      .cm-input-disabled {
        color: #bbbec4;
      }
      /*progress*/
      .cm-light-progress-bg {
        background-color: #fff;
      }
      .cm-light-progress-bd {
        border-color: #f8f8f9;
      }
      .cm-light-progress-info-bg {
        background-color: #2db7f5;
      }
      .cm-light-progress-info-color {
        color: #495060;
      }
      .cm-light-progress-failed-bg {
        background-color: #ed3f14;
      }
      .cm-light-progress-failed-color {
        color: #ed3f14;
      }
      .cm-light-progress-success-bg {
        background-color: #19be6b;
      }
      .cm-light-progress-success-color {
        color: #19be6b;
      }
      /*inputNumber*/
      .cm-light-inputNumber {
        border-color: #dddee1;
        color: #495060;
      }
      .cm-light-inputNumber:hover {
        border: 1px solid #57a3f3;
      }
      .cm-light-inputNumber:focus {
        border: 1px solid #57a3f3;
      }
      .cm-light-inputNumber-inner-color {
        color: #495060;
      }
      .cm-light-inputNumber-inner-border {
        border-color: #57a3f3;
      }
      .cm-light-inputNumber-inner-disabled {
        color: #bbbec4;
        background-color: #f4f4f4;
      }
      .cm-light-inputNumber-inner-disabled:hover {
        color: #bbbec4;
      }
      /*弹框下箭头：白色背景*/
      .cm-light-popper-arrow-down {
        border-color: transparent;
        border-width: 6px;
        border-top-color: #dddee1;
        border-bottom-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-light-popper-arrow-down:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        bottom: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-top-color: #fff;
        border-bottom-width: 0;
      }
      /*弹框上箭头：白色背景*/
      .cm-light-popper-arrow-up {
        border-color: transparent;
        border-width: 6px;
        border-bottom-color: #dddee1;
        border-top-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-light-popper-arrow-up:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        top: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-bottom-color: #fff;
        border-top-width: 0;
      }
      /*提示框下三角：黑色背景*/
      .cm-light-tooltip-arrow-down {
        border-width: 6px;
        border-style: solid;
        border-top-color: #000;
        border-bottom-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框上三角：黑色背景*/
      .cm-light-tooltip-arrow-up {
        border-width: 6px;
        border-style: solid;
        border-bottom-color: #000;
        border-top-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框左三角：黑色背景*/
      .cm-light-tooltip-arrow-left {
        border-width: 6px;
        border-style: solid;
        border-right-color: #000;
        border-left-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*提示框右三角：黑色背景*/
      .cm-light-tooltip-arrow-right {
        border-width: 6px;
        border-style: solid;
        border-left-color: #000;
        border-right-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*timeline*/
      /*timeline 圆圈颜色*/
      .cm-light-timeline-circle-blue {
        color: #2d8cf0;
        border-color: #2d8cf0;
      }
      .cm-light-timeline-circle-green {
        color: #19be6b;
        border-color: #19be6b;
      }
      .cm-light-timeline-circle-red {
        color: #ed3f14;
        border-color: #ed3f14;
      }
      /*timeline 字体颜色*/
      .cm-light-timeline-text {
        color: #2d8cf0;
      }
      /*timeline 图标颜色*/
      .cm-light-timeline-icon {
        color: #ed3f14;
      }
      /*steps*/
      .cm-light-step {
        border: 1px solid #dddee1;
        border-radius: 50%;
      }
      .cm-light-step-finish {
        border-color: #57a3f3;
        background-color: #fff;
      }
      .cm-light-step-wait {
        border-color: #dddee1;
        background-color: #fff;
      }
      .cm-light-step-progress {
        border-color: #dddee1;
        background-color: #57a3f3;
      }
      .cm-light-step-line-error {
        border-color: #ed3f14;
        background-color: #fff;
      }
      .cm-light-step-line-wait {
        box-shadow: 0 0 0 0 #dddee1;
        background: #dddee1;
      }
      .cm-light-step-line-progress {
        box-shadow: 0 0 0 0 #57a3f3;
        background: #57a3f3;
      }
      .cm-light-step-color1 {
        color: #dddee1;
      }
      .cm-light-step-color2 {
        color: #fff;
      }
      .cm-light-step-color3 {
        color: #57a3f3;
      }
      .cm-light-step-color4 {
        color: #ed3f14;
      }
      .cm-light-step-color5 {
        color: #495060;
      }
      /*cascader*/
      .cm-light-a-color {
        color: #2d8cf0;
      }
      .cm-light-a-color:hover {
        color: #2DacF0;
      }
      /*breadcrumb*/
      .cm-light-text-color {
        color: #464c5b;
      }
      .cm-light-text-color:hover {
        color: #2DacF0;
      }
      .cm-light-start-color {
        color: #657180;
      }
      .cm-light-interval-color {
        color: #dddee1;
      }
      .cm-light-lasttext-color {
        color: #1c2438;
      }
      /*upload*/
      .cm-light-texterror-color {
        color: #ed3f14;
      }
      .cm-light-border-color {
        border-color: #dddee1;
      }
      .cm-light-text-color {
        color: #495060;
      }
      .cm-light-color {
        color: #2a6cF0;
      }
      .cm-light-hover-color {
        border-color: #2a6cF0;
      }
      .cm-light-bg-color {
        background-color: #f4f4f4;
      }
      /*字体图标颜色*/
      .cm-icon-nav-1:before {
        color: #495060;
      }
      .cm-icon-nav-2:before {
        color: #495060;
      }
      .cm-icon-nav-3:before {
        color: #495060;
      }
      .cm-icon-arrow-down:before {
        color: #dddee1;
      }
      .cm-icon-basic-components:before {
        color: #dddee1;
      }
      .cm-icon-professional-components:before {
        color: #495060;
      }
      .cm-icon-checkbox-focus:before {
        color: #2DacF0;
      }
      .cm-icon-functional-components:before {
        color: #495060;
      }
      .cm-icon-document:before {
        color: #495060;
      }
      .cm-icon-checkbox-default:before {
        color: #495060;
      }
      .cm-icon-info:before {
        color: #2db7f5;
      }
      .cm-icon-arrow-up:before {
        color: #dddee1;
      }
      .cm-icon-close:before {
        color: #dddee1;
      }
      .cm-icon-clock:before {
        color: #dddee1;
      }
      .cm-icon-arrow-left:before {
        color: #dddee1;
      }
      .cm-icon-radio-disabled:before {
        color: #bbbec4;
      }
      .cm-icon-arrow-right:before {
        color: #dddee1;
      }
      .cm-icon-not-all-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-radio-default:before {
        color: #dddee1;
      }
      .cm-icon-menu-fold:before {
        color: #dddee1;
      }
      .cm-icon-menu-unfold:before {
        color: #dddee1;
      }
      .cm-icon-radio-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-error:before {
        color: #ed3f14;
      }
      .cm-icon-disabled-checked:before {
        color: #bbbec4;
      }
      .cm-icon-warning:before {
        color: #f90;
      }
      .cm-icon-checkbox-checked:before {
        color: #2d8cf0;
      }
      .cm-icon-success:before {
        color: #19be6b;
      }
      a {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
      }
      .large {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 14px;
        padding: 12px 20px;
        height: auto;
      }
      .medium {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 14px;
        padding: 10px 20px;
        height: auto;
      }
      .small {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 12px;
        padding: 9px 15px;
        height: auto;
      }
      .mini {
        font-family: "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif;
        font-size: 12px;
        padding: 7px 15px;
        height: auto;
      }
      .theme-light .cm-cascader-selection {
        background-color: #fff;
      }
      /* button*/
      .cm-dark-btn-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-btn-default:hover {
        color: #080F26;
        border-color: #080F26;
      }
      .cm-dark-btn-default:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-info {
        color: #fff;
        background-color: #0D6DA8;
        border: 1px solid #0D6DA8;
        border-radius: 4px;
      }
      .cm-dark-btn-info:hover {
        color: #fff;
        background-color: #1B90CF;
      }
      .cm-dark-btn-info:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-primary {
        color: #fff;
        background-color: #14224D;
        border: 1px solid #14224D;
        border-radius: 4px;
      }
      .cm-dark-btn-primary:hover {
        color: #fff;
        background-color: #253459;
        border-color: #14224D;
      }
      .cm-dark-btn-primary:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-success {
        color: #fff;
        background-color: #037343;
        border: 1px solid #037343;
        border-radius: 4px;
      }
      .cm-dark-btn-success:hover {
        color: #fff;
        background-color: #0C9957;
        border-color: #037343;
      }
      .cm-dark-btn-success:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-warning {
        color: #fff;
        background-color: #B35F00;
        border: 1px solid #B35F00;
        border-radius: 4px;
      }
      .cm-dark-btn-warning:hover {
        color: #fff;
        background-color: #D97B00;
        border-color: #B35F00;
      }
      .cm-dark-btn-warning:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-failed {
        color: #fff;
        background-color: #A11500;
        border: 1px solid #A11500;
        border-radius: 4px;
      }
      .cm-dark-btn-failed:hover {
        color: #fff;
        background-color: #C72606;
        border-color: #A11500;
      }
      .cm-dark-btn-failed:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-text {
        color: #fff;
        background-color: #fff;
        border: 1px solid #fff;
        border-radius: 4px;
      }
      .cm-dark-btn-text:hover {
        color: #253459;
        border-color: #fff;
      }
      .cm-dark-btn-text:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-ghost {
        background: transparent!important;
        border-color: #fff;
        color: #fff;
      }
      .cm-dark-btn-ghost:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-ghost-primary {
        background: transparent;
        border-color: #14224D;
        color: #14224D;
      }
      .cm-dark-btn-ghost-primary:hover {
        border-color: #253459;
        color: #253459;
      }
      .cm-dark-btn-ghost-danger {
        background: transparent;
        border-color: #A11500;
        color: #A11500;
      }
      .cm-dark-btn-ghost-danger:hover {
        border-color: #C72606;
        color: #C72606;
      }
      .cm-dark-btn-dashed {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px dashed #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-btn-dashed:hover {
        color: #1B90CF;
        border-color: #080F26;
      }
      .cm-dark-btn-dashed:disabled:hover {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-btn-radio {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border-color: #9CA0A6;
      }
      .cm-dark-btn-radio:first-child {
        border-radius: 4px 0 0 4px;
        border: 1px solid #9CA0A6;
      }
      .cm-dark-btn-radio:hover {
        color: #253459;
      }
      .cm-dark-btn-radio-checked {
        background: #fff;
        border-color: #14224D;
        color: #14224D;
        box-shadow: -1px 0 0 0 #14224D;
      }
      .cm-dark-btn-radio-checked:first-child {
        box-shadow: none!important;
        border-color: #14224D;
      }
      .cm-dark-btn-radio-checked:hover {
        border-color: #253459;
        box-shadow: -1px 0 0 0 #253459;
        color: #253459;
      }
      .cm-dark-btn-disabled {
        color: #84878C;
        border-color: #84878C;
        background-color: #d8cbcb;
        border-radius: 4px;
        cursor: not-allowed;
      }
      /*Dropdown*/
      .cm-dark-dropdown-default {
        color: #14224D;
        background-color: #fff;
        border: none;
      }
      .cm-dark-dropdown-default:hover {
        color: #253459;
      }
      .cm-dark-dropdown-btn {
        color: #fff;
        background-color: #14224D;
        border: transparent;
        border-radius: 4px;
      }
      .cm-dark-dropdown-btn:hover {
        background-color: #253459;
      }
      .cm-dark-dropdown-menu {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        border-radius: 4px;
      }
      .cm-dark-dropdown-menu-item {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-dropdown-menu-item:hover {
        background-color: #9CA0A6;
      }
      .cm-dark-dropdown-disbabled-item {
        color: #84878C;
        background-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-dropdown-menu-current {
        background-color: #253459;
      }
      /*Menu*/
      .cm-dark-menu-color {
        color: #a4a5a7;
      }
      .cm-dark-menu-color:hover {
        color: #d3d5d8;
      }
      .cm-dark-menu-bg {
        background-color: #14224D;
      }
      .cm-dark-submenu-bg {
        background-color: #253459;
      }
      .cm-dark-menu-border {
        border-color: #9CA0A6;
      }
      .cm-dark-menu-border-hover {
        background-color: #0D6DA8;
      }
      .cm-dark-menu-border-hover:hover {
        border-color: #080F26;
      }
      .cm-dark-menu-active {
        color: #d3d5d8;
        border-color: #080F26;
        background-color: #0D6DA8;
      }
      .cm-dark-menu-disabled {
        color: #84878C;
      }
      /*Pagination*/
      .cm-dark-pagination-current-page {
        color: #fff;
        border-color: #0D6DA8;
        background-color: #0D6DA8;
      }
      .cm-dark-pagination-current-page:hover a,
      .cm-dark-pagination-current-page:focus a {
        color: #fff !important;
      }
      .cm-dark-pagination-page {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-pagination-page:hover {
        border-color: #0D6DA8;
        color: #0D6DA8;
      }
      .cm-dark-pagination-page:hover a {
        color: #0D6DA8;
      }
      .cm-dark-pagination-border:hover {
        border-color: #0D6DA8;
      }
      .cm-dark-pagination-text {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-pagination-disabled {
        color: #84878C;
        border-color: #84878C;
      }
      /*Checkbox*/
      .cm-dark-checkbox-checked {
        color: #080F26;
      }
      .cm-dark-checkbox-unchecked {
        color: #fff;
      }
      .cm-dark-checkbox-label {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-checkbox-text {
        color: #14224D;
      }
      .cm-dark-checkbox-disabled {
        color: #84878C;
      }
      .cm-dark-checkbox-move {
        border-color: #14224D;
      }
      .cm-dark-checkbox-base {
        border-color: #0D6DA8;
        background-color: #0D6DA8;
      }
      /*DatePicker*/
      .cm-dark-datepicker-default {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-datepicker-default:hover {
        border-color: #080F26;
      }
      .cm-dark-datepicker-date {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-datepicker-active-font {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-datepicker-active-font:hover {
        background-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-datepicker-unactive-font {
        color: #84878C;
      }
      .cm-dark-datepicker-current-font {
        background-color: #14224D;
      }
      .cm-dark-datepicker-disabled-date {
        color: #84878C;
      }
      /*Radio*/
      .cm-dark-radio-label {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-radio-default {
        border-color: #9CA0A6;
        background: #fff;
      }
      .cm-dark-radio-default:hover {
        border-color: #14224D;
      }
      .cm-dark-radio-bg {
        background: #fff;
      }
      .cm-dark-radio-inner {
        background: #14224D;
      }
      .cm-dark-radio-border-checked {
        border-color: #14224D;
      }
      .cm-dark-radio-border-shadow {
        box-shadow: -1px 0 0 0 #14224D;
      }
      .cm-dark-radio-disabled {
        color: #84878C;
        border-color: #84878C !important;
        background-color: #d8cbcb;
        cursor: not-allowed;
      }
      /*Select*/
      .cm-dark-select-default {
        color: rgba(0, 0, 0, 0.56);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        background-color: #fff;
      }
      .cm-dark-select-default:hover {
        border-color: #080F26;
      }
      .cm-dark-select-disabled {
        color: #84878C;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        background-color: #d8cbcb;
      }
      .cm-dark-select-menu {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-select-menu-item {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-select-menu-item:hover {
        background-color: #9CA0A6;
      }
      .cm-dark-select-menu-current {
        color: #fff;
        background-color: #080F26;
      }
      .cm-dark-select-item-disabled {
        color: rgba(0, 0, 0, 0.56);
      }
      /*Slider*/
      .cm-dark-slider-color {
        color: #253459;
        background-color: #253459;
      }
      .cm-dark-slider-border-color {
        border-color: #253459;
      }
      .cm-dark-slider-range {
        color: #9CA0A6;
        background-color: #9CA0A6;
      }
      .cm-dark-slider-disabled {
        color: #84878C;
        background-color: #84878C;
      }
      .cm-dark-slider-step {
        color: rgba(0, 0, 0, 0.4);
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-slider-step:hover {
        color: rgba(0, 0, 0, 0.56);
        background-color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-slider-input {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border: 2px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-slider-border {
        border: 1px solid #9CA0A6;
        border-radius: 50%;
      }
      .cm-dark-slider-border:hover {
        border-color: #253459;
      }
      /*Switch*/
      .cm-dark-switch-default {
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-switch-default:focus {
        box-shadow: 0 0 0 1px rgba(28, 28, 29, 0.2);
      }
      .cm-dark-switch-default-bg {
        background-color: rgba(0, 0, 0, 0.4);
      }
      .cm-dark-switch-default-color {
        color: #fff;
      }
      .cm-dark-switch-open {
        background-color: #14224D;
      }
      .cm-dark-switch-disabled {
        background-color: #84878C;
      }
      /*Popover*/
      .cm-dark-popover-default {
        background-color: #fff;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      .cm-dark-popover-title {
        color: #000;
        border-bottom: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-popovert-body {
        color: rgba(0, 0, 0, 0.56);
      }
      /*Tooltip*/
      .cm-dark-tooltip-default {
        color: #fff;
        background-color: #1c2438;
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
      }
      /*Table*/
      .cm-dark-table-thead {
        color: #000;
        background-color: #acacbb;
      }
      .cm-dark-table-tbody {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
      }
      .cm-dark-table-tbody:hover {
        background-color: #253459;
      }
      .cm-dark-table-border {
        border-color: #9CA0A6;
      }
      .cm-dark-table-cell {
        padding: 0;
        padding-left: 18px;
        padding-right: 18px;
        height: 40px;
      }
      .cm-dark-table-tbody-td-hover {
        background-color: #ebf7ff;
      }
      /*Tabs*/
      .cm-dark-tabs-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: #fff;
        border-bottom: 1px solid #9CA0A6;
      }
      .cm-dark-tabs-default:hover {
        background-color: #253459;
      }
      .cm-dark-tabs-current {
        color: #080F26;
        border-bottom: 1px solid #080F26;
      }
      .cm-dark-tabs-disabled {
        color: #84878C;
        border-bottom: 1px solid #84878C;
      }
      .cm-dark-tabs-hover-color {
        color: #0D6DA8;
      }
      .cm-dark-tabs-hover-background {
        background-color: #0D6DA8;
      }
      /*Alert*/
      .cm-dark-alert-info {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #0D6DA8;
      }
      .cm-dark-alert-info-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-info-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-success {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #037343;
      }
      .cm-dark-alert-success-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-success-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-warning {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #B35F00;
      }
      .cm-dark-alert-warning-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-warning-main-color {
        color: #d3d5d8;
      }
      .cm-dark-alert-error {
        color: #a4a5a7;
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #A11500;
      }
      .cm-dark-alert-error-color {
        color: #a4a5a7;
      }
      .cm-dark-alert-error-main-color {
        color: #d3d5d8;
      }
      /*Message*/
      .cm-dark-message-default {
        color: rgba(0, 0, 0, 0.56);
        background-color: rgba(0, 0, 0, 0.8);
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
      }
      /*Modal*/
      .cm-dark-modal-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-dark-modal-title {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-body {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-footer {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-modal-border {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      /*Notification*/
      .cm-dark-norification-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
      }
      .cm-dark-norification-title {
        color: #000;
      }
      .cm-dark-norification-body {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-norification-border {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
      }
      .cm-dark-norification-info {
        color: #0D6DA8;
      }
      .cm-dark-norification-success {
        color: #037343;
      }
      .cm-dark-norification-warning {
        color: #B35F00;
      }
      .cm-dark-norification-error {
        color: #A11500;
      }
      /*Popconfirm*/
      .cm-dark-popconfirm-default {
        border: 1px solid #9CA0A6;
        border-radius: 4px;
        box-shadow: 0 2px 12px rgba(0, 0, 0, 0.2);
        background-color: #fff;
        color: rgba(0, 0, 0, 0.56);
      }
      /*input*/
      .cm-dark-input {
        background-color: #fff;
        border-radius: 4px;
        border-style: solid;
        border-width: 1px;
        border-color: #9CA0A6;
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-input:hover {
        border: 1px solid #253459;
      }
      .cm-dark-input:focus {
        border: 1px solid #253459;
      }
      .cm-dark-input-disabled {
        color: #84878C;
        background-color: #d8cbcb;
        border: 1px solid #9CA0A6;
      }
      .cm-dark-input-disabled:hover {
        border: 1px solid #253459;
      }
      .cm-dark-input-disabled:focus {
        border: 1px solid #253459;
      }
      .cm-dark-input-group-border-last {
        border-left-width: 0;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border-last:hover {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border-last:focus {
        border-left-width: 1px;
        border-right-width: 1px;
      }
      .cm-dark-input-group-border {
        border-right-width: 0;
      }
      .cm-dark-input-group-border:hover {
        border-right-width: 1px;
      }
      .cm-input-disabled {
        color: #84878C;
      }
      /*progress*/
      .cm-dark-progress-bg {
        background-color: #fbfbfb;
      }
      .cm-dark-progress-bd {
        border-color: rgba(0, 0, 0, 0.8);
      }
      .cm-dark-progress-info-bg {
        background-color: #0D6DA8;
      }
      .cm-dark-progress-info-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-progress-failed-bg {
        background-color: #A11500;
      }
      .cm-dark-progress-failed-color {
        color: #A11500;
      }
      .cm-dark-progress-success-bg {
        background-color: #037343;
      }
      .cm-dark-progress-success-color {
        color: #037343;
      }
      /*inputNumber*/
      .cm-dark-inputNumber {
        border-color: #9CA0A6;
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber:hover {
        border: 1px solid #253459;
      }
      .cm-dark-inputNumber:focus {
        border: 1px solid #253459;
      }
      .cm-dark-inputNumber-inner-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber-inner-color:hover {
        color: #14224D;
      }
      .cm-dark-inputNumber-inner-border {
        border-color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-inputNumber-inner-disabled {
        color: #84878C;
        background-color: #d8cbcb;
      }
      .cm-dark-inputNumber-inner-disabled:hover {
        color: #84878C;
      }
      /*弹框下箭头：白色背景*/
      .cm-dark-popper-arrow-down {
        border-color: transparent;
        border-width: 6px;
        border-top-color: #9CA0A6;
        border-bottom-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-dark-popper-arrow-down:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        bottom: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-top-color: #fff;
        border-bottom-width: 0;
      }
      /*弹框上箭头：白色背景*/
      .cm-dark-popper-arrow-up {
        border-color: transparent;
        border-width: 6px;
        border-bottom-color: #9CA0A6;
        border-top-width: 0;
        border-style: solid;
        position: absolute;
      }
      .cm-dark-popper-arrow-up:after {
        content: '';
        position: absolute;
        display: block;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        top: 1px;
        margin-left: -6px;
        border-width: 6px;
        border-bottom-color: #fff;
        border-top-width: 0;
      }
      /*提示框下三角：黑色背景*/
      .cm-dark-tooltip-arrow-down {
        border-width: 6px;
        border-style: solid;
        border-top-color: #000;
        border-bottom-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框上三角：黑色背景*/
      .cm-dark-tooltip-arrow-up {
        border-width: 6px;
        border-style: solid;
        border-bottom-color: #000;
        border-top-width: 0;
        border-left-color: transparent;
        border-right-color: transparent;
      }
      /*提示框左三角：黑色背景*/
      .cm-dark-tooltip-arrow-left {
        border-width: 6px;
        border-style: solid;
        border-right-color: #000;
        border-left-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*提示框右三角：黑色背景*/
      .cm-dark-tooltip-arrow-right {
        border-width: 6px;
        border-style: solid;
        border-left-color: #000;
        border-right-width: 0;
        border-top-color: transparent;
        border-bottom-color: transparent;
      }
      /*timeline*/
      /*timeline 圆圈颜色*/
      .cm-dark-timeline-circle-blue {
        color: #14224D;
        border-color: #14224D;
      }
      .cm-dark-timeline-circle-green {
        color: #037343;
        border-color: #037343;
      }
      .cm-dark-timeline-circle-red {
        color: #A11500;
        border-color: #A11500;
      }
      /*timeline 字体颜色*/
      .cm-dark-timeline-text {
        color: #14224D;
      }
      /*timeline 图标颜色*/
      .cm-dark-timeline-icon {
        color: #A11500;
      }
      /*steps*/
      .cm-dark-step {
        border: 1px solid #9CA0A6;
        border-radius: 50%;
      }
      .cm-dark-step-finish {
        border-color: #253459;
        background-color: #fff;
      }
      .cm-dark-step-wait {
        border-color: #9CA0A6;
        background-color: #fff;
      }
      .cm-dark-step-progress {
        border-color: #9CA0A6;
        background-color: #253459;
      }
      .cm-dark-step-line-error {
        border-color: #A11500;
        background-color: #fff;
      }
      .cm-dark-step-line-wait {
        box-shadow: 0 0 0 0 #9CA0A6;
        background: #9CA0A6;
      }
      .cm-dark-step-line-progress {
        box-shadow: 0 0 0 0 #253459;
        background: #253459;
      }
      .cm-dark-step-color1 {
        color: #9CA0A6;
      }
      .cm-dark-step-color2 {
        color: #fff;
      }
      .cm-dark-step-color3 {
        color: #253459;
      }
      .cm-dark-step-color4 {
        color: #A11500;
      }
      .cm-dark-step-color5 {
        color: rgba(0, 0, 0, 0.56);
      }
      /*cascader*/
      .cm-dark-a-color {
        color: #14224D;
      }
      .cm-dark-a-color:hover {
        color: #253459;
      }
      /*breadcrumb*/
      .cm-dark-text-color {
        color: #d3d5d8;
      }
      .cm-dark-text-color:hover {
        color: #253459;
      }
      .cm-dark-start-color {
        color: #a4a5a7;
      }
      .cm-dark-interval-color {
        color: #9CA0A6;
      }
      .cm-dark-lasttext-color {
        color: #1c2438;
      }
      /*upload*/
      .cm-dark-texterror-color {
        color: #A11500;
      }
      .cm-dark-border-color {
        border-color: #9CA0A6;
      }
      .cm-dark-text-color {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-dark-color {
        color: #080F26;
      }
      .cm-dark-hover-color {
        border-color: #080F26;
      }
      .cm-dark-bg-color {
        background-color: #d8cbcb;
      }
      /*字体图标颜色*/
      .cm-icon-nav-1:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-nav-2:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-nav-3:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-arrow-down:before {
        color: #9CA0A6;
      }
      .cm-icon-basic-components:before {
        color: #9CA0A6;
      }
      .cm-icon-professional-components:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-checkbox-focus:before {
        color: #253459;
      }
      .cm-icon-functional-components:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-document:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-checkbox-default:before {
        color: rgba(0, 0, 0, 0.56);
      }
      .cm-icon-info:before {
        color: #0D6DA8;
      }
      .cm-icon-arrow-up:before {
        color: #9CA0A6;
      }
      .cm-icon-close:before {
        color: #9CA0A6;
      }
      .cm-icon-clock:before {
        color: #9CA0A6;
      }
      .cm-icon-arrow-left:before {
        color: #9CA0A6;
      }
      .cm-icon-radio-disabled:before {
        color: #84878C;
      }
      .cm-icon-arrow-right:before {
        color: #9CA0A6;
      }
      .cm-icon-not-all-checked:before {
        color: #14224D;
      }
      .cm-icon-radio-default:before {
        color: #9CA0A6;
      }
      .cm-icon-menu-fold:before {
        color: #9CA0A6;
      }
      .cm-icon-menu-unfold:before {
        color: #9CA0A6;
      }
      .cm-icon-radio-checked:before {
        color: #14224D;
      }
      .cm-icon-error:before {
        color: #A11500;
      }
      .cm-icon-disabled-checked:before {
        color: #84878C;
      }
      .cm-icon-warning:before {
        color: #B35F00;
      }
      .cm-icon-checkbox-checked:before {
        color: #14224D;
      }
      .cm-icon-success:before {
        color: #037343;
      }
      .theme-dark .cm-cascader-selection {
        background-color: #fff;
      }
      .ant-cascader-picker {
        position: relative;
        display: inline-block;
        cursor: pointer;
        font-size: 12px;
        background-color: #fff;
        border-radius: 4px;
        outline: 0;
      }
      .ant-cascader {
        display: block;
      }
      .ant-cascader-menus {
        font-size: 12px;
        background: #fff;
        position: absolute;
        z-index: 1050;
        border-radius: 4px;
        -webkit-box-shadow: 0 1px 6px rgba(0, 0, 0, 0.2);
        box-shadow: 0 1px 6px rgba(0, 0, 0, 0.2);
        white-space: nowrap;
      }
      .ant-cascader-menus {
        top: 100%;
        left: 0;
        position: relative;
        width: 100%;
        margin-top: 4px;
        margin-bottom: 4px;
      }
      .ant-cascader-menu {
        display: inline-block;
        vertical-align: top;
        min-width: 111px;
        height: 180px;
        list-style: none;
        margin: 0;
        padding: 0;
        border-right: 1px solid #e9e9e9;
        overflow: auto;
      }
      .ant-cascader-menu:first-child {
        border-radius: 4px 0 0 4px;
      }
      .ant-cascader-menu-item {
        padding: 7px 8px;
        cursor: pointer;
        white-space: nowrap;
        -webkit-transition: all .3s;
        transition: all .3s;
      }
      .ant-cascader-menu-item-expand {
        position: relative;
        padding-right: 24px;
      }
      .ant-cascader-menu-item-active:not(.ant-cascader-menu-item-disabled),
      .ant-cascader-menu-item-active:not(.ant-cascader-menu-item-disabled):hover {
        background: #f7f7f7;
        font-weight: 600;
      }
      .ant-cascader-menu-item-expand:after {
        font-family: anticon;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        content: "\\E61F";
        display: inline-block;
        font-size: 12px;
        font-size: 8px\\9;
        -webkit-transform: scale(0.66666667) rotate(0deg);
        transform: scale(0.66666667) rotate(0deg);
        -ms-filter: "progid:DXImageTransform.Microsoft.Matrix(sizingMethod='auto expand', M11=1, M12=0, M21=0, M22=1)";
        zoom: 1;
        color: rgba(0, 0, 0, 0.43);
        position: absolute;
        right: 8px;
      }
      .ant-cascader-menu-item-expand:after {
        -webkit-filter: none;
        filter: none;
        font-size: 12px;
      }
      .ant-input {
        position: relative;
        display: inline-block;
        padding: 4px 7px;
        width: 100%;
        height: 28px;
        font-size: 12px;
        line-height: 1.5;
        color: rgba(0, 0, 0, 0.65);
        background-color: #fff;
        background-image: none;
        border: 1px solid #d9d9d9;
        border-radius: 4px;
        -webkit-transition: all .3s;
        transition: all .3s;
      }
      .ant-input-sm {
        padding: 1px 7px;
        height: 22px;
      }
      .ant-input-lg {
        padding: 6px 7px;
        height: 32px;
      }
      .ant-cascader-input.ant-input {
        background-color: transparent!important;
        cursor: pointer;
        width: 100%;
        display: block;
      }
      .ant-cascader-picker:focus .ant-cascader-input {
        border-color: #49a9ee;
        outline: 0;
        -webkit-box-shadow: 0 0 0 2px rgba(16, 142, 233, 0.2);
        box-shadow: 0 0 0 2px rgba(16, 142, 233, 0.2);
      }
      .ant-cascader-picker-clear {
        opacity: 0;
        position: absolute;
        right: 8px;
        z-index: 2;
        background: #fff;
        top: 50%;
        font-size: 12px;
        color: rgba(0, 0, 0, 0.25);
        width: 12px;
        height: 12px;
        margin-top: -6px;
        line-height: 12px;
        cursor: pointer;
        -webkit-transition: color .3s ease,opacity .15s ease;
        transition: color .3s ease,opacity .15s ease;
      }
      .ant-cascader-picker-clear:hover {
        color: rgba(0, 0, 0, 0.43);
      }
      .ant-cascader-picker:hover,
      .ant-cascader-picker-clear {
        opacity: 1;
      }
      .ant-cascader-picker-arrow {
        position: absolute;
        z-index: 1;
        top: 50%;
        right: 8px;
        width: 12px;
        height: 12px;
        margin-top: -6px;
        line-height: 12px;
        color: rgba(0, 0, 0, 0.43);
        display: inline-block;
        font-size: 12px;
        font-size: 9px\\9;
        -webkit-transform: scale(0.75) rotate(0deg);
        transform: scale(0.75) rotate(0deg);
        -ms-filter: "progid:DXImageTransform.Microsoft.Matrix(sizingMethod='auto expand', M11=1, M12=0, M21=0, M22=1)";
        zoom: 1;
      }
      .ant-cascader-picker-arrow {
        -webkit-filter: none;
        filter: none;
        font-size: 12px;
      }
      .ant-cascader-picker-arrow::before {
        content: "\\E61D";
        display: block;
        font-family: anticon!important;
        -webkit-transition: -webkit-transform 0.2s ease;
        transition: -webkit-transform 0.2s ease;
        transition: transform .2s ease;
        transition: transform 0.2s ease, -webkit-transform 0.2s ease;
      }
      .ant-cascader-picker-arrow.ant-cascader-picker-arrow-expand:before {
        -webkit-transform: rotate(180deg);
        transform: rotate(180deg);
      }
      .ant-cascader-picker-label {
        position: absolute;
        left: 0;
        height: 20px;
        line-height: 20px;
        top: 50%;
        margin-top: -10px;
        white-space: nowrap;
        text-overflow: ellipsis;
        overflow: hidden;
        width: 100%;
        padding: 0 12px 0 8px;
      }
      .anticon-close-circle:before,
      .anticon-cross-circle:before {
        content: "\\E62E";
        display: block;
        font-family: anticon!important;
      }
    `],
        providers: [
            {
                provide: forms_1.NG_VALUE_ACCESSOR,
                useExisting: core_1.forwardRef(() => CheckCascaderComponent_1),
                multi: true
            }
        ]
    }),
    __metadata("design:paramtypes", [core_1.ElementRef,
        core_1.Renderer2,
        core_1.ChangeDetectorRef])
], CheckCascaderComponent);
exports.CheckCascaderComponent = CheckCascaderComponent;
