// ng-cosmos-td-ui/src/base/code-highlight/code-highlight.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const platform_browser_1 = require("@angular/platform-browser");
const hljs = require("highlight.js");
const Clipboard = require("clipboard");
let CodeHighlightComponent = class CodeHighlightComponent {
    constructor(el, sanitizer) {
        this.el = el;
        this.sanitizer = sanitizer;
        this.onCopy = new core_1.EventEmitter();
    }
    set hlString(hlString) {
        this._hlString = hlString;
        this.safeHtml = this.sanitizer.bypassSecurityTrustHtml(hljs.highlightAuto(this._hlString).value);
    }
    ngOnInit() {
        let self = this;
        let clipAction = this.el.nativeElement.querySelector('.clip-button');
        self.clipboard = new Clipboard(clipAction, {
            text: function (trigger) {
                return self._hlString;
            }
        });
        self.clipboard.on('success', function (e) {
            self.onCopy.emit(true);
        });
        self.clipboard.on('error', function (e) {
            self.onCopy.emit(false);
        });
    }
    ngOnDestroy() {
        this.clipboard.destroy();
    }
};
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], CodeHighlightComponent.prototype, "onCopy", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], CodeHighlightComponent.prototype, "hlString", null);
CodeHighlightComponent = __decorate([
    core_1.Component({
        selector: 'cm-code-highlight',
        template: `
      <main class="container">
          <div class="clip-content">
              <pre>
                  <code class="hljs" [innerHtml]="safeHtml">...</code>
              </pre>
          </div>
          <div class="clip-button"><i class="cm-iconfont cm-icon-document"></i></div>
      </main>
    `,
        styles: [`
      .container {
        display: flex;
        width: 100%;
        padding: 10px;
      }
      .clip-content {
        width: 100%;
      }
      .clip-button {
        width: 20px;
      }
      .hljs {
        display: block;
        overflow-x: auto;
        padding: 0px;
        color: #000;
        background: #ffffff;
      }
      .hljs-comment,
      .hljs-quote {
        color: #408080;
        font-style: italic;
      }
      .hljs-keyword,
      .hljs-selector-tag,
      .hljs-literal,
      .hljs-subst {
        color: #954121;
      }
      .hljs-number {
        color: #40a070;
      }
      .hljs-string,
      .hljs-doctag {
        color: #219161;
      }
      .hljs-selector-id,
      .hljs-selector-class,
      .hljs-section,
      .hljs-type {
        color: #19469d;
      }
      .hljs-params {
        color: #00f;
      }
      .hljs-title {
        color: #458;
        font-weight: bold;
      }
      .hljs-tag,
      .hljs-name,
      .hljs-attribute {
        color: #000080;
        font-weight: normal;
      }
      .hljs-variable,
      .hljs-template-variable {
        color: #008080;
      }
      .hljs-regexp,
      .hljs-link {
        color: #b68;
      }
      .hljs-symbol,
      .hljs-bullet {
        color: #990073;
      }
      .hljs-built_in,
      .hljs-builtin-name {
        color: #0086b3;
      }
      .hljs-meta {
        color: #999;
        font-weight: bold;
      }
      .hljs-deletion {
        background: #fdd;
      }
      .hljs-addition {
        background: #dfd;
      }
      .hljs-emphasis {
        font-style: italic;
      }
      .hljs-strong {
        font-weight: bold;
      }
    `]
    }),
    __metadata("design:paramtypes", [core_1.ElementRef,
        platform_browser_1.DomSanitizer])
], CodeHighlightComponent);
exports.CodeHighlightComponent = CodeHighlightComponent;
