// ng-cosmos-td-ui/src/base/cmtTable/cmthighlight.directive.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
/* tslint:disable:member-ordering */
const core_1 = require("@angular/core");
const cmTableFixex_service_1 = require("./cmTableFixex.service");
let RowHighlightDirective = class RowHighlightDirective {
    constructor(el, highlithService) {
        this.el = el;
        this.highlithService = highlithService;
        let that = this;
        that.highlithService.missionHighlight$.subscribe((highlight) => {
            if (highlight.index == that.rowHighlight.rowIndex) {
                if (highlight.isSelect) {
                    that.highlightColor(that.rowHighlight.hoverColor ? that.rowHighlight.hoverColor : "#ebf7ff");
                }
                else {
                    that.highlightColor("#ffffff");
                }
            }
        });
    }
    onMouseEnter() {
        this.highlithService.highlightMission({ isSelect: true, index: this.rowHighlight.rowIndex });
    }
    onMouseLeave() {
        this.highlithService.highlightMission({ isSelect: false, index: this.rowHighlight.rowIndex });
    }
    highlightColor(color) {
        this.el.nativeElement.style.backgroundColor = color;
    }
};
__decorate([
    core_1.Input('rowHighlight'),
    __metadata("design:type", Object)
], RowHighlightDirective.prototype, "rowHighlight", void 0);
__decorate([
    core_1.HostListener('mouseenter'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], RowHighlightDirective.prototype, "onMouseEnter", null);
__decorate([
    core_1.HostListener('mouseleave'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], RowHighlightDirective.prototype, "onMouseLeave", null);
RowHighlightDirective = __decorate([
    core_1.Directive({
        selector: '[rowHighlight]'
    }),
    __metadata("design:paramtypes", [core_1.ElementRef, cmTableFixex_service_1.CmTableFixedService])
], RowHighlightDirective);
exports.RowHighlightDirective = RowHighlightDirective;
/*
Copyright 2017 Google Inc. All Rights Reserved.
Use of this source code is governed by an MIT-style license that
can be found in the LICENSE file at http://angular.io/license
*/
