// ng-cosmos-td-ui/src/business/help/help.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
let HelpComponent = class HelpComponent {
    constructor() {
        this._cmMessage = '这不是帮助提示这不是帮助提示这不是帮助提示这不是帮助提示这不是帮助提示这不是帮助提示这不是帮助提示这不是帮助提示';
        this._cmPlace = 'top';
        this._cmOverlayStyle = {};
        this._cmTrigger = 'click';
    }
    set cmMessage(cmMessage) {
        this._cmMessage = cmMessage;
    }
    ;
    set cmPlace(cmPlace) {
        this._cmPlace = cmPlace;
    }
    ;
    set cmOverlayStyle(cmOverlayStyle) {
        this._cmOverlayStyle = cmOverlayStyle;
    }
    ;
    set cmTrigger(cmTrigger) {
        this._cmTrigger = cmTrigger;
    }
    ;
    ngOnInit() {
    }
};
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], HelpComponent.prototype, "cmMessage", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], HelpComponent.prototype, "cmPlace", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], HelpComponent.prototype, "cmOverlayStyle", null);
__decorate([
    core_1.Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], HelpComponent.prototype, "cmTrigger", null);
HelpComponent = __decorate([
    core_1.Component({
        selector: 'cm-help',
        template: `
      <ng-container>
          <nz-tooltip [nzTitle]="'prompt text'" [nzPlacement]="_cmPlace" [nzTrigger]="_cmTrigger" [nzOverlayStyle]="_cmOverlayStyle" [nzOverlayClassName]="'color'">
              <div>
                  <i class="anticon anticon-question-circle-o" nz-tooltip></i>
                  <span>帮助提示</span>
              </div>
              <ng-template #nzTemplate>
                  <span>{{_cmMessage}}</span>
              </ng-template>
          </nz-tooltip>
      </ng-container>
    `,
        styles: [`
      ::ng-deep .ant-tooltip-placement-top .ant-tooltip-arrow,
      ::ng-deep .ant-tooltip-placement-topLeft .ant-tooltip-arrow,
      ::ng-deep .ant-tooltip-placement-topRight .ant-tooltip-arrow {
        bottom: 3px;
        border-width: 5px 5px 0;
        border-top-color: rgba(0, 0, 0, 0.75) !important;
      }
      ::ng-deep .ant-tooltip-placement-bottom .ant-tooltip-arrow,
      ::ng-deep .ant-tooltip-placement-bottomLeft .ant-tooltip-arrow,
      ::ng-deep .ant-tooltip-placement-bottomRight .ant-tooltip-arrow {
        top: 3px;
        border-width: 0 5px 5px;
        border-bottom-color: rgba(0, 0, 0, 0.75) !important;
      }
      ::ng-deep .ant-tooltip-inner {
        max-width: 250px;
        padding: 8px 10px;
        color: #fff;
        text-align: left;
        text-decoration: none;
        background-color: rgba(0, 0, 0, 0.75) !important;
        border-radius: 4px;
        box-shadow: 0 1px 6px rgba(0, 0, 0, 0.2);
        min-height: 34px;
      }
      ::ng-deep .color {
        color: #333;
      }
    `]
    }),
    __metadata("design:paramtypes", [])
], HelpComponent);
exports.HelpComponent = HelpComponent;
